{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "d0955c26-f77a-4cb5-a157-e2187a8804a9",
   "metadata": {},
   "source": [
    "# INBRE Google Cloud Training Tutorials\n",
    "\n",
    "## Overview\n",
    "\n",
    "Included here are several tutorials in the form of 'Jupyter notebooks'.\n",
    "\n",
    "The purpose of these tutorials is to help users familiarize themselves with the Google Cloud environment in the specific context of running bioinformatics workflows.\n",
    "\n",
    "These tutorials do this by going step-by-step through specific workflows. These workflows cover the start to finish of basic bioinformatics analysis; starting from downloading raw sequence data, and extending to differential gene expression analysis, and producing common plots in R. \n",
    "\n",
    "## Workflows\n",
    "\n",
    "![RNA-Seq workflow](images/RNA-Seq_Notebook_Homepage.png)\n",
    "\n",
    "Each notebook covers a specific workflow, and contains a written and visual summary at its head for further information. Access each tutorial for more information:\n",
    "\n",
    "[Workflow One:](training_demo_short.ipynb) A short introduction to downloading and mapping sequences to a transcriptome using Trimmomatic and Salmon.\n",
    "\n",
    "[Workflow One (Snakemake):](training_demo_snakemake.ipynb) Using snakemake to run workflow one.\n",
    "\n",
    "[Extended Workflow One:](training_demo_extended.ipynb) An extended version of workflow one. Once you have got your feet wet, you can retry workflow one with this extended version that covers the entire dataset, and includes elaboration such as using SRA tools for sequence downloading, and examples of running batches of fastq files through the pipeline. This workflow may take around an hour to run.\n",
    "\n",
    "[Workflow Two:](r_training_demo.ipynb) Using Deseq2 and R to conduct clustering and differential gene expression analysis.\n",
    "\n",
    "\n",
    "\n",
    "## Requirements\n",
    "\n",
    "These piplines run entirely within the virtual machine environment of Google Cloud, and so should require nothing more than the ability to produce a virtual machine environment 'notebook' instance on the cloud. \n",
    "\n",
    "We chose for the framework of our virtual machines the basic Python 3 framework, for flexibility.\n",
    "\n",
    "## Data\n",
    "\n",
    "These tutorials use example sequence data procured from the Sally Molloy labratory here at University of Maine; which investigates the transcriptome change in prophage infected, versus non-prophage infected M. Chelonae bacteria. The respective article can be found <a href=\"https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8191103/\">here</a>."
   ]
  }
 ],
 "metadata": {
  "environment": {
   "kernel": "python3",
   "name": "common-cpu.m91",
   "type": "gcloud",
   "uri": "gcr.io/deeplearning-platform-release/base-cpu:m91"
  },
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
