{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {
    "tags": []
   },
   "source": [
    "# Extended RNA-Seq Analysis Training Demo"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Overview"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "For simplicity and time, The short tutorial workflow uses only partial run data from the Cushman et al., project.\n",
    "\n",
    "The tutorial repeats the short tutorial, but with extended steps.\n",
    "\n",
    "This tutorial demonstrates how to run an RNA-Seq workflow using a prokaryotic data set. Steps in the workflow include read trimming, read QC, read mapping, and counting mapped reads per gene to quantitate gene expression. Extended steps include using SRA tools to download the full dataset, joining gene count results, and formatting those results for use in R for further downstream analysis.\n",
    "\n",
    "For those seeking practice, we encourage you to create your own notebook and attempt to repeat through the full work flow yourself from scratch, using this extended tutorial as a reference as needed.\n",
    "\n",
    "![RNA-Seq workflow](images/rnaseq-workflow.png)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### STEP 1: Install Mambaforge and then install snakemake using bioconda."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "First install Mambaforge.\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "!curl -L -O https://github.com/conda-forge/miniforge/releases/latest/download/Mambaforge-$(uname)-$(uname -m).sh\n",
    "!bash Mambaforge-$(uname)-$(uname -m).sh -b -u -p $HOME/mambaforge\n",
    "!export PATH=\"$HOME/mambaforge/bin:$PATH\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Next, using mambaforge and bioconda, install the tools that will be used in this tutorial."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "!$HOME/mambaforge/bin/mamba install -y -c conda-forge -c bioconda trimmomatic\n",
    "!$HOME/mambaforge/bin/mamba install -y -c conda-forge -c bioconda fastqc\n",
    "!$HOME/mambaforge/bin/mamba install -y -c conda-forge -c bioconda multiqc\n",
    "!$HOME/mambaforge/bin/mamba install -y -c conda-forge -c bioconda salmon\n",
    "!$HOME/mambaforge/bin/mamba install -y -c conda-forge -c bioconda sra-tools\n",
    "!$HOME/mambaforge/bin/mamba install -y -c conda-forge -c bioconda parallel-fastq-dump\n",
    "!$HOME/mambaforge/bin/mamba install -y -c conda-forge -c bioconda entrez-direct\n",
    "!$HOME/mambaforge/bin/mamba install -y -c conda-forge -c bioconda gffread"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### STEP 2: Setup Environment"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Create a set of directories to store the reads, reference sequence files, and output files.\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "!cd $HOMEDIR\n",
    "!echo $PWD\n",
    "!mkdir -p data\n",
    "!mkdir -p data/raw_fastq\n",
    "!mkdir -p data/trimmed\n",
    "!mkdir -p data/fastqc\n",
    "!mkdir -p data/aligned\n",
    "!mkdir -p data/reference"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "tags": []
   },
   "source": [
    "### STEP 3: Downloading relevant FASTQ files using SRA Tools\n",
    "\n",
    "Next we will need to download the relevant fastq files.\n",
    "\n",
    "The sequence data for this tutorial comes from work by Cushman et al., <em><a href='https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8191103/'>Increased whiB7 expression and antibiotic resistance in Mycobacterium chelonae carrying two prophages</a><em>.\n",
    "\n",
    "We will be downloading the sample runs from this project using SRA tools, downloading from the NCBI's SRA (Sequence Run Archives).\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### STEP 3.1: Finding run accession numbers.\n",
    "\n",
    "The SRA stores sequence data in terms of runs, (SRR stands for Sequence Read Run). To download runs, we will need the accession ID for each run we wish to download. \n",
    "\n",
    "The Cushman et al., project contains 12 runs. To make it easier, these are the run IDs associated with this project:\n",
    "\n",
    "SRR13349122\n",
    "SRR13349123\n",
    "SRR13349124\n",
    "SRR13349125\n",
    "SRR13349126\n",
    "SRR13349127\n",
    "SRR13349128\n",
    "SRR13349129\n",
    "SRR13349130\n",
    "SRR13349131\n",
    "SRR13349132\n",
    "SRR13349133\n",
    "\n",
    "\n",
    "In this case, all these runs belong to the SRP (Sequence Run Project): SRP300216.\n",
    "\n",
    "Sequence run experiments can be searched for using the SRA database on the NCBI website; and article specific sample run information can be found in the supplementary section of that article.\n",
    "\n",
    "For instance, here, the the authors posted a link to the sequence data GSE (Gene Series number), <a href='https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE164210'>GSE164210</a>. This leads to the appropriate 'Gene Expression Omnibus' page where, among other useful files and information, the relevant SRA database link can be found. "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### STEP 3.2: Using the SRA-toolkit.\n",
    "\n",
    "Now use the Sequence Run accession ID to download the sequence data."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "2022-03-08T00:45:13 prefetch.2.11.0: 1) Downloading 'SRR13349122'...\n",
      "2022-03-08T00:45:13 prefetch.2.11.0:  Downloading via HTTPS...\n",
      "2022-03-08T00:45:31 prefetch.2.11.0:  HTTPS download succeed\n",
      "2022-03-08T00:45:33 prefetch.2.11.0:  'SRR13349122' is valid\n",
      "2022-03-08T00:45:33 prefetch.2.11.0: 1) 'SRR13349122' was downloaded successfully\n",
      "2022-03-08T00:45:33 prefetch.2.11.0: 'SRR13349122' has 0 unresolved dependencies\n"
     ]
    }
   ],
   "source": [
    "!prefetch  -O data/raw_fastq/ SRR13349122"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Notice the SRA archives sequence files in the SRA format. \n",
    "\n",
    "Typically genome workflows processed data in the form of zipped or unzipped .fastq, or .fasta files\n",
    "\n",
    "So before we move on, we need to convert the files from .sra to .fastq using the fastq-dump tool:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Read 10827590 spots for data/raw_fastq/SRR13349122\n",
      "Written 10827590 spots for data/raw_fastq/SRR13349122\n"
     ]
    }
   ],
   "source": [
    "#paired-end reads use the --split-files flag.\n",
    "#to save space, convert to a zipped fastq file with the --gzip flag.\n",
    "!fastq-dump -O data/raw_fastq/ --split-files --gzip data/raw_fastq/SRR13349122"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "tags": []
   },
   "source": [
    "### STEP 3.3 Downloading multiple files using the SRA-toolkit.\n",
    "\n",
    "One may, as in our case, wish to download multiple runs at once.\n",
    "\n",
    "To aid in this, SRA-tools supports batch downloading.\n",
    "\n",
    "We can download multiple SRA files using a single line of code by creating a list of the SRA IDs we wish to download, and inputting that into the prefetch command.\n",
    "\n",
    "For instance, one can first create a list of the 12 accession IDs for this project however they wish:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SRR13349122\n",
      "SRR13349123\n",
      "SRR13349124\n",
      "SRR13349125\n",
      "SRR13349126\n",
      "SRR13349127\n",
      "SRR13349128\n",
      "SRR13349129\n",
      "SRR13349130\n",
      "SRR13349131\n",
      "SRR13349132\n",
      "SRR13349133\n"
     ]
    }
   ],
   "source": [
    "!for i in {22..33}; do echo \"SRR133491$i\"; done > data/raw_fastq/list_of_accesionIDS.txt\n",
    "!cat data/raw_fastq/list_of_accesionIDS.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "And then feed that list into the sra-toolkit prefetch command"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "2022-03-08T01:04:26 prefetch.2.11.0: 1) 'SRR13349122' is found locally\n",
      "2022-03-08T01:04:26 prefetch.2.11.0: 'SRR13349122' has 0 unresolved dependencies\n",
      "\n",
      "2022-03-08T01:04:26 prefetch.2.11.0: 2) Downloading 'SRR13349123'...\n",
      "2022-03-08T01:04:26 prefetch.2.11.0:  Downloading via HTTPS...\n",
      "2022-03-08T01:04:45 prefetch.2.11.0:  HTTPS download succeed\n",
      "2022-03-08T01:04:46 prefetch.2.11.0:  'SRR13349123' is valid\n",
      "2022-03-08T01:04:46 prefetch.2.11.0: 2) 'SRR13349123' was downloaded successfully\n",
      "2022-03-08T01:04:46 prefetch.2.11.0: 'SRR13349123' has 0 unresolved dependencies\n",
      "\n",
      "2022-03-08T01:04:46 prefetch.2.11.0: 3) Downloading 'SRR13349124'...\n",
      "2022-03-08T01:04:46 prefetch.2.11.0:  Downloading via HTTPS...\n",
      "2022-03-08T01:05:07 prefetch.2.11.0:  HTTPS download succeed\n",
      "2022-03-08T01:05:08 prefetch.2.11.0:  'SRR13349124' is valid\n",
      "2022-03-08T01:05:08 prefetch.2.11.0: 3) 'SRR13349124' was downloaded successfully\n",
      "2022-03-08T01:05:08 prefetch.2.11.0: 'SRR13349124' has 0 unresolved dependencies\n",
      "\n",
      "2022-03-08T01:05:08 prefetch.2.11.0: 4) Downloading 'SRR13349125'...\n",
      "2022-03-08T01:05:08 prefetch.2.11.0:  Downloading via HTTPS...\n",
      "2022-03-08T01:05:26 prefetch.2.11.0:  HTTPS download succeed\n",
      "2022-03-08T01:05:27 prefetch.2.11.0:  'SRR13349125' is valid\n",
      "2022-03-08T01:05:27 prefetch.2.11.0: 4) 'SRR13349125' was downloaded successfully\n",
      "2022-03-08T01:05:27 prefetch.2.11.0: 'SRR13349125' has 0 unresolved dependencies\n",
      "\n",
      "2022-03-08T01:05:29 prefetch.2.11.0: 5) Downloading 'SRR13349126'...\n",
      "2022-03-08T01:05:29 prefetch.2.11.0:  Downloading via HTTPS...\n",
      "2022-03-08T01:05:54 prefetch.2.11.0:  HTTPS download succeed\n",
      "2022-03-08T01:05:56 prefetch.2.11.0:  'SRR13349126' is valid\n",
      "2022-03-08T01:05:56 prefetch.2.11.0: 5) 'SRR13349126' was downloaded successfully\n",
      "2022-03-08T01:05:56 prefetch.2.11.0: 'SRR13349126' has 0 unresolved dependencies\n",
      "\n",
      "2022-03-08T01:05:56 prefetch.2.11.0: 6) Downloading 'SRR13349127'...\n",
      "2022-03-08T01:05:56 prefetch.2.11.0:  Downloading via HTTPS...\n",
      "2022-03-08T01:06:17 prefetch.2.11.0:  HTTPS download succeed\n",
      "2022-03-08T01:06:18 prefetch.2.11.0:  'SRR13349127' is valid\n",
      "2022-03-08T01:06:18 prefetch.2.11.0: 6) 'SRR13349127' was downloaded successfully\n",
      "2022-03-08T01:06:18 prefetch.2.11.0: 'SRR13349127' has 0 unresolved dependencies\n",
      "\n",
      "2022-03-08T01:06:21 prefetch.2.11.0: 7) Downloading 'SRR13349128'...\n",
      "2022-03-08T01:06:21 prefetch.2.11.0:  Downloading via HTTPS...\n",
      "2022-03-08T01:06:45 prefetch.2.11.0:  HTTPS download succeed\n",
      "2022-03-08T01:06:47 prefetch.2.11.0:  'SRR13349128' is valid\n",
      "2022-03-08T01:06:47 prefetch.2.11.0: 7) 'SRR13349128' was downloaded successfully\n",
      "2022-03-08T01:06:47 prefetch.2.11.0: 'SRR13349128' has 0 unresolved dependencies\n",
      "\n",
      "2022-03-08T01:06:47 prefetch.2.11.0: 8) Downloading 'SRR13349129'...\n",
      "2022-03-08T01:06:47 prefetch.2.11.0:  Downloading via HTTPS...\n",
      "2022-03-08T01:07:08 prefetch.2.11.0:  HTTPS download succeed\n",
      "2022-03-08T01:07:10 prefetch.2.11.0:  'SRR13349129' is valid\n",
      "2022-03-08T01:07:10 prefetch.2.11.0: 8) 'SRR13349129' was downloaded successfully\n",
      "2022-03-08T01:07:10 prefetch.2.11.0: 'SRR13349129' has 0 unresolved dependencies\n",
      "\n",
      "2022-03-08T01:07:10 prefetch.2.11.0: 9) Downloading 'SRR13349130'...\n",
      "2022-03-08T01:07:10 prefetch.2.11.0:  Downloading via HTTPS...\n",
      "2022-03-08T01:07:27 prefetch.2.11.0:  HTTPS download succeed\n",
      "2022-03-08T01:07:28 prefetch.2.11.0:  'SRR13349130' is valid\n",
      "2022-03-08T01:07:28 prefetch.2.11.0: 9) 'SRR13349130' was downloaded successfully\n",
      "2022-03-08T01:07:28 prefetch.2.11.0: 'SRR13349130' has 0 unresolved dependencies\n",
      "\n",
      "2022-03-08T01:07:28 prefetch.2.11.0: 10) Downloading 'SRR13349131'...\n",
      "2022-03-08T01:07:28 prefetch.2.11.0:  Downloading via HTTPS...\n",
      "2022-03-08T01:07:45 prefetch.2.11.0:  HTTPS download succeed\n",
      "2022-03-08T01:07:47 prefetch.2.11.0:  'SRR13349131' is valid\n",
      "2022-03-08T01:07:47 prefetch.2.11.0: 10) 'SRR13349131' was downloaded successfully\n",
      "2022-03-08T01:07:47 prefetch.2.11.0: 'SRR13349131' has 0 unresolved dependencies\n",
      "\n",
      "2022-03-08T01:07:47 prefetch.2.11.0: 11) Downloading 'SRR13349132'...\n",
      "2022-03-08T01:07:47 prefetch.2.11.0:  Downloading via HTTPS...\n",
      "2022-03-08T01:08:06 prefetch.2.11.0:  HTTPS download succeed\n",
      "2022-03-08T01:08:08 prefetch.2.11.0:  'SRR13349132' is valid\n",
      "2022-03-08T01:08:08 prefetch.2.11.0: 11) 'SRR13349132' was downloaded successfully\n",
      "2022-03-08T01:08:08 prefetch.2.11.0: 'SRR13349132' has 0 unresolved dependencies\n",
      "\n",
      "2022-03-08T01:08:08 prefetch.2.11.0: 12) Downloading 'SRR13349133'...\n",
      "2022-03-08T01:08:08 prefetch.2.11.0:  Downloading via HTTPS...\n",
      "2022-03-08T01:08:27 prefetch.2.11.0:  HTTPS download succeed\n",
      "2022-03-08T01:08:28 prefetch.2.11.0:  'SRR13349133' is valid\n",
      "2022-03-08T01:08:28 prefetch.2.11.0: 12) 'SRR13349133' was downloaded successfully\n",
      "2022-03-08T01:08:28 prefetch.2.11.0: 'SRR13349133' has 0 unresolved dependencies\n"
     ]
    }
   ],
   "source": [
    "!prefetch -O data/raw_fastq/ --option-file data/raw_fastq/list_of_accesionIDS.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### STEP 3.3 Converting Multiple SRA files to Fastq\n",
    "\n",
    "Fastq-dump does not have native batch compatibility.\n",
    "\n",
    "There are various ways to get around this, for instance, commonly by using bash pipes or for-loops.\n",
    "\n",
    "One can also attempt to speed up the process of converting SRA to fastq files, by using fasterq-dump from the SRA-Tools library, or 'parallel-fastq-dump' from the separate parralel-fastq-dump library.\n",
    "\n",
    "There is more than one right way to do anything, but parallel-fastq-dump has the benefit of automatic zipping, whereas fasterq-dump does not have --gzip flag.\n",
    "\n",
    "Below is an example of one way to batch convert SRA to fastq files, piping in the same accession list used to download the SRAs with prefetech."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2022-03-08 01:12:14,434 - SRR ids: ['SRR13349122']\n",
      "2022-03-08 01:12:14,434 - extra args: ['--gzip', '--split-files']\n",
      "2022-03-08 01:12:14,434 - tempdir: ./pfd_6j6anfig\n",
      "2022-03-08 01:12:14,434 - CMD: sra-stat --meta --quick SRR13349122\n",
      "2022-03-08 01:12:15,307 - SRR13349122 spots: 10827590\n",
      "2022-03-08 01:12:15,307 - blocks: [[1, 1353448], [1353449, 2706896], [2706897, 4060344], [4060345, 5413792], [5413793, 6767240], [6767241, 8120688], [8120689, 9474136], [9474137, 10827590]]\n",
      "2022-03-08 01:12:15,308 - CMD: fastq-dump -N 1 -X 1353448 -O ./pfd_6j6anfig/0 --gzip --split-files SRR13349122\n",
      "2022-03-08 01:12:15,309 - CMD: fastq-dump -N 1353449 -X 2706896 -O ./pfd_6j6anfig/1 --gzip --split-files SRR13349122\n",
      "2022-03-08 01:12:15,310 - CMD: fastq-dump -N 2706897 -X 4060344 -O ./pfd_6j6anfig/2 --gzip --split-files SRR13349122\n",
      "2022-03-08 01:12:15,312 - CMD: fastq-dump -N 4060345 -X 5413792 -O ./pfd_6j6anfig/3 --gzip --split-files SRR13349122\n",
      "2022-03-08 01:12:15,314 - CMD: fastq-dump -N 5413793 -X 6767240 -O ./pfd_6j6anfig/4 --gzip --split-files SRR13349122\n",
      "2022-03-08 01:12:15,325 - CMD: fastq-dump -N 6767241 -X 8120688 -O ./pfd_6j6anfig/5 --gzip --split-files SRR13349122\n",
      "2022-03-08 01:12:15,343 - CMD: fastq-dump -N 8120689 -X 9474136 -O ./pfd_6j6anfig/6 --gzip --split-files SRR13349122\n",
      "2022-03-08 01:12:15,358 - CMD: fastq-dump -N 9474137 -X 10827590 -O ./pfd_6j6anfig/7 --gzip --split-files SRR13349122\n",
      "Read 1353448 spots for SRR13349122\n",
      "Written 1353448 spots for SRR13349122\n",
      "Read 1353448 spots for SRR13349122\n",
      "Written 1353448 spots for SRR13349122\n",
      "Read 1353448 spots for SRR13349122\n",
      "Written 1353448 spots for SRR13349122\n",
      "Read 1353448 spots for SRR13349122\n",
      "Written 1353448 spots for SRR13349122\n",
      "Read 1353454 spots for SRR13349122\n",
      "Written 1353454 spots for SRR13349122\n",
      "Read 1353448 spots for SRR13349122\n",
      "Written 1353448 spots for SRR13349122\n",
      "Read 1353448 spots for SRR13349122\n",
      "Written 1353448 spots for SRR13349122\n",
      "Read 1353448 spots for SRR13349122\n",
      "Written 1353448 spots for SRR13349122\n",
      "2022-03-08 01:14:27,801 - SRR ids: ['SRR13349123']\n",
      "2022-03-08 01:14:27,802 - extra args: ['--gzip', '--split-files']\n",
      "2022-03-08 01:14:27,802 - tempdir: ./pfd_sb4vn3nr\n",
      "2022-03-08 01:14:27,802 - CMD: sra-stat --meta --quick SRR13349123\n",
      "2022-03-08 01:14:28,529 - SRR13349123 spots: 11165256\n",
      "2022-03-08 01:14:28,530 - blocks: [[1, 1395657], [1395658, 2791314], [2791315, 4186971], [4186972, 5582628], [5582629, 6978285], [6978286, 8373942], [8373943, 9769599], [9769600, 11165256]]\n",
      "2022-03-08 01:14:28,530 - CMD: fastq-dump -N 1 -X 1395657 -O ./pfd_sb4vn3nr/0 --gzip --split-files SRR13349123\n",
      "2022-03-08 01:14:28,542 - CMD: fastq-dump -N 1395658 -X 2791314 -O ./pfd_sb4vn3nr/1 --gzip --split-files SRR13349123\n",
      "2022-03-08 01:14:28,543 - CMD: fastq-dump -N 2791315 -X 4186971 -O ./pfd_sb4vn3nr/2 --gzip --split-files SRR13349123\n",
      "2022-03-08 01:14:28,545 - CMD: fastq-dump -N 4186972 -X 5582628 -O ./pfd_sb4vn3nr/3 --gzip --split-files SRR13349123\n",
      "2022-03-08 01:14:28,546 - CMD: fastq-dump -N 5582629 -X 6978285 -O ./pfd_sb4vn3nr/4 --gzip --split-files SRR13349123\n",
      "2022-03-08 01:14:28,564 - CMD: fastq-dump -N 6978286 -X 8373942 -O ./pfd_sb4vn3nr/5 --gzip --split-files SRR13349123\n",
      "2022-03-08 01:14:28,576 - CMD: fastq-dump -N 8373943 -X 9769599 -O ./pfd_sb4vn3nr/6 --gzip --split-files SRR13349123\n",
      "2022-03-08 01:14:28,584 - CMD: fastq-dump -N 9769600 -X 11165256 -O ./pfd_sb4vn3nr/7 --gzip --split-files SRR13349123\n",
      "Read 1395657 spots for SRR13349123\n",
      "Written 1395657 spots for SRR13349123\n",
      "Read 1395657 spots for SRR13349123\n",
      "Written 1395657 spots for SRR13349123\n",
      "Read 1395657 spots for SRR13349123\n",
      "Written 1395657 spots for SRR13349123\n",
      "Read 1395657 spots for SRR13349123\n",
      "Written 1395657 spots for SRR13349123\n",
      "Read 1395657 spots for SRR13349123\n",
      "Written 1395657 spots for SRR13349123\n",
      "Read 1395657 spots for SRR13349123\n",
      "Written 1395657 spots for SRR13349123\n",
      "Read 1395657 spots for SRR13349123\n",
      "Written 1395657 spots for SRR13349123\n",
      "Read 1395657 spots for SRR13349123\n",
      "Written 1395657 spots for SRR13349123\n",
      "2022-03-08 01:16:34,041 - SRR ids: ['SRR13349124']\n",
      "2022-03-08 01:16:34,041 - extra args: ['--gzip', '--split-files']\n",
      "2022-03-08 01:16:34,041 - tempdir: ./pfd_40gvj9re\n",
      "2022-03-08 01:16:34,041 - CMD: sra-stat --meta --quick SRR13349124\n",
      "2022-03-08 01:16:34,886 - SRR13349124 spots: 10727273\n",
      "2022-03-08 01:16:34,887 - blocks: [[1, 1340909], [1340910, 2681818], [2681819, 4022727], [4022728, 5363636], [5363637, 6704545], [6704546, 8045454], [8045455, 9386363], [9386364, 10727273]]\n",
      "2022-03-08 01:16:34,887 - CMD: fastq-dump -N 1 -X 1340909 -O ./pfd_40gvj9re/0 --gzip --split-files SRR13349124\n",
      "2022-03-08 01:16:34,888 - CMD: fastq-dump -N 1340910 -X 2681818 -O ./pfd_40gvj9re/1 --gzip --split-files SRR13349124\n",
      "2022-03-08 01:16:34,889 - CMD: fastq-dump -N 2681819 -X 4022727 -O ./pfd_40gvj9re/2 --gzip --split-files SRR13349124\n",
      "2022-03-08 01:16:34,891 - CMD: fastq-dump -N 4022728 -X 5363636 -O ./pfd_40gvj9re/3 --gzip --split-files SRR13349124\n",
      "2022-03-08 01:16:34,893 - CMD: fastq-dump -N 5363637 -X 6704545 -O ./pfd_40gvj9re/4 --gzip --split-files SRR13349124\n",
      "2022-03-08 01:16:34,909 - CMD: fastq-dump -N 6704546 -X 8045454 -O ./pfd_40gvj9re/5 --gzip --split-files SRR13349124\n",
      "2022-03-08 01:16:34,924 - CMD: fastq-dump -N 8045455 -X 9386363 -O ./pfd_40gvj9re/6 --gzip --split-files SRR13349124\n",
      "2022-03-08 01:16:34,932 - CMD: fastq-dump -N 9386364 -X 10727273 -O ./pfd_40gvj9re/7 --gzip --split-files SRR13349124\n",
      "Read 1340909 spots for SRR13349124\n",
      "Written 1340909 spots for SRR13349124\n",
      "Read 1340909 spots for SRR13349124\n",
      "Written 1340909 spots for SRR13349124\n",
      "Read 1340909 spots for SRR13349124\n",
      "Written 1340909 spots for SRR13349124\n",
      "Read 1340910 spots for SRR13349124\n",
      "Written 1340910 spots for SRR13349124\n",
      "Read 1340909 spots for SRR13349124\n",
      "Written 1340909 spots for SRR13349124\n",
      "Read 1340909 spots for SRR13349124\n",
      "Written 1340909 spots for SRR13349124\n",
      "Read 1340909 spots for SRR13349124\n",
      "Written 1340909 spots for SRR13349124\n",
      "Read 1340909 spots for SRR13349124\n",
      "Written 1340909 spots for SRR13349124\n",
      "2022-03-08 01:18:38,268 - SRR ids: ['SRR13349125']\n",
      "2022-03-08 01:18:38,268 - extra args: ['--gzip', '--split-files']\n",
      "2022-03-08 01:18:38,268 - tempdir: ./pfd__9k79mvi\n",
      "2022-03-08 01:18:38,269 - CMD: sra-stat --meta --quick SRR13349125\n",
      "2022-03-08 01:18:38,764 - SRR13349125 spots: 10992686\n",
      "2022-03-08 01:18:38,764 - blocks: [[1, 1374085], [1374086, 2748170], [2748171, 4122255], [4122256, 5496340], [5496341, 6870425], [6870426, 8244510], [8244511, 9618595], [9618596, 10992686]]\n",
      "2022-03-08 01:18:38,764 - CMD: fastq-dump -N 1 -X 1374085 -O ./pfd__9k79mvi/0 --gzip --split-files SRR13349125\n",
      "2022-03-08 01:18:38,766 - CMD: fastq-dump -N 1374086 -X 2748170 -O ./pfd__9k79mvi/1 --gzip --split-files SRR13349125\n",
      "2022-03-08 01:18:38,768 - CMD: fastq-dump -N 2748171 -X 4122255 -O ./pfd__9k79mvi/2 --gzip --split-files SRR13349125\n",
      "2022-03-08 01:18:38,769 - CMD: fastq-dump -N 4122256 -X 5496340 -O ./pfd__9k79mvi/3 --gzip --split-files SRR13349125\n",
      "2022-03-08 01:18:38,771 - CMD: fastq-dump -N 5496341 -X 6870425 -O ./pfd__9k79mvi/4 --gzip --split-files SRR13349125\n",
      "2022-03-08 01:18:38,785 - CMD: fastq-dump -N 6870426 -X 8244510 -O ./pfd__9k79mvi/5 --gzip --split-files SRR13349125\n",
      "2022-03-08 01:18:38,801 - CMD: fastq-dump -N 8244511 -X 9618595 -O ./pfd__9k79mvi/6 --gzip --split-files SRR13349125\n",
      "2022-03-08 01:18:38,808 - CMD: fastq-dump -N 9618596 -X 10992686 -O ./pfd__9k79mvi/7 --gzip --split-files SRR13349125\n",
      "Read 1374085 spots for SRR13349125\n",
      "Written 1374085 spots for SRR13349125\n",
      "Read 1374085 spots for SRR13349125\n",
      "Written 1374085 spots for SRR13349125\n",
      "Read 1374085 spots for SRR13349125\n",
      "Written 1374085 spots for SRR13349125\n",
      "Read 1374085 spots for SRR13349125\n",
      "Written 1374085 spots for SRR13349125\n",
      "Read 1374091 spots for SRR13349125\n",
      "Written 1374091 spots for SRR13349125\n",
      "Read 1374085 spots for SRR13349125\n",
      "Written 1374085 spots for SRR13349125\n",
      "Read 1374085 spots for SRR13349125\n",
      "Written 1374085 spots for SRR13349125\n",
      "Read 1374085 spots for SRR13349125\n",
      "Written 1374085 spots for SRR13349125\n",
      "2022-03-08 01:20:41,977 - SRR ids: ['SRR13349126']\n",
      "2022-03-08 01:20:41,977 - extra args: ['--gzip', '--split-files']\n",
      "2022-03-08 01:20:41,977 - tempdir: ./pfd_32rixkl8\n",
      "2022-03-08 01:20:41,977 - CMD: sra-stat --meta --quick SRR13349126\n",
      "2022-03-08 01:20:42,518 - SRR13349126 spots: 12267497\n",
      "2022-03-08 01:20:42,518 - blocks: [[1, 1533437], [1533438, 3066874], [3066875, 4600311], [4600312, 6133748], [6133749, 7667185], [7667186, 9200622], [9200623, 10734059], [10734060, 12267497]]\n",
      "2022-03-08 01:20:42,518 - CMD: fastq-dump -N 1 -X 1533437 -O ./pfd_32rixkl8/0 --gzip --split-files SRR13349126\n",
      "2022-03-08 01:20:42,520 - CMD: fastq-dump -N 1533438 -X 3066874 -O ./pfd_32rixkl8/1 --gzip --split-files SRR13349126\n",
      "2022-03-08 01:20:42,521 - CMD: fastq-dump -N 3066875 -X 4600311 -O ./pfd_32rixkl8/2 --gzip --split-files SRR13349126\n",
      "2022-03-08 01:20:42,523 - CMD: fastq-dump -N 4600312 -X 6133748 -O ./pfd_32rixkl8/3 --gzip --split-files SRR13349126\n",
      "2022-03-08 01:20:42,525 - CMD: fastq-dump -N 6133749 -X 7667185 -O ./pfd_32rixkl8/4 --gzip --split-files SRR13349126\n",
      "2022-03-08 01:20:42,545 - CMD: fastq-dump -N 7667186 -X 9200622 -O ./pfd_32rixkl8/5 --gzip --split-files SRR13349126\n",
      "2022-03-08 01:20:42,556 - CMD: fastq-dump -N 9200623 -X 10734059 -O ./pfd_32rixkl8/6 --gzip --split-files SRR13349126\n",
      "2022-03-08 01:20:42,558 - CMD: fastq-dump -N 10734060 -X 12267497 -O ./pfd_32rixkl8/7 --gzip --split-files SRR13349126\n",
      "Read 1533437 spots for SRR13349126\n",
      "Written 1533437 spots for SRR13349126\n",
      "Read 1533437 spots for SRR13349126\n",
      "Written 1533437 spots for SRR13349126\n",
      "Read 1533437 spots for SRR13349126\n",
      "Written 1533437 spots for SRR13349126\n",
      "Read 1533437 spots for SRR13349126\n",
      "Written 1533437 spots for SRR13349126\n",
      "Read 1533437 spots for SRR13349126\n",
      "Written 1533437 spots for SRR13349126\n",
      "Read 1533438 spots for SRR13349126\n",
      "Written 1533438 spots for SRR13349126\n",
      "Read 1533437 spots for SRR13349126\n",
      "Written 1533437 spots for SRR13349126\n",
      "Read 1533437 spots for SRR13349126\n",
      "Written 1533437 spots for SRR13349126\n",
      "2022-03-08 01:23:01,413 - SRR ids: ['SRR13349127']\n",
      "2022-03-08 01:23:01,413 - extra args: ['--gzip', '--split-files']\n",
      "2022-03-08 01:23:01,413 - tempdir: ./pfd_qveyrq4n\n",
      "2022-03-08 01:23:01,413 - CMD: sra-stat --meta --quick SRR13349127\n",
      "2022-03-08 01:23:02,085 - SRR13349127 spots: 12563032\n",
      "2022-03-08 01:23:02,085 - blocks: [[1, 1570379], [1570380, 3140758], [3140759, 4711137], [4711138, 6281516], [6281517, 7851895], [7851896, 9422274], [9422275, 10992653], [10992654, 12563032]]\n",
      "2022-03-08 01:23:02,085 - CMD: fastq-dump -N 1 -X 1570379 -O ./pfd_qveyrq4n/0 --gzip --split-files SRR13349127\n",
      "2022-03-08 01:23:02,087 - CMD: fastq-dump -N 1570380 -X 3140758 -O ./pfd_qveyrq4n/1 --gzip --split-files SRR13349127\n",
      "2022-03-08 01:23:02,089 - CMD: fastq-dump -N 3140759 -X 4711137 -O ./pfd_qveyrq4n/2 --gzip --split-files SRR13349127\n",
      "2022-03-08 01:23:02,090 - CMD: fastq-dump -N 4711138 -X 6281516 -O ./pfd_qveyrq4n/3 --gzip --split-files SRR13349127\n",
      "2022-03-08 01:23:02,092 - CMD: fastq-dump -N 6281517 -X 7851895 -O ./pfd_qveyrq4n/4 --gzip --split-files SRR13349127\n",
      "2022-03-08 01:23:02,105 - CMD: fastq-dump -N 7851896 -X 9422274 -O ./pfd_qveyrq4n/5 --gzip --split-files SRR13349127\n",
      "2022-03-08 01:23:02,113 - CMD: fastq-dump -N 9422275 -X 10992653 -O ./pfd_qveyrq4n/6 --gzip --split-files SRR13349127\n",
      "2022-03-08 01:23:02,123 - CMD: fastq-dump -N 10992654 -X 12563032 -O ./pfd_qveyrq4n/7 --gzip --split-files SRR13349127\n",
      "Read 1570379 spots for SRR13349127\n",
      "Written 1570379 spots for SRR13349127\n",
      "Read 1570379 spots for SRR13349127\n",
      "Written 1570379 spots for SRR13349127\n",
      "Read 1570379 spots for SRR13349127\n",
      "Written 1570379 spots for SRR13349127\n",
      "Read 1570379 spots for SRR13349127\n",
      "Written 1570379 spots for SRR13349127\n",
      "Read 1570379 spots for SRR13349127\n",
      "Written 1570379 spots for SRR13349127\n",
      "Read 1570379 spots for SRR13349127\n",
      "Written 1570379 spots for SRR13349127\n",
      "Read 1570379 spots for SRR13349127\n",
      "Written 1570379 spots for SRR13349127\n",
      "Read 1570379 spots for SRR13349127\n",
      "Written 1570379 spots for SRR13349127\n",
      "2022-03-08 01:25:20,505 - SRR ids: ['SRR13349128']\n",
      "2022-03-08 01:25:20,505 - extra args: ['--gzip', '--split-files']\n",
      "2022-03-08 01:25:20,506 - tempdir: ./pfd_ubd_vx2e\n",
      "2022-03-08 01:25:20,506 - CMD: sra-stat --meta --quick SRR13349128\n",
      "2022-03-08 01:25:21,003 - SRR13349128 spots: 12652387\n",
      "2022-03-08 01:25:21,003 - blocks: [[1, 1581548], [1581549, 3163096], [3163097, 4744644], [4744645, 6326192], [6326193, 7907740], [7907741, 9489288], [9489289, 11070836], [11070837, 12652387]]\n",
      "2022-03-08 01:25:21,003 - CMD: fastq-dump -N 1 -X 1581548 -O ./pfd_ubd_vx2e/0 --gzip --split-files SRR13349128\n",
      "2022-03-08 01:25:21,005 - CMD: fastq-dump -N 1581549 -X 3163096 -O ./pfd_ubd_vx2e/1 --gzip --split-files SRR13349128\n",
      "2022-03-08 01:25:21,006 - CMD: fastq-dump -N 3163097 -X 4744644 -O ./pfd_ubd_vx2e/2 --gzip --split-files SRR13349128\n",
      "2022-03-08 01:25:21,008 - CMD: fastq-dump -N 4744645 -X 6326192 -O ./pfd_ubd_vx2e/3 --gzip --split-files SRR13349128\n",
      "2022-03-08 01:25:21,010 - CMD: fastq-dump -N 6326193 -X 7907740 -O ./pfd_ubd_vx2e/4 --gzip --split-files SRR13349128\n",
      "2022-03-08 01:25:21,021 - CMD: fastq-dump -N 7907741 -X 9489288 -O ./pfd_ubd_vx2e/5 --gzip --split-files SRR13349128\n",
      "2022-03-08 01:25:21,033 - CMD: fastq-dump -N 9489289 -X 11070836 -O ./pfd_ubd_vx2e/6 --gzip --split-files SRR13349128\n",
      "2022-03-08 01:25:21,041 - CMD: fastq-dump -N 11070837 -X 12652387 -O ./pfd_ubd_vx2e/7 --gzip --split-files SRR13349128\n",
      "Read 1581548 spots for SRR13349128\n",
      "Written 1581548 spots for SRR13349128\n",
      "Read 1581548 spots for SRR13349128\n",
      "Written 1581548 spots for SRR13349128\n",
      "Read 1581548 spots for SRR13349128\n",
      "Written 1581548 spots for SRR13349128\n",
      "Read 1581548 spots for SRR13349128\n",
      "Written 1581548 spots for SRR13349128\n",
      "Read 1581551 spots for SRR13349128\n",
      "Written 1581551 spots for SRR13349128\n",
      "Read 1581548 spots for SRR13349128\n",
      "Written 1581548 spots for SRR13349128\n",
      "Read 1581548 spots for SRR13349128\n",
      "Written 1581548 spots for SRR13349128\n",
      "Read 1581548 spots for SRR13349128\n",
      "Written 1581548 spots for SRR13349128\n",
      "2022-03-08 01:27:43,628 - SRR ids: ['SRR13349129']\n",
      "2022-03-08 01:27:43,628 - extra args: ['--gzip', '--split-files']\n",
      "2022-03-08 01:27:43,629 - tempdir: ./pfd_yheb20cg\n",
      "2022-03-08 01:27:43,629 - CMD: sra-stat --meta --quick SRR13349129\n",
      "2022-03-08 01:27:44,153 - SRR13349129 spots: 12961793\n",
      "2022-03-08 01:27:44,153 - blocks: [[1, 1620224], [1620225, 3240448], [3240449, 4860672], [4860673, 6480896], [6480897, 8101120], [8101121, 9721344], [9721345, 11341568], [11341569, 12961793]]\n",
      "2022-03-08 01:27:44,153 - CMD: fastq-dump -N 1 -X 1620224 -O ./pfd_yheb20cg/0 --gzip --split-files SRR13349129\n",
      "2022-03-08 01:27:44,155 - CMD: fastq-dump -N 1620225 -X 3240448 -O ./pfd_yheb20cg/1 --gzip --split-files SRR13349129\n",
      "2022-03-08 01:27:44,157 - CMD: fastq-dump -N 3240449 -X 4860672 -O ./pfd_yheb20cg/2 --gzip --split-files SRR13349129\n",
      "2022-03-08 01:27:44,158 - CMD: fastq-dump -N 4860673 -X 6480896 -O ./pfd_yheb20cg/3 --gzip --split-files SRR13349129\n",
      "2022-03-08 01:27:44,160 - CMD: fastq-dump -N 6480897 -X 8101120 -O ./pfd_yheb20cg/4 --gzip --split-files SRR13349129\n",
      "2022-03-08 01:27:44,177 - CMD: fastq-dump -N 8101121 -X 9721344 -O ./pfd_yheb20cg/5 --gzip --split-files SRR13349129\n",
      "2022-03-08 01:27:44,192 - CMD: fastq-dump -N 9721345 -X 11341568 -O ./pfd_yheb20cg/6 --gzip --split-files SRR13349129\n",
      "2022-03-08 01:27:44,200 - CMD: fastq-dump -N 11341569 -X 12961793 -O ./pfd_yheb20cg/7 --gzip --split-files SRR13349129\n",
      "Read 1620224 spots for SRR13349129\n",
      "Written 1620224 spots for SRR13349129\n",
      "Read 1620224 spots for SRR13349129\n",
      "Written 1620224 spots for SRR13349129\n",
      "Read 1620224 spots for SRR13349129\n",
      "Written 1620224 spots for SRR13349129\n",
      "Read 1620224 spots for SRR13349129\n",
      "Written 1620224 spots for SRR13349129\n",
      "Read 1620224 spots for SRR13349129\n",
      "Written 1620224 spots for SRR13349129\n",
      "Read 1620224 spots for SRR13349129\n",
      "Written 1620224 spots for SRR13349129\n",
      "Read 1620225 spots for SRR13349129\n",
      "Written 1620225 spots for SRR13349129\n",
      "Read 1620224 spots for SRR13349129\n",
      "Written 1620224 spots for SRR13349129\n",
      "2022-03-08 01:30:06,539 - SRR ids: ['SRR13349130']\n",
      "2022-03-08 01:30:06,540 - extra args: ['--gzip', '--split-files']\n",
      "2022-03-08 01:30:06,540 - tempdir: ./pfd_hv6y6lr8\n",
      "2022-03-08 01:30:06,540 - CMD: sra-stat --meta --quick SRR13349130\n",
      "2022-03-08 01:30:07,085 - SRR13349130 spots: 10083015\n",
      "2022-03-08 01:30:07,085 - blocks: [[1, 1260376], [1260377, 2520752], [2520753, 3781128], [3781129, 5041504], [5041505, 6301880], [6301881, 7562256], [7562257, 8822632], [8822633, 10083015]]\n",
      "2022-03-08 01:30:07,085 - CMD: fastq-dump -N 1 -X 1260376 -O ./pfd_hv6y6lr8/0 --gzip --split-files SRR13349130\n",
      "2022-03-08 01:30:07,086 - CMD: fastq-dump -N 1260377 -X 2520752 -O ./pfd_hv6y6lr8/1 --gzip --split-files SRR13349130\n",
      "2022-03-08 01:30:07,088 - CMD: fastq-dump -N 2520753 -X 3781128 -O ./pfd_hv6y6lr8/2 --gzip --split-files SRR13349130\n",
      "2022-03-08 01:30:07,090 - CMD: fastq-dump -N 3781129 -X 5041504 -O ./pfd_hv6y6lr8/3 --gzip --split-files SRR13349130\n",
      "2022-03-08 01:30:07,091 - CMD: fastq-dump -N 5041505 -X 6301880 -O ./pfd_hv6y6lr8/4 --gzip --split-files SRR13349130\n",
      "2022-03-08 01:30:07,105 - CMD: fastq-dump -N 6301881 -X 7562256 -O ./pfd_hv6y6lr8/5 --gzip --split-files SRR13349130\n",
      "2022-03-08 01:30:07,121 - CMD: fastq-dump -N 7562257 -X 8822632 -O ./pfd_hv6y6lr8/6 --gzip --split-files SRR13349130\n",
      "2022-03-08 01:30:07,125 - CMD: fastq-dump -N 8822633 -X 10083015 -O ./pfd_hv6y6lr8/7 --gzip --split-files SRR13349130\n",
      "Read 1260376 spots for SRR13349130\n",
      "Written 1260376 spots for SRR13349130\n",
      "Read 1260376 spots for SRR13349130\n",
      "Written 1260376 spots for SRR13349130\n",
      "Read 1260376 spots for SRR13349130\n",
      "Written 1260376 spots for SRR13349130\n",
      "Read 1260376 spots for SRR13349130\n",
      "Written 1260376 spots for SRR13349130\n",
      "Read 1260376 spots for SRR13349130\n",
      "Written 1260376 spots for SRR13349130\n",
      "Read 1260376 spots for SRR13349130\n",
      "Written 1260376 spots for SRR13349130\n",
      "Read 1260376 spots for SRR13349130\n",
      "Written 1260376 spots for SRR13349130\n",
      "Read 1260383 spots for SRR13349130\n",
      "Written 1260383 spots for SRR13349130\n",
      "2022-03-08 01:32:01,143 - SRR ids: ['SRR13349131']\n",
      "2022-03-08 01:32:01,143 - extra args: ['--gzip', '--split-files']\n",
      "2022-03-08 01:32:01,143 - tempdir: ./pfd_djfmj2ml\n",
      "2022-03-08 01:32:01,143 - CMD: sra-stat --meta --quick SRR13349131\n",
      "2022-03-08 01:32:01,795 - SRR13349131 spots: 10491160\n",
      "2022-03-08 01:32:01,795 - blocks: [[1, 1311395], [1311396, 2622790], [2622791, 3934185], [3934186, 5245580], [5245581, 6556975], [6556976, 7868370], [7868371, 9179765], [9179766, 10491160]]\n",
      "2022-03-08 01:32:01,795 - CMD: fastq-dump -N 1 -X 1311395 -O ./pfd_djfmj2ml/0 --gzip --split-files SRR13349131\n",
      "2022-03-08 01:32:01,797 - CMD: fastq-dump -N 1311396 -X 2622790 -O ./pfd_djfmj2ml/1 --gzip --split-files SRR13349131\n",
      "2022-03-08 01:32:01,798 - CMD: fastq-dump -N 2622791 -X 3934185 -O ./pfd_djfmj2ml/2 --gzip --split-files SRR13349131\n",
      "2022-03-08 01:32:01,800 - CMD: fastq-dump -N 3934186 -X 5245580 -O ./pfd_djfmj2ml/3 --gzip --split-files SRR13349131\n",
      "2022-03-08 01:32:01,802 - CMD: fastq-dump -N 5245581 -X 6556975 -O ./pfd_djfmj2ml/4 --gzip --split-files SRR13349131\n",
      "2022-03-08 01:32:01,821 - CMD: fastq-dump -N 6556976 -X 7868370 -O ./pfd_djfmj2ml/5 --gzip --split-files SRR13349131\n",
      "2022-03-08 01:32:01,833 - CMD: fastq-dump -N 7868371 -X 9179765 -O ./pfd_djfmj2ml/6 --gzip --split-files SRR13349131\n",
      "2022-03-08 01:32:01,843 - CMD: fastq-dump -N 9179766 -X 10491160 -O ./pfd_djfmj2ml/7 --gzip --split-files SRR13349131\n",
      "Read 1311395 spots for SRR13349131\n",
      "Written 1311395 spots for SRR13349131\n",
      "Read 1311395 spots for SRR13349131\n",
      "Written 1311395 spots for SRR13349131\n",
      "Read 1311395 spots for SRR13349131\n",
      "Written 1311395 spots for SRR13349131\n",
      "Read 1311395 spots for SRR13349131\n",
      "Written 1311395 spots for SRR13349131\n",
      "Read 1311395 spots for SRR13349131\n",
      "Written 1311395 spots for SRR13349131\n",
      "Read 1311395 spots for SRR13349131\n",
      "Written 1311395 spots for SRR13349131\n",
      "Read 1311395 spots for SRR13349131\n",
      "Written 1311395 spots for SRR13349131\n",
      "Read 1311395 spots for SRR13349131\n",
      "Written 1311395 spots for SRR13349131\n",
      "2022-03-08 01:33:58,441 - SRR ids: ['SRR13349132']\n",
      "2022-03-08 01:33:58,442 - extra args: ['--gzip', '--split-files']\n",
      "2022-03-08 01:33:58,442 - tempdir: ./pfd_4r3ydpaq\n",
      "2022-03-08 01:33:58,442 - CMD: sra-stat --meta --quick SRR13349132\n",
      "2022-03-08 01:33:58,956 - SRR13349132 spots: 11341357\n",
      "2022-03-08 01:33:58,957 - blocks: [[1, 1417669], [1417670, 2835338], [2835339, 4253007], [4253008, 5670676], [5670677, 7088345], [7088346, 8506014], [8506015, 9923683], [9923684, 11341357]]\n",
      "2022-03-08 01:33:58,957 - CMD: fastq-dump -N 1 -X 1417669 -O ./pfd_4r3ydpaq/0 --gzip --split-files SRR13349132\n",
      "2022-03-08 01:33:58,974 - CMD: fastq-dump -N 1417670 -X 2835338 -O ./pfd_4r3ydpaq/1 --gzip --split-files SRR13349132\n",
      "2022-03-08 01:33:58,976 - CMD: fastq-dump -N 2835339 -X 4253007 -O ./pfd_4r3ydpaq/2 --gzip --split-files SRR13349132\n",
      "2022-03-08 01:33:58,977 - CMD: fastq-dump -N 4253008 -X 5670676 -O ./pfd_4r3ydpaq/3 --gzip --split-files SRR13349132\n",
      "2022-03-08 01:33:59,050 - CMD: fastq-dump -N 5670677 -X 7088345 -O ./pfd_4r3ydpaq/4 --gzip --split-files SRR13349132\n",
      "2022-03-08 01:33:59,052 - CMD: fastq-dump -N 7088346 -X 8506014 -O ./pfd_4r3ydpaq/5 --gzip --split-files SRR13349132\n",
      "2022-03-08 01:33:59,068 - CMD: fastq-dump -N 8506015 -X 9923683 -O ./pfd_4r3ydpaq/6 --gzip --split-files SRR13349132\n",
      "2022-03-08 01:33:59,077 - CMD: fastq-dump -N 9923684 -X 11341357 -O ./pfd_4r3ydpaq/7 --gzip --split-files SRR13349132\n",
      "Read 1417669 spots for SRR13349132\n",
      "Written 1417669 spots for SRR13349132\n",
      "Read 1417669 spots for SRR13349132\n",
      "Written 1417669 spots for SRR13349132\n",
      "Read 1417669 spots for SRR13349132\n",
      "Written 1417669 spots for SRR13349132\n",
      "Read 1417669 spots for SRR13349132\n",
      "Written 1417669 spots for SRR13349132\n",
      "Read 1417669 spots for SRR13349132\n",
      "Written 1417669 spots for SRR13349132\n",
      "Read 1417669 spots for SRR13349132\n",
      "Written 1417669 spots for SRR13349132\n",
      "Read 1417674 spots for SRR13349132\n",
      "Written 1417674 spots for SRR13349132\n",
      "Read 1417669 spots for SRR13349132\n",
      "Written 1417669 spots for SRR13349132\n",
      "2022-03-08 01:36:09,537 - SRR ids: ['SRR13349133']\n",
      "2022-03-08 01:36:09,537 - extra args: ['--gzip', '--split-files']\n",
      "2022-03-08 01:36:09,537 - tempdir: ./pfd_1rixt0la\n",
      "2022-03-08 01:36:09,537 - CMD: sra-stat --meta --quick SRR13349133\n",
      "2022-03-08 01:36:10,086 - SRR13349133 spots: 11603881\n",
      "2022-03-08 01:36:10,086 - blocks: [[1, 1450485], [1450486, 2900970], [2900971, 4351455], [4351456, 5801940], [5801941, 7252425], [7252426, 8702910], [8702911, 10153395], [10153396, 11603881]]\n",
      "2022-03-08 01:36:10,086 - CMD: fastq-dump -N 1 -X 1450485 -O ./pfd_1rixt0la/0 --gzip --split-files SRR13349133\n",
      "2022-03-08 01:36:10,088 - CMD: fastq-dump -N 1450486 -X 2900970 -O ./pfd_1rixt0la/1 --gzip --split-files SRR13349133\n",
      "2022-03-08 01:36:10,089 - CMD: fastq-dump -N 2900971 -X 4351455 -O ./pfd_1rixt0la/2 --gzip --split-files SRR13349133\n",
      "2022-03-08 01:36:10,091 - CMD: fastq-dump -N 4351456 -X 5801940 -O ./pfd_1rixt0la/3 --gzip --split-files SRR13349133\n",
      "2022-03-08 01:36:10,093 - CMD: fastq-dump -N 5801941 -X 7252425 -O ./pfd_1rixt0la/4 --gzip --split-files SRR13349133\n",
      "2022-03-08 01:36:10,113 - CMD: fastq-dump -N 7252426 -X 8702910 -O ./pfd_1rixt0la/5 --gzip --split-files SRR13349133\n",
      "2022-03-08 01:36:10,121 - CMD: fastq-dump -N 8702911 -X 10153395 -O ./pfd_1rixt0la/6 --gzip --split-files SRR13349133\n",
      "2022-03-08 01:36:10,131 - CMD: fastq-dump -N 10153396 -X 11603881 -O ./pfd_1rixt0la/7 --gzip --split-files SRR13349133\n",
      "Read 1450485 spots for SRR13349133\n",
      "Written 1450485 spots for SRR13349133\n",
      "Read 1450485 spots for SRR13349133\n",
      "Written 1450485 spots for SRR13349133\n",
      "Read 1450485 spots for SRR13349133\n",
      "Written 1450485 spots for SRR13349133\n",
      "Read 1450485 spots for SRR13349133\n",
      "Written 1450485 spots for SRR13349133\n",
      "Read 1450485 spots for SRR13349133\n",
      "Written 1450485 spots for SRR13349133\n",
      "Read 1450485 spots for SRR13349133\n",
      "Written 1450485 spots for SRR13349133\n",
      "Read 1450486 spots for SRR13349133\n",
      "Written 1450486 spots for SRR13349133\n",
      "Read 1450485 spots for SRR13349133\n",
      "Written 1450485 spots for SRR13349133\n"
     ]
    }
   ],
   "source": [
    "!cat data/raw_fastq/list_of_accesionIDS.txt |xargs -I {} parallel-fastq-dump -O data/raw_fastq/ --tmpdir . --threads 8 --gzip --split-files --sra-id {}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "tags": []
   },
   "source": [
    "### STEP 4: Copy reference transcriptome files that will be used by Salmon using E-Direct\n",
    "\n",
    "Salmon is a tool that aligns RNA-Seq reads to a set of transcripts rather than the entire genome.\n",
    "\n",
    "So we will need a transcriptome reference.\n",
    "\n",
    "To get one, we can search through the NCBI assembly database, find an assembly, and download transcriptome reference files from that assembly using FTP links.\n",
    "\n",
    "For instance, we will use the <a href='https://www.ncbi.nlm.nih.gov/assembly/GCF_001632805.1'>ASM163280v1</a> refseq assembly, found by searching through the NCBI assembly database. The FTP links can be accessed through the website in various ways, one way is to click the 'FTP directory for RefSeq assembly' link, found under 'Access the data', section.\n",
    "\n",
    "Alternatively, this process can be done through the NCBI command line set of tools called 'Entrez Direct' (EDirect).\n",
    "\n",
    "The bottom lines of code show how FTP links for specific files in a given assembly can be performed using EDirect."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 147,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n",
      "                                 Dload  Upload   Total   Spent    Left  Speed\n",
      "100 1436k  100 1436k    0     0  6544k      0 --:--:-- --:--:-- --:--:-- 6561k\n",
      "  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n",
      "                                 Dload  Upload   Total   Spent    Left  Speed\n",
      "100  329k  100  329k    0     0  2926k      0 --:--:-- --:--:-- --:--:-- 2918k\n",
      "  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n",
      "                                 Dload  Upload   Total   Spent    Left  Speed\n",
      "100  222k  100  222k    0     0  2068k      0 --:--:-- --:--:-- --:--:-- 2080k\n"
     ]
    }
   ],
   "source": [
    "#parse for the ftp link and download the genome reference fasta file\n",
    "\n",
    "!esearch -db assembly -query GCF_001632805.1 | efetch -format docsum \\\n",
    "| xtract -pattern DocumentSummary -element FtpPath_RefSeq \\\n",
    "| awk -F\"/\" '{print \"curl -o data/reference/\"$NF\"_genomic.fna.gz \" $0\"/\"$NF\"_genomic.fna.gz\"}' \\\n",
    "| bash\n",
    "\n",
    "#parse for the ftp link and download the gtf reference fasta file\n",
    "\n",
    "!esearch -db assembly -query GCF_001632805.1 | efetch -format docsum \\\n",
    "| xtract -pattern DocumentSummary -element FtpPath_RefSeq \\\n",
    "| awk -F\"/\" '{print \"curl -o data/reference/\"$NF\"_genomic.gff.gz \" $0\"/\"$NF\"_genomic.gff.gz\"}' \\\n",
    "| bash\n",
    "\n",
    "# parse for the ftp link and download the feature-table reference file \n",
    "# (for later use for merging readcounts with gene names in R code).\n",
    "\n",
    "!esearch -db assembly -query GCF_001632805.1 | efetch -format docsum \\\n",
    "| xtract -pattern DocumentSummary -element FtpPath_RefSeq \\\n",
    "| awk -F\"/\" '{print \"curl -o data/reference/\"$NF\"_feature_table.txt.gz \" $0\"/\"$NF\"_feature_table.txt.gz\"}' \\\n",
    "| bash\n",
    "\n",
    "\n",
    "#unzip the compresseed fasta files\n",
    "\n",
    "!gzip -d data/reference/*.gz\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Next we can use gffread to create a transcriptome reference file using the gtf and genome files we downloaded."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 148,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "FASTA index file data/reference/GCF_001632805.1_ASM163280v1_genomic.fna.fai created.\n"
     ]
    }
   ],
   "source": [
    "!gffread -w data/reference/GCF_001632805.1_transcriptome_reference.fa -g data/reference/GCF_001632805.1_ASM163280v1_genomic.fna data/reference/GCF_001632805.1_ASM163280v1_genomic.gff"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We can also include the genome at the end of the reference file, for the purpose of creating a 'decoy-aware' transcriptome file, more information for which can be found in the Salmon documentation.\n",
    "\n",
    "Accordingly, also for purpose of decoy-aware indexing, we will also create a 'decoy file', which salmon needs to point to the genome sequence in our transcriptome reference file."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 133,
   "metadata": {},
   "outputs": [],
   "source": [
    "!cat data/reference/GCF_001632805.1_transcriptome_reference.fa <(echo) data/reference/GCF_001632805.1_ASM163280v1_genomic.fna > data/reference/GCF_001632805.1_transcriptome_reference_w_decoy.fa\n",
    "!echo \"NZ_CP007220.1\" > data/reference/decoys.txt\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### STEP 5: Copy data file for Trimmomatic\n",
    "\n",
    "One of trimmomatics functions is to trim sequence machine specific adapter sequences. These are usually within the trimmomatic installation directory in a folder called adapters.\n",
    "\n",
    "Directories of packages within conda installations can be confusing, so in the case of using conda with trimmomatic, it may be easier to simply download or create a file with the relevant adapter sequencecs and store it in an easy to find directory."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n",
      "                                 Dload  Upload   Total   Spent    Left  Speed\n",
      "100    95  100    95    0     0   5027      0 --:--:-- --:--:-- --:--:--  5277\n",
      ">PrefixPE/1\n",
      "TACACTCTTTCCCTACACGACGCTCTTCCGATCT\n",
      ">PrefixPE/2\n",
      "GTGACTGGAGTTCAGACGTGTGCTCTTCCGATCT\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!curl https://storage.googleapis.com/me-inbre-rnaseq-pipelinev2/config/TruSeq3-PE.fa --output TruSeq3-PE.fa\n",
    "!head TruSeq3-PE.fa "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### STEP 6: Run Trimmomatic\n",
    "Trimmomatic will trim off any adapter sequences or low quality sequence it detects in the FASTQ files.\n",
    "\n",
    "For simplicity sake, it is possible to use our original list of run IDs to run trimmomatic on all of our fastq files in one line of code."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TrimmomaticPE: Started with arguments:\n",
      " -threads 2 data/raw_fastq/SRR13349122_1.fastq.gz data/raw_fastq/SRR13349122_2.fastq.gz data/trimmed/SRR13349122_1_trimmed.fastq.gz data/trimmed/SRR13349122_2_trimmed.fastq.gz data/trimmed/SRR13349122_1_trimmed_unpaired.fastq.gz data/trimmed/SRR13349122_2_trimmed_unpaired.fastq.gz ILLUMINACLIP:TruSeq3-PE.fa:2:30:10:2:keepBothReads LEADING:3 TRAILING:3 MINLEN:36\n",
      "Using PrefixPair: 'TACACTCTTTCCCTACACGACGCTCTTCCGATCT' and 'GTGACTGGAGTTCAGACGTGTGCTCTTCCGATCT'\n",
      "ILLUMINACLIP: Using 1 prefix pairs, 0 forward/reverse sequences, 0 forward only sequences, 0 reverse only sequences\n",
      "Quality encoding detected as phred33\n",
      "Input Read Pairs: 10827590 Both Surviving: 10810267 (99.84%) Forward Only Surviving: 17297 (0.16%) Reverse Only Surviving: 0 (0.00%) Dropped: 26 (0.00%)\n",
      "TrimmomaticPE: Completed successfully\n",
      "TrimmomaticPE: Started with arguments:\n",
      " -threads 2 data/raw_fastq/SRR13349123_1.fastq.gz data/raw_fastq/SRR13349123_2.fastq.gz data/trimmed/SRR13349123_1_trimmed.fastq.gz data/trimmed/SRR13349123_2_trimmed.fastq.gz data/trimmed/SRR13349123_1_trimmed_unpaired.fastq.gz data/trimmed/SRR13349123_2_trimmed_unpaired.fastq.gz ILLUMINACLIP:TruSeq3-PE.fa:2:30:10:2:keepBothReads LEADING:3 TRAILING:3 MINLEN:36\n",
      "Using PrefixPair: 'TACACTCTTTCCCTACACGACGCTCTTCCGATCT' and 'GTGACTGGAGTTCAGACGTGTGCTCTTCCGATCT'\n",
      "ILLUMINACLIP: Using 1 prefix pairs, 0 forward/reverse sequences, 0 forward only sequences, 0 reverse only sequences\n",
      "Quality encoding detected as phred33\n",
      "Input Read Pairs: 11165256 Both Surviving: 11096258 (99.38%) Forward Only Surviving: 68966 (0.62%) Reverse Only Surviving: 0 (0.00%) Dropped: 32 (0.00%)\n",
      "TrimmomaticPE: Completed successfully\n",
      "TrimmomaticPE: Started with arguments:\n",
      " -threads 2 data/raw_fastq/SRR13349124_1.fastq.gz data/raw_fastq/SRR13349124_2.fastq.gz data/trimmed/SRR13349124_1_trimmed.fastq.gz data/trimmed/SRR13349124_2_trimmed.fastq.gz data/trimmed/SRR13349124_1_trimmed_unpaired.fastq.gz data/trimmed/SRR13349124_2_trimmed_unpaired.fastq.gz ILLUMINACLIP:TruSeq3-PE.fa:2:30:10:2:keepBothReads LEADING:3 TRAILING:3 MINLEN:36\n",
      "Using PrefixPair: 'TACACTCTTTCCCTACACGACGCTCTTCCGATCT' and 'GTGACTGGAGTTCAGACGTGTGCTCTTCCGATCT'\n",
      "ILLUMINACLIP: Using 1 prefix pairs, 0 forward/reverse sequences, 0 forward only sequences, 0 reverse only sequences\n",
      "Quality encoding detected as phred33\n",
      "Input Read Pairs: 10727273 Both Surviving: 10710165 (99.84%) Forward Only Surviving: 17076 (0.16%) Reverse Only Surviving: 0 (0.00%) Dropped: 32 (0.00%)\n",
      "TrimmomaticPE: Completed successfully\n",
      "TrimmomaticPE: Started with arguments:\n",
      " -threads 2 data/raw_fastq/SRR13349125_1.fastq.gz data/raw_fastq/SRR13349125_2.fastq.gz data/trimmed/SRR13349125_1_trimmed.fastq.gz data/trimmed/SRR13349125_2_trimmed.fastq.gz data/trimmed/SRR13349125_1_trimmed_unpaired.fastq.gz data/trimmed/SRR13349125_2_trimmed_unpaired.fastq.gz ILLUMINACLIP:TruSeq3-PE.fa:2:30:10:2:keepBothReads LEADING:3 TRAILING:3 MINLEN:36\n",
      "Using PrefixPair: 'TACACTCTTTCCCTACACGACGCTCTTCCGATCT' and 'GTGACTGGAGTTCAGACGTGTGCTCTTCCGATCT'\n",
      "ILLUMINACLIP: Using 1 prefix pairs, 0 forward/reverse sequences, 0 forward only sequences, 0 reverse only sequences\n",
      "Quality encoding detected as phred33\n",
      "Input Read Pairs: 10992686 Both Surviving: 10924466 (99.38%) Forward Only Surviving: 68182 (0.62%) Reverse Only Surviving: 0 (0.00%) Dropped: 38 (0.00%)\n",
      "TrimmomaticPE: Completed successfully\n",
      "TrimmomaticPE: Started with arguments:\n",
      " -threads 2 data/raw_fastq/SRR13349126_1.fastq.gz data/raw_fastq/SRR13349126_2.fastq.gz data/trimmed/SRR13349126_1_trimmed.fastq.gz data/trimmed/SRR13349126_2_trimmed.fastq.gz data/trimmed/SRR13349126_1_trimmed_unpaired.fastq.gz data/trimmed/SRR13349126_2_trimmed_unpaired.fastq.gz ILLUMINACLIP:TruSeq3-PE.fa:2:30:10:2:keepBothReads LEADING:3 TRAILING:3 MINLEN:36\n",
      "Using PrefixPair: 'TACACTCTTTCCCTACACGACGCTCTTCCGATCT' and 'GTGACTGGAGTTCAGACGTGTGCTCTTCCGATCT'\n",
      "ILLUMINACLIP: Using 1 prefix pairs, 0 forward/reverse sequences, 0 forward only sequences, 0 reverse only sequences\n",
      "Quality encoding detected as phred33\n",
      "Input Read Pairs: 12267497 Both Surviving: 12248005 (99.84%) Forward Only Surviving: 19411 (0.16%) Reverse Only Surviving: 0 (0.00%) Dropped: 81 (0.00%)\n",
      "TrimmomaticPE: Completed successfully\n",
      "TrimmomaticPE: Started with arguments:\n",
      " -threads 2 data/raw_fastq/SRR13349127_1.fastq.gz data/raw_fastq/SRR13349127_2.fastq.gz data/trimmed/SRR13349127_1_trimmed.fastq.gz data/trimmed/SRR13349127_2_trimmed.fastq.gz data/trimmed/SRR13349127_1_trimmed_unpaired.fastq.gz data/trimmed/SRR13349127_2_trimmed_unpaired.fastq.gz ILLUMINACLIP:TruSeq3-PE.fa:2:30:10:2:keepBothReads LEADING:3 TRAILING:3 MINLEN:36\n",
      "Using PrefixPair: 'TACACTCTTTCCCTACACGACGCTCTTCCGATCT' and 'GTGACTGGAGTTCAGACGTGTGCTCTTCCGATCT'\n",
      "ILLUMINACLIP: Using 1 prefix pairs, 0 forward/reverse sequences, 0 forward only sequences, 0 reverse only sequences\n",
      "Quality encoding detected as phred33\n",
      "Input Read Pairs: 12563032 Both Surviving: 12485561 (99.38%) Forward Only Surviving: 77386 (0.62%) Reverse Only Surviving: 0 (0.00%) Dropped: 85 (0.00%)\n",
      "TrimmomaticPE: Completed successfully\n",
      "TrimmomaticPE: Started with arguments:\n",
      " -threads 2 data/raw_fastq/SRR13349128_1.fastq.gz data/raw_fastq/SRR13349128_2.fastq.gz data/trimmed/SRR13349128_1_trimmed.fastq.gz data/trimmed/SRR13349128_2_trimmed.fastq.gz data/trimmed/SRR13349128_1_trimmed_unpaired.fastq.gz data/trimmed/SRR13349128_2_trimmed_unpaired.fastq.gz ILLUMINACLIP:TruSeq3-PE.fa:2:30:10:2:keepBothReads LEADING:3 TRAILING:3 MINLEN:36\n",
      "Using PrefixPair: 'TACACTCTTTCCCTACACGACGCTCTTCCGATCT' and 'GTGACTGGAGTTCAGACGTGTGCTCTTCCGATCT'\n",
      "ILLUMINACLIP: Using 1 prefix pairs, 0 forward/reverse sequences, 0 forward only sequences, 0 reverse only sequences\n",
      "Quality encoding detected as phred33\n",
      "Input Read Pairs: 12652387 Both Surviving: 12631972 (99.84%) Forward Only Surviving: 20327 (0.16%) Reverse Only Surviving: 0 (0.00%) Dropped: 88 (0.00%)\n",
      "TrimmomaticPE: Completed successfully\n",
      "TrimmomaticPE: Started with arguments:\n",
      " -threads 2 data/raw_fastq/SRR13349129_1.fastq.gz data/raw_fastq/SRR13349129_2.fastq.gz data/trimmed/SRR13349129_1_trimmed.fastq.gz data/trimmed/SRR13349129_2_trimmed.fastq.gz data/trimmed/SRR13349129_1_trimmed_unpaired.fastq.gz data/trimmed/SRR13349129_2_trimmed_unpaired.fastq.gz ILLUMINACLIP:TruSeq3-PE.fa:2:30:10:2:keepBothReads LEADING:3 TRAILING:3 MINLEN:36\n",
      "Using PrefixPair: 'TACACTCTTTCCCTACACGACGCTCTTCCGATCT' and 'GTGACTGGAGTTCAGACGTGTGCTCTTCCGATCT'\n",
      "ILLUMINACLIP: Using 1 prefix pairs, 0 forward/reverse sequences, 0 forward only sequences, 0 reverse only sequences\n",
      "Quality encoding detected as phred33\n",
      "Input Read Pairs: 12961793 Both Surviving: 12881935 (99.38%) Forward Only Surviving: 79783 (0.62%) Reverse Only Surviving: 0 (0.00%) Dropped: 75 (0.00%)\n",
      "TrimmomaticPE: Completed successfully\n",
      "TrimmomaticPE: Started with arguments:\n",
      " -threads 2 data/raw_fastq/SRR13349130_1.fastq.gz data/raw_fastq/SRR13349130_2.fastq.gz data/trimmed/SRR13349130_1_trimmed.fastq.gz data/trimmed/SRR13349130_2_trimmed.fastq.gz data/trimmed/SRR13349130_1_trimmed_unpaired.fastq.gz data/trimmed/SRR13349130_2_trimmed_unpaired.fastq.gz ILLUMINACLIP:TruSeq3-PE.fa:2:30:10:2:keepBothReads LEADING:3 TRAILING:3 MINLEN:36\n",
      "Using PrefixPair: 'TACACTCTTTCCCTACACGACGCTCTTCCGATCT' and 'GTGACTGGAGTTCAGACGTGTGCTCTTCCGATCT'\n",
      "ILLUMINACLIP: Using 1 prefix pairs, 0 forward/reverse sequences, 0 forward only sequences, 0 reverse only sequences\n",
      "Quality encoding detected as phred33\n",
      "Input Read Pairs: 10083015 Both Surviving: 10067073 (99.84%) Forward Only Surviving: 15920 (0.16%) Reverse Only Surviving: 0 (0.00%) Dropped: 22 (0.00%)\n",
      "TrimmomaticPE: Completed successfully\n",
      "TrimmomaticPE: Started with arguments:\n",
      " -threads 2 data/raw_fastq/SRR13349131_1.fastq.gz data/raw_fastq/SRR13349131_2.fastq.gz data/trimmed/SRR13349131_1_trimmed.fastq.gz data/trimmed/SRR13349131_2_trimmed.fastq.gz data/trimmed/SRR13349131_1_trimmed_unpaired.fastq.gz data/trimmed/SRR13349131_2_trimmed_unpaired.fastq.gz ILLUMINACLIP:TruSeq3-PE.fa:2:30:10:2:keepBothReads LEADING:3 TRAILING:3 MINLEN:36\n",
      "Using PrefixPair: 'TACACTCTTTCCCTACACGACGCTCTTCCGATCT' and 'GTGACTGGAGTTCAGACGTGTGCTCTTCCGATCT'\n",
      "ILLUMINACLIP: Using 1 prefix pairs, 0 forward/reverse sequences, 0 forward only sequences, 0 reverse only sequences\n",
      "Quality encoding detected as phred33\n",
      "Input Read Pairs: 10491160 Both Surviving: 10427189 (99.39%) Forward Only Surviving: 63929 (0.61%) Reverse Only Surviving: 0 (0.00%) Dropped: 42 (0.00%)\n",
      "TrimmomaticPE: Completed successfully\n",
      "TrimmomaticPE: Started with arguments:\n",
      " -threads 2 data/raw_fastq/SRR13349132_1.fastq.gz data/raw_fastq/SRR13349132_2.fastq.gz data/trimmed/SRR13349132_1_trimmed.fastq.gz data/trimmed/SRR13349132_2_trimmed.fastq.gz data/trimmed/SRR13349132_1_trimmed_unpaired.fastq.gz data/trimmed/SRR13349132_2_trimmed_unpaired.fastq.gz ILLUMINACLIP:TruSeq3-PE.fa:2:30:10:2:keepBothReads LEADING:3 TRAILING:3 MINLEN:36\n",
      "Using PrefixPair: 'TACACTCTTTCCCTACACGACGCTCTTCCGATCT' and 'GTGACTGGAGTTCAGACGTGTGCTCTTCCGATCT'\n",
      "ILLUMINACLIP: Using 1 prefix pairs, 0 forward/reverse sequences, 0 forward only sequences, 0 reverse only sequences\n",
      "Quality encoding detected as phred33\n",
      "Input Read Pairs: 11341357 Both Surviving: 11323093 (99.84%) Forward Only Surviving: 18224 (0.16%) Reverse Only Surviving: 0 (0.00%) Dropped: 40 (0.00%)\n",
      "TrimmomaticPE: Completed successfully\n",
      "TrimmomaticPE: Started with arguments:\n",
      " -threads 2 data/raw_fastq/SRR13349133_1.fastq.gz data/raw_fastq/SRR13349133_2.fastq.gz data/trimmed/SRR13349133_1_trimmed.fastq.gz data/trimmed/SRR13349133_2_trimmed.fastq.gz data/trimmed/SRR13349133_1_trimmed_unpaired.fastq.gz data/trimmed/SRR13349133_2_trimmed_unpaired.fastq.gz ILLUMINACLIP:TruSeq3-PE.fa:2:30:10:2:keepBothReads LEADING:3 TRAILING:3 MINLEN:36\n",
      "Using PrefixPair: 'TACACTCTTTCCCTACACGACGCTCTTCCGATCT' and 'GTGACTGGAGTTCAGACGTGTGCTCTTCCGATCT'\n",
      "ILLUMINACLIP: Using 1 prefix pairs, 0 forward/reverse sequences, 0 forward only sequences, 0 reverse only sequences\n",
      "Quality encoding detected as phred33\n",
      "Input Read Pairs: 11603881 Both Surviving: 11531393 (99.38%) Forward Only Surviving: 72435 (0.62%) Reverse Only Surviving: 0 (0.00%) Dropped: 53 (0.00%)\n",
      "TrimmomaticPE: Completed successfully\n"
     ]
    }
   ],
   "source": [
    "!cat data/raw_fastq/list_of_accesionIDS.txt | xargs -I {} trimmomatic PE -threads 2 'data/raw_fastq/{}_1.fastq.gz' 'data/raw_fastq/{}_2.fastq.gz' 'data/trimmed/{}_1_trimmed.fastq.gz' 'data/trimmed/{}_2_trimmed.fastq.gz' 'data/trimmed/{}_1_trimmed_unpaired.fastq.gz'  'data/trimmed/{}_2_trimmed_unpaired.fastq.gz' ILLUMINACLIP:TruSeq3-PE.fa:2:30:10:2:keepBothReads LEADING:3 TRAILING:3 MINLEN:36"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### STEP 7: Run FastQC\n",
    "FastQC is an invaluable tool that allows you to evaluate whether there are problems with a set of reads. For example, it will provide a report of whether there is any bias in the sequence composition of the reads.\n",
    "\n",
    "If you notice the results of the trimming, you may have noted the sequences in the reverse reads were few, and largely unpaired. This may be an artifact from how the original sequencing process. This is okay, we can proceed from here simply using the forward reads."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Started analysis of SRR13349122_1_trimmed.fastq.gz\n",
      "Approx 5% complete for SRR13349122_1_trimmed.fastq.gz\n",
      "Approx 10% complete for SRR13349122_1_trimmed.fastq.gz\n",
      "Approx 15% complete for SRR13349122_1_trimmed.fastq.gz\n",
      "Approx 20% complete for SRR13349122_1_trimmed.fastq.gz\n",
      "Approx 25% complete for SRR13349122_1_trimmed.fastq.gz\n",
      "Approx 30% complete for SRR13349122_1_trimmed.fastq.gz\n",
      "Approx 35% complete for SRR13349122_1_trimmed.fastq.gz\n",
      "Approx 40% complete for SRR13349122_1_trimmed.fastq.gz\n",
      "Approx 45% complete for SRR13349122_1_trimmed.fastq.gz\n",
      "Approx 50% complete for SRR13349122_1_trimmed.fastq.gz\n",
      "Approx 55% complete for SRR13349122_1_trimmed.fastq.gz\n",
      "Approx 60% complete for SRR13349122_1_trimmed.fastq.gz\n",
      "Approx 65% complete for SRR13349122_1_trimmed.fastq.gz\n",
      "Approx 70% complete for SRR13349122_1_trimmed.fastq.gz\n",
      "Approx 75% complete for SRR13349122_1_trimmed.fastq.gz\n",
      "Approx 80% complete for SRR13349122_1_trimmed.fastq.gz\n",
      "Approx 85% complete for SRR13349122_1_trimmed.fastq.gz\n",
      "Approx 90% complete for SRR13349122_1_trimmed.fastq.gz\n",
      "Approx 95% complete for SRR13349122_1_trimmed.fastq.gz\n",
      "Analysis complete for SRR13349122_1_trimmed.fastq.gz\n",
      "Started analysis of SRR13349123_1_trimmed.fastq.gz\n",
      "Approx 5% complete for SRR13349123_1_trimmed.fastq.gz\n",
      "Approx 10% complete for SRR13349123_1_trimmed.fastq.gz\n",
      "Approx 15% complete for SRR13349123_1_trimmed.fastq.gz\n",
      "Approx 20% complete for SRR13349123_1_trimmed.fastq.gz\n",
      "Approx 25% complete for SRR13349123_1_trimmed.fastq.gz\n",
      "Approx 30% complete for SRR13349123_1_trimmed.fastq.gz\n",
      "Approx 35% complete for SRR13349123_1_trimmed.fastq.gz\n",
      "Approx 40% complete for SRR13349123_1_trimmed.fastq.gz\n",
      "Approx 45% complete for SRR13349123_1_trimmed.fastq.gz\n",
      "Approx 50% complete for SRR13349123_1_trimmed.fastq.gz\n",
      "Approx 55% complete for SRR13349123_1_trimmed.fastq.gz\n",
      "Approx 60% complete for SRR13349123_1_trimmed.fastq.gz\n",
      "Approx 65% complete for SRR13349123_1_trimmed.fastq.gz\n",
      "Approx 70% complete for SRR13349123_1_trimmed.fastq.gz\n",
      "Approx 75% complete for SRR13349123_1_trimmed.fastq.gz\n",
      "Approx 80% complete for SRR13349123_1_trimmed.fastq.gz\n",
      "Approx 85% complete for SRR13349123_1_trimmed.fastq.gz\n",
      "Approx 90% complete for SRR13349123_1_trimmed.fastq.gz\n",
      "Approx 95% complete for SRR13349123_1_trimmed.fastq.gz\n",
      "Analysis complete for SRR13349123_1_trimmed.fastq.gz\n",
      "Started analysis of SRR13349124_1_trimmed.fastq.gz\n",
      "Approx 5% complete for SRR13349124_1_trimmed.fastq.gz\n",
      "Approx 10% complete for SRR13349124_1_trimmed.fastq.gz\n",
      "Approx 15% complete for SRR13349124_1_trimmed.fastq.gz\n",
      "Approx 20% complete for SRR13349124_1_trimmed.fastq.gz\n",
      "Approx 25% complete for SRR13349124_1_trimmed.fastq.gz\n",
      "Approx 30% complete for SRR13349124_1_trimmed.fastq.gz\n",
      "Approx 35% complete for SRR13349124_1_trimmed.fastq.gz\n",
      "Approx 40% complete for SRR13349124_1_trimmed.fastq.gz\n",
      "Approx 45% complete for SRR13349124_1_trimmed.fastq.gz\n",
      "Approx 50% complete for SRR13349124_1_trimmed.fastq.gz\n",
      "Approx 55% complete for SRR13349124_1_trimmed.fastq.gz\n",
      "Approx 60% complete for SRR13349124_1_trimmed.fastq.gz\n",
      "Approx 65% complete for SRR13349124_1_trimmed.fastq.gz\n",
      "Approx 70% complete for SRR13349124_1_trimmed.fastq.gz\n",
      "Approx 75% complete for SRR13349124_1_trimmed.fastq.gz\n",
      "Approx 80% complete for SRR13349124_1_trimmed.fastq.gz\n",
      "Approx 85% complete for SRR13349124_1_trimmed.fastq.gz\n",
      "Approx 90% complete for SRR13349124_1_trimmed.fastq.gz\n",
      "Approx 95% complete for SRR13349124_1_trimmed.fastq.gz\n",
      "Analysis complete for SRR13349124_1_trimmed.fastq.gz\n",
      "Started analysis of SRR13349125_1_trimmed.fastq.gz\n",
      "Approx 5% complete for SRR13349125_1_trimmed.fastq.gz\n",
      "Approx 10% complete for SRR13349125_1_trimmed.fastq.gz\n",
      "Approx 15% complete for SRR13349125_1_trimmed.fastq.gz\n",
      "Approx 20% complete for SRR13349125_1_trimmed.fastq.gz\n",
      "Approx 25% complete for SRR13349125_1_trimmed.fastq.gz\n",
      "Approx 30% complete for SRR13349125_1_trimmed.fastq.gz\n",
      "Approx 35% complete for SRR13349125_1_trimmed.fastq.gz\n",
      "Approx 40% complete for SRR13349125_1_trimmed.fastq.gz\n",
      "Approx 45% complete for SRR13349125_1_trimmed.fastq.gz\n",
      "Approx 50% complete for SRR13349125_1_trimmed.fastq.gz\n",
      "Approx 55% complete for SRR13349125_1_trimmed.fastq.gz\n",
      "Approx 60% complete for SRR13349125_1_trimmed.fastq.gz\n",
      "Approx 65% complete for SRR13349125_1_trimmed.fastq.gz\n",
      "Approx 70% complete for SRR13349125_1_trimmed.fastq.gz\n",
      "Approx 75% complete for SRR13349125_1_trimmed.fastq.gz\n",
      "Approx 80% complete for SRR13349125_1_trimmed.fastq.gz\n",
      "Approx 85% complete for SRR13349125_1_trimmed.fastq.gz\n",
      "Approx 90% complete for SRR13349125_1_trimmed.fastq.gz\n",
      "Approx 95% complete for SRR13349125_1_trimmed.fastq.gz\n",
      "Analysis complete for SRR13349125_1_trimmed.fastq.gz\n",
      "Started analysis of SRR13349126_1_trimmed.fastq.gz\n",
      "Approx 5% complete for SRR13349126_1_trimmed.fastq.gz\n",
      "Approx 10% complete for SRR13349126_1_trimmed.fastq.gz\n",
      "Approx 15% complete for SRR13349126_1_trimmed.fastq.gz\n",
      "Approx 20% complete for SRR13349126_1_trimmed.fastq.gz\n",
      "Approx 25% complete for SRR13349126_1_trimmed.fastq.gz\n",
      "Approx 30% complete for SRR13349126_1_trimmed.fastq.gz\n",
      "Approx 35% complete for SRR13349126_1_trimmed.fastq.gz\n",
      "Approx 40% complete for SRR13349126_1_trimmed.fastq.gz\n",
      "Approx 45% complete for SRR13349126_1_trimmed.fastq.gz\n",
      "Approx 50% complete for SRR13349126_1_trimmed.fastq.gz\n",
      "Approx 55% complete for SRR13349126_1_trimmed.fastq.gz\n",
      "Approx 60% complete for SRR13349126_1_trimmed.fastq.gz\n",
      "Approx 65% complete for SRR13349126_1_trimmed.fastq.gz\n",
      "Approx 70% complete for SRR13349126_1_trimmed.fastq.gz\n",
      "Approx 75% complete for SRR13349126_1_trimmed.fastq.gz\n",
      "Approx 80% complete for SRR13349126_1_trimmed.fastq.gz\n",
      "Approx 85% complete for SRR13349126_1_trimmed.fastq.gz\n",
      "Approx 90% complete for SRR13349126_1_trimmed.fastq.gz\n",
      "Approx 95% complete for SRR13349126_1_trimmed.fastq.gz\n",
      "Approx 100% complete for SRR13349126_1_trimmed.fastq.gz\n",
      "Analysis complete for SRR13349126_1_trimmed.fastq.gz\n",
      "Started analysis of SRR13349127_1_trimmed.fastq.gz\n",
      "Approx 5% complete for SRR13349127_1_trimmed.fastq.gz\n",
      "Approx 10% complete for SRR13349127_1_trimmed.fastq.gz\n",
      "Approx 15% complete for SRR13349127_1_trimmed.fastq.gz\n",
      "Approx 20% complete for SRR13349127_1_trimmed.fastq.gz\n",
      "Approx 25% complete for SRR13349127_1_trimmed.fastq.gz\n",
      "Approx 30% complete for SRR13349127_1_trimmed.fastq.gz\n",
      "Approx 35% complete for SRR13349127_1_trimmed.fastq.gz\n",
      "Approx 40% complete for SRR13349127_1_trimmed.fastq.gz\n",
      "Approx 45% complete for SRR13349127_1_trimmed.fastq.gz\n",
      "Approx 50% complete for SRR13349127_1_trimmed.fastq.gz\n",
      "Approx 55% complete for SRR13349127_1_trimmed.fastq.gz\n",
      "Approx 60% complete for SRR13349127_1_trimmed.fastq.gz\n",
      "Approx 65% complete for SRR13349127_1_trimmed.fastq.gz\n",
      "Approx 70% complete for SRR13349127_1_trimmed.fastq.gz\n",
      "Approx 75% complete for SRR13349127_1_trimmed.fastq.gz\n",
      "Approx 80% complete for SRR13349127_1_trimmed.fastq.gz\n",
      "Approx 85% complete for SRR13349127_1_trimmed.fastq.gz\n",
      "Approx 90% complete for SRR13349127_1_trimmed.fastq.gz\n",
      "Approx 95% complete for SRR13349127_1_trimmed.fastq.gz\n",
      "Analysis complete for SRR13349127_1_trimmed.fastq.gz\n",
      "Started analysis of SRR13349128_1_trimmed.fastq.gz\n",
      "Approx 5% complete for SRR13349128_1_trimmed.fastq.gz\n",
      "Approx 10% complete for SRR13349128_1_trimmed.fastq.gz\n",
      "Approx 15% complete for SRR13349128_1_trimmed.fastq.gz\n",
      "Approx 20% complete for SRR13349128_1_trimmed.fastq.gz\n",
      "Approx 25% complete for SRR13349128_1_trimmed.fastq.gz\n",
      "Approx 30% complete for SRR13349128_1_trimmed.fastq.gz\n",
      "Approx 35% complete for SRR13349128_1_trimmed.fastq.gz\n",
      "Approx 40% complete for SRR13349128_1_trimmed.fastq.gz\n",
      "Approx 45% complete for SRR13349128_1_trimmed.fastq.gz\n",
      "Approx 50% complete for SRR13349128_1_trimmed.fastq.gz\n",
      "Approx 55% complete for SRR13349128_1_trimmed.fastq.gz\n",
      "Approx 60% complete for SRR13349128_1_trimmed.fastq.gz\n",
      "Approx 65% complete for SRR13349128_1_trimmed.fastq.gz\n",
      "Approx 70% complete for SRR13349128_1_trimmed.fastq.gz\n",
      "Approx 75% complete for SRR13349128_1_trimmed.fastq.gz\n",
      "Approx 80% complete for SRR13349128_1_trimmed.fastq.gz\n",
      "Approx 85% complete for SRR13349128_1_trimmed.fastq.gz\n",
      "Approx 90% complete for SRR13349128_1_trimmed.fastq.gz\n",
      "Approx 95% complete for SRR13349128_1_trimmed.fastq.gz\n",
      "Analysis complete for SRR13349128_1_trimmed.fastq.gz\n",
      "Started analysis of SRR13349129_1_trimmed.fastq.gz\n",
      "Approx 5% complete for SRR13349129_1_trimmed.fastq.gz\n",
      "Approx 10% complete for SRR13349129_1_trimmed.fastq.gz\n",
      "Approx 15% complete for SRR13349129_1_trimmed.fastq.gz\n",
      "Approx 20% complete for SRR13349129_1_trimmed.fastq.gz\n",
      "Approx 25% complete for SRR13349129_1_trimmed.fastq.gz\n",
      "Approx 30% complete for SRR13349129_1_trimmed.fastq.gz\n",
      "Approx 35% complete for SRR13349129_1_trimmed.fastq.gz\n",
      "Approx 40% complete for SRR13349129_1_trimmed.fastq.gz\n",
      "Approx 45% complete for SRR13349129_1_trimmed.fastq.gz\n",
      "Approx 50% complete for SRR13349129_1_trimmed.fastq.gz\n",
      "Approx 55% complete for SRR13349129_1_trimmed.fastq.gz\n",
      "Approx 60% complete for SRR13349129_1_trimmed.fastq.gz\n",
      "Approx 65% complete for SRR13349129_1_trimmed.fastq.gz\n",
      "Approx 70% complete for SRR13349129_1_trimmed.fastq.gz\n",
      "Approx 75% complete for SRR13349129_1_trimmed.fastq.gz\n",
      "Approx 80% complete for SRR13349129_1_trimmed.fastq.gz\n",
      "Approx 85% complete for SRR13349129_1_trimmed.fastq.gz\n",
      "Approx 90% complete for SRR13349129_1_trimmed.fastq.gz\n",
      "Approx 95% complete for SRR13349129_1_trimmed.fastq.gz\n",
      "Analysis complete for SRR13349129_1_trimmed.fastq.gz\n",
      "Started analysis of SRR13349130_1_trimmed.fastq.gz\n",
      "Approx 5% complete for SRR13349130_1_trimmed.fastq.gz\n",
      "Approx 10% complete for SRR13349130_1_trimmed.fastq.gz\n",
      "Approx 15% complete for SRR13349130_1_trimmed.fastq.gz\n",
      "Approx 20% complete for SRR13349130_1_trimmed.fastq.gz\n",
      "Approx 25% complete for SRR13349130_1_trimmed.fastq.gz\n",
      "Approx 30% complete for SRR13349130_1_trimmed.fastq.gz\n",
      "Approx 35% complete for SRR13349130_1_trimmed.fastq.gz\n",
      "Approx 40% complete for SRR13349130_1_trimmed.fastq.gz\n",
      "Approx 45% complete for SRR13349130_1_trimmed.fastq.gz\n",
      "Approx 50% complete for SRR13349130_1_trimmed.fastq.gz\n",
      "Approx 55% complete for SRR13349130_1_trimmed.fastq.gz\n",
      "Approx 60% complete for SRR13349130_1_trimmed.fastq.gz\n",
      "Approx 65% complete for SRR13349130_1_trimmed.fastq.gz\n",
      "Approx 70% complete for SRR13349130_1_trimmed.fastq.gz\n",
      "Approx 75% complete for SRR13349130_1_trimmed.fastq.gz\n",
      "Approx 80% complete for SRR13349130_1_trimmed.fastq.gz\n",
      "Approx 85% complete for SRR13349130_1_trimmed.fastq.gz\n",
      "Approx 90% complete for SRR13349130_1_trimmed.fastq.gz\n",
      "Approx 95% complete for SRR13349130_1_trimmed.fastq.gz\n",
      "Analysis complete for SRR13349130_1_trimmed.fastq.gz\n",
      "Started analysis of SRR13349131_1_trimmed.fastq.gz\n",
      "Approx 5% complete for SRR13349131_1_trimmed.fastq.gz\n",
      "Approx 10% complete for SRR13349131_1_trimmed.fastq.gz\n",
      "Approx 15% complete for SRR13349131_1_trimmed.fastq.gz\n",
      "Approx 20% complete for SRR13349131_1_trimmed.fastq.gz\n",
      "Approx 25% complete for SRR13349131_1_trimmed.fastq.gz\n",
      "Approx 30% complete for SRR13349131_1_trimmed.fastq.gz\n",
      "Approx 35% complete for SRR13349131_1_trimmed.fastq.gz\n",
      "Approx 40% complete for SRR13349131_1_trimmed.fastq.gz\n",
      "Approx 45% complete for SRR13349131_1_trimmed.fastq.gz\n",
      "Approx 50% complete for SRR13349131_1_trimmed.fastq.gz\n",
      "Approx 55% complete for SRR13349131_1_trimmed.fastq.gz\n",
      "Approx 60% complete for SRR13349131_1_trimmed.fastq.gz\n",
      "Approx 65% complete for SRR13349131_1_trimmed.fastq.gz\n",
      "Approx 70% complete for SRR13349131_1_trimmed.fastq.gz\n",
      "Approx 75% complete for SRR13349131_1_trimmed.fastq.gz\n",
      "Approx 80% complete for SRR13349131_1_trimmed.fastq.gz\n",
      "Approx 85% complete for SRR13349131_1_trimmed.fastq.gz\n",
      "Approx 90% complete for SRR13349131_1_trimmed.fastq.gz\n",
      "Approx 95% complete for SRR13349131_1_trimmed.fastq.gz\n",
      "Analysis complete for SRR13349131_1_trimmed.fastq.gz\n",
      "Started analysis of SRR13349132_1_trimmed.fastq.gz\n",
      "Approx 5% complete for SRR13349132_1_trimmed.fastq.gz\n",
      "Approx 10% complete for SRR13349132_1_trimmed.fastq.gz\n",
      "Approx 15% complete for SRR13349132_1_trimmed.fastq.gz\n",
      "Approx 20% complete for SRR13349132_1_trimmed.fastq.gz\n",
      "Approx 25% complete for SRR13349132_1_trimmed.fastq.gz\n",
      "Approx 30% complete for SRR13349132_1_trimmed.fastq.gz\n",
      "Approx 35% complete for SRR13349132_1_trimmed.fastq.gz\n",
      "Approx 40% complete for SRR13349132_1_trimmed.fastq.gz\n",
      "Approx 45% complete for SRR13349132_1_trimmed.fastq.gz\n",
      "Approx 50% complete for SRR13349132_1_trimmed.fastq.gz\n",
      "Approx 55% complete for SRR13349132_1_trimmed.fastq.gz\n",
      "Approx 60% complete for SRR13349132_1_trimmed.fastq.gz\n",
      "Approx 65% complete for SRR13349132_1_trimmed.fastq.gz\n",
      "Approx 70% complete for SRR13349132_1_trimmed.fastq.gz\n",
      "Approx 75% complete for SRR13349132_1_trimmed.fastq.gz\n",
      "Approx 80% complete for SRR13349132_1_trimmed.fastq.gz\n",
      "Approx 85% complete for SRR13349132_1_trimmed.fastq.gz\n",
      "Approx 90% complete for SRR13349132_1_trimmed.fastq.gz\n",
      "Approx 95% complete for SRR13349132_1_trimmed.fastq.gz\n",
      "Analysis complete for SRR13349132_1_trimmed.fastq.gz\n",
      "Started analysis of SRR13349133_1_trimmed.fastq.gz\n",
      "Approx 5% complete for SRR13349133_1_trimmed.fastq.gz\n",
      "Approx 10% complete for SRR13349133_1_trimmed.fastq.gz\n",
      "Approx 15% complete for SRR13349133_1_trimmed.fastq.gz\n",
      "Approx 20% complete for SRR13349133_1_trimmed.fastq.gz\n",
      "Approx 25% complete for SRR13349133_1_trimmed.fastq.gz\n",
      "Approx 30% complete for SRR13349133_1_trimmed.fastq.gz\n",
      "Approx 35% complete for SRR13349133_1_trimmed.fastq.gz\n",
      "Approx 40% complete for SRR13349133_1_trimmed.fastq.gz\n",
      "Approx 45% complete for SRR13349133_1_trimmed.fastq.gz\n",
      "Approx 50% complete for SRR13349133_1_trimmed.fastq.gz\n",
      "Approx 55% complete for SRR13349133_1_trimmed.fastq.gz\n",
      "Approx 60% complete for SRR13349133_1_trimmed.fastq.gz\n",
      "Approx 65% complete for SRR13349133_1_trimmed.fastq.gz\n",
      "Approx 70% complete for SRR13349133_1_trimmed.fastq.gz\n",
      "Approx 75% complete for SRR13349133_1_trimmed.fastq.gz\n",
      "Approx 80% complete for SRR13349133_1_trimmed.fastq.gz\n",
      "Approx 85% complete for SRR13349133_1_trimmed.fastq.gz\n",
      "Approx 90% complete for SRR13349133_1_trimmed.fastq.gz\n",
      "Approx 95% complete for SRR13349133_1_trimmed.fastq.gz\n",
      "Analysis complete for SRR13349133_1_trimmed.fastq.gz\n"
     ]
    }
   ],
   "source": [
    "!cat data/raw_fastq/list_of_accesionIDS.txt | xargs -I {} fastqc \"data/trimmed/{}_1_trimmed.fastq.gz\" -o data/fastqc/\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### STEP 8: Run MultiQC\n",
    "MultiQC reads in the FastQQ reports and generate a compiled report for all the analyzed FASTQ files."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "  \u001b[34m/\u001b[0m\u001b[32m/\u001b[0m\u001b[31m/\u001b[0m \u001b]8;id=257313;https://multiqc.info\u001b\\\u001b[1mMultiQC\u001b[0m\u001b]8;;\u001b\\ 🔍 \u001b[2m| v1.11\u001b[0m\n",
      "\n",
      "\u001b[34m|           multiqc\u001b[0m | \u001b[33mMultiQC Version v1.12 now available!\u001b[0m\n",
      "\u001b[34m|           multiqc\u001b[0m | Search path : /home/jupyter/rnaseq-myco-notebook/data/fastqc\n",
      "\u001b[2K\u001b[34m|\u001b[0m         \u001b[34msearching\u001b[0m | \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[35m100%\u001b[0m \u001b[32m26/26\u001b[0m  rimmed_fastqc.html\u001b[0m\n",
      "\u001b[?25h\u001b[34m|            fastqc\u001b[0m | Found 12 reports\n",
      "\u001b[34m|           multiqc\u001b[0m | Compressing plot data\n",
      "\u001b[34m|           multiqc\u001b[0m | \u001b[33mDeleting    : multiqc_report.html   (-f was specified)\u001b[0m\n",
      "\u001b[34m|           multiqc\u001b[0m | \u001b[33mDeleting    : multiqc_data   (-f was specified)\u001b[0m\n",
      "\u001b[34m|           multiqc\u001b[0m | Report      : multiqc_report.html\n",
      "\u001b[34m|           multiqc\u001b[0m | Data        : multiqc_data\n",
      "\u001b[34m|           multiqc\u001b[0m | MultiQC complete\n"
     ]
    }
   ],
   "source": [
    "!multiqc -f data/fastqc/\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### STEP 9: Index the Transcriptome so that Trimmed Reads Can Be Mapped Using Salmon"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 136,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Version Info: ### PLEASE UPGRADE SALMON ###\n",
      "### A newer version of salmon with important bug fixes and improvements is available. ####\n",
      "###\n",
      "The newest version, available at https://github.com/COMBINE-lab/salmon/releases\n",
      "contains new features, improvements, and bug fixes; please upgrade at your\n",
      "earliest convenience.\n",
      "###\n",
      "Sign up for the salmon mailing list to hear about new versions, features and updates at:\n",
      "https://oceangenomics.com/subscribe\n",
      "###[2022-03-15 02:39:43.132] [jLog] [info] building index\n",
      "out : data/reference/transcriptome_index\n",
      "\u001b[00m[2022-03-15 02:39:43.132] [puff::index::jointLog] [info] Running fixFasta\n",
      "\u001b[00m\n",
      "[Step 1 of 4] : counting k-mers\n",
      "\n",
      "\u001b[35m[2022-03-15 02:39:43.356] [puff::index::jointLog] [warning] There were 1 transcripts that would need to be removed to avoid duplicates.\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:39:43.357] [puff::index::jointLog] [info] Replaced 0 non-ATCG nucleotides\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:39:43.357] [puff::index::jointLog] [info] Clipped poly-A tails from 0 transcripts\n",
      "\u001b[00mwrote 4907 cleaned references\n",
      "\u001b[00m[2022-03-15 02:39:43.376] [puff::index::jointLog] [info] Filter size not provided; estimating from number of distinct k-mers\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:39:43.469] [puff::index::jointLog] [info] ntHll estimated 4966944 distinct k-mers, setting filter size to 2^27\n",
      "\u001b[00mThreads = 8\n",
      "Vertex length = 31\n",
      "Hash functions = 5\n",
      "Filter size = 134217728\n",
      "Capacity = 2\n",
      "Files: \n",
      "data/reference/transcriptome_index/ref_k31_fixed.fa\n",
      "--------------------------------------------------------------------------------\n",
      "Round 0, 0:134217728\n",
      "Pass\tFilling\tFiltering\n",
      "1\t1\t2\t\n",
      "2\t1\t0\n",
      "True junctions count = 10210\n",
      "False junctions count = 5883\n",
      "Hash table size = 16093\n",
      "Candidate marks count = 32705\n",
      "--------------------------------------------------------------------------------\n",
      "Reallocating bifurcations time: 0\n",
      "True marks count: 21435\n",
      "Edges construction time: 25\n",
      "--------------------------------------------------------------------------------\n",
      "Distinct junctions = 10210\n",
      "\n",
      "allowedIn: 17\n",
      "Max Junction ID: 10253\n",
      "seen.size():82033 kmerInfo.size():10254\n",
      "approximateContigTotalLength: 4949554\n",
      "counters for complex kmers:\n",
      "(prec>1 & succ>1)=8 | (succ>1 & isStart)=0 | (prec>1 & isEnd)=0 | (isStart & isEnd)=1\n",
      "contig count: 10431 element count: 5330463 complex nodes: 9\n",
      "# of ones in rank vector: 10430\n",
      "\u001b[00m[2022-03-15 02:40:12.888] [puff::index::jointLog] [info] Starting the Pufferfish indexing by reading the GFA binary file.\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:40:12.888] [puff::index::jointLog] [info] Setting the index/BinaryGfa directory data/reference/transcriptome_index\n",
      "\u001b[00msize = 5330463\n",
      "-----------------------------------------\n",
      "| Loading contigs | Time = 862.17 us\n",
      "-----------------------------------------\n",
      "size = 5330463\n",
      "-----------------------------------------\n",
      "| Loading contig boundaries | Time = 465.86 us\n",
      "-----------------------------------------\n",
      "Number of ones: 10430\n",
      "Number of ones per inventory item: 512\n",
      "Inventory entries filled: 21\n",
      "10430\n",
      "\u001b[00m[2022-03-15 02:40:12.902] [puff::index::jointLog] [info] Done wrapping the rank vector with a rank9sel structure.\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:40:12.902] [puff::index::jointLog] [info] contig count for validation: 10430\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:40:12.905] [puff::index::jointLog] [info] Total # of Contigs : 10430\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:40:12.905] [puff::index::jointLog] [info] Total # of numerical Contigs : 10430\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:40:12.905] [puff::index::jointLog] [info] Total # of contig vec entries: 16558\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:40:12.905] [puff::index::jointLog] [info] bits per offset entry 15\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:40:12.905] [puff::index::jointLog] [info] Done constructing the contig vector. 10431\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:40:12.908] [puff::index::jointLog] [info] # segments = 10430\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:40:12.908] [puff::index::jointLog] [info] total length = 5330463\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:40:12.909] [puff::index::jointLog] [info] Reading the reference files ...\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:40:12.957] [puff::index::jointLog] [info] positional integer width = 23\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:40:12.957] [puff::index::jointLog] [info] seqSize = 5330463\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:40:12.957] [puff::index::jointLog] [info] rankSize = 5330463\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:40:12.957] [puff::index::jointLog] [info] edgeVecSize = 0\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:40:12.957] [puff::index::jointLog] [info] num keys = 5017563\n",
      "\u001b[00mfor info, total work write each  : 2.331    total work inram from level 3 : 4.322  total work raw : 25.000 \n",
      "[Building BooPHF]  100  %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec\n",
      "Bitarray        26296128  bits (100.00 %)   (array + ranks )\n",
      "final hash             0  bits (0.00 %) (nb in final hash 0)\n",
      "\u001b[00m[2022-03-15 02:40:13.256] [puff::index::jointLog] [info] mphf size = 3.13474 MB\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:40:13.260] [puff::index::jointLog] [info] chunk size = 666308\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:40:13.260] [puff::index::jointLog] [info] chunk 0 = [0, 666308)\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:40:13.260] [puff::index::jointLog] [info] chunk 1 = [666308, 1332616)\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:40:13.260] [puff::index::jointLog] [info] chunk 2 = [1332616, 1998924)\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:40:13.260] [puff::index::jointLog] [info] chunk 3 = [1998924, 2665232)\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:40:13.260] [puff::index::jointLog] [info] chunk 4 = [2665232, 3331540)\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:40:13.260] [puff::index::jointLog] [info] chunk 5 = [3331540, 3997848)\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:40:13.260] [puff::index::jointLog] [info] chunk 6 = [3997848, 4664156)\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:40:13.260] [puff::index::jointLog] [info] chunk 7 = [4664156, 5330433)\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:40:13.532] [puff::index::jointLog] [info] finished populating pos vector\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:40:13.532] [puff::index::jointLog] [info] writing index components\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:40:13.598] [puff::index::jointLog] [info] finished writing dense pufferfish index\n",
      "\u001b[00m[2022-03-15 02:40:13.601] [jLog] [info] done building index\n"
     ]
    }
   ],
   "source": [
    "!salmon index -t data/reference/GCF_001632805.1_transcriptome_reference_w_decoy.fa -p 8 -i data/reference/transcriptome_index --decoys data/reference/decoys.txt -k 31 --keepDuplicates"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### STEP 10: Run Salmon to Map Reads to Transcripts and Quantify Expression Levels\n",
    "Salmon aligns the trimmed reads to the reference transcriptome and generates the read counts per transcript. In this analysis, each gene has a single transcript."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 137,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Version Info: ### PLEASE UPGRADE SALMON ###\n",
      "### A newer version of salmon with important bug fixes and improvements is available. ####\n",
      "###\n",
      "The newest version, available at https://github.com/COMBINE-lab/salmon/releases\n",
      "contains new features, improvements, and bug fixes; please upgrade at your\n",
      "earliest convenience.\n",
      "###\n",
      "Sign up for the salmon mailing list to hear about new versions, features and updates at:\n",
      "https://oceangenomics.com/subscribe\n",
      "###### salmon (selective-alignment-based) v1.5.2\n",
      "### [ program ] => salmon \n",
      "### [ command ] => quant \n",
      "### [ index ] => { /home/jupyter/rnaseq-myco-notebook/data/reference/transcriptome_index }\n",
      "### [ libType ] => { SR }\n",
      "### [ unmatedReads ] => { data/trimmed/SRR13349122_1_trimmed.fastq.gz }\n",
      "### [ threads ] => { 8 }\n",
      "### [ validateMappings ] => { }\n",
      "### [ output ] => { data/quants/SRR13349122_quant }\n",
      "Logs will be written to data/quants/SRR13349122_quant/logs\n",
      "\u001b[00m[2022-03-15 02:40:36.766] [jointLog] [info] setting maxHashResizeThreads to 8\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:40:36.766] [jointLog] [info] Fragment incompatibility prior below threshold.  Incompatible fragments will be ignored.\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:40:36.766] [jointLog] [info] Usage of --validateMappings implies use of minScoreFraction. Since not explicitly specified, it is being set to 0.65\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:40:36.766] [jointLog] [info] Setting consensusSlack to selective-alignment default of 0.35.\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:40:36.766] [jointLog] [info] parsing read library format\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:40:36.779] [jointLog] [info] There is 1 library.\n",
      "\u001b[00m-----------------------------------------\n",
      "| Loading contig table | Time = 1.8167 ms\n",
      "-----------------------------------------\n",
      "size = 10431\n",
      "-----------------------------------------\n",
      "| Loading contig offsets | Time = 87.73 us\n",
      "-----------------------------------------\n",
      "-----------------------------------------\n",
      "| Loading reference lengths | Time = 18.344 us\n",
      "-----------------------------------------\n",
      "-----------------------------------------\n",
      "| Loading mphf table | Time = 701.42 us\n",
      "-----------------------------------------\n",
      "size = 5330463\n",
      "Number of ones: 10430\n",
      "Number of ones per inventory item: 512\n",
      "Inventory entries filled: 21\n",
      "\u001b[00m[2022-03-15 02:40:36.834] [jointLog] [info] Loading pufferfish index\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:40:36.834] [jointLog] [info] Loading dense pufferfish index.\n",
      "\u001b[00m-----------------------------------------\n",
      "| Loading contig boundaries | Time = 13.29 ms\n",
      "-----------------------------------------\n",
      "size = 5330463\n",
      "-----------------------------------------\n",
      "| Loading sequence | Time = 1.074 ms\n",
      "-----------------------------------------\n",
      "size = 5017563\n",
      "-----------------------------------------\n",
      "| Loading positions | Time = 8.6457 ms\n",
      "-----------------------------------------\n",
      "size = 9684819\n",
      "-----------------------------------------\n",
      "| Loading reference sequence | Time = 1.5094 ms\n",
      "-----------------------------------------\n",
      "-----------------------------------------\n",
      "| Loading reference accumulative lengths | Time = 41.181 us\n",
      "-----------------------------------------\n",
      "\u001b[00m[2022-03-15 02:40:36.862] [jointLog] [info] done\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:40:36.862] [jointLog] [info] Index contained 4907 targets\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:40:36.863] [jointLog] [info] Number of decoys : 1\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:40:36.863] [jointLog] [info] First decoy index : 4906 \n",
      "\u001b[00m\n",
      "\n",
      "\n",
      "\n",
      "\u001b[32mprocessed\u001b[31m 500001 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 1000000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 1000036 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 1500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 2000000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 2500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 3000000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 3500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 4000001 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 4500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 5000000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 5500001 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 6000000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 6500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 7000000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 7500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 8000000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 8500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 9000000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 9500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 10000000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 10500000 \u001b[32mfragments\u001b[0m\n",
      "hits: 10012965; hits per frag:  0.954886\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\u001b[00m[2022-03-15 02:40:56.364] [jointLog] [info] Thread saw mini-batch with a maximum of 0.44% zero probability fragments\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:40:56.364] [jointLog] [info] Thread saw mini-batch with a maximum of 0.40% zero probability fragments\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:40:56.364] [jointLog] [info] Thread saw mini-batch with a maximum of 0.36% zero probability fragments\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:40:56.365] [jointLog] [info] Thread saw mini-batch with a maximum of 0.36% zero probability fragments\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:40:56.365] [jointLog] [info] Thread saw mini-batch with a maximum of 0.34% zero probability fragments\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:40:56.367] [jointLog] [info] Thread saw mini-batch with a maximum of 0.34% zero probability fragments\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:40:56.373] [jointLog] [info] Thread saw mini-batch with a maximum of 0.42% zero probability fragments\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:40:56.377] [jointLog] [info] Thread saw mini-batch with a maximum of 0.36% zero probability fragments\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:40:56.394] [jointLog] [info] Computed 4595 rich equivalence classes for further processing\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:40:56.394] [jointLog] [info] Counted 10287659 total reads in the equivalence classes \n",
      "\u001b[00m\u001b[00m[2022-03-15 02:40:56.398] [jointLog] [info] Number of mappings discarded because of alignment score : 152096\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:40:56.398] [jointLog] [info] Number of fragments entirely discarded because of alignment score : 133954\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:40:56.398] [jointLog] [info] Number of fragments discarded because they are best-mapped to decoys : 72939\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:40:56.398] [jointLog] [info] Number of fragments discarded because they have only dovetail (discordant) mappings to valid targets : 0\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:40:56.398] [jointLog] [info] Mapping rate = 95.1656%\n",
      "\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:40:56.398] [jointLog] [info] finished quantifyLibrary()\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:40:56.398] [jointLog] [info] Starting optimizer\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:40:56.401] [jointLog] [info] Marked 0 weighted equivalence classes as degenerate\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:40:56.401] [jointLog] [info] iteration = 0 | max rel diff. = 3330.2\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:40:56.461] [jointLog] [info] iteration = 100 | max rel diff. = 1.62461e-16\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:40:56.462] [jointLog] [info] Finished optimizer\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:40:56.462] [jointLog] [info] writing output \n",
      "\n",
      "\u001b[00mVersion Info: ### PLEASE UPGRADE SALMON ###\n",
      "### A newer version of salmon with important bug fixes and improvements is available. ####\n",
      "###\n",
      "The newest version, available at https://github.com/COMBINE-lab/salmon/releases\n",
      "contains new features, improvements, and bug fixes; please upgrade at your\n",
      "earliest convenience.\n",
      "###\n",
      "Sign up for the salmon mailing list to hear about new versions, features and updates at:\n",
      "https://oceangenomics.com/subscribe\n",
      "###### salmon (selective-alignment-based) v1.5.2\n",
      "### [ program ] => salmon \n",
      "### [ command ] => quant \n",
      "### [ index ] => { /home/jupyter/rnaseq-myco-notebook/data/reference/transcriptome_index }\n",
      "### [ libType ] => { SR }\n",
      "### [ unmatedReads ] => { data/trimmed/SRR13349123_1_trimmed.fastq.gz }\n",
      "### [ threads ] => { 8 }\n",
      "### [ validateMappings ] => { }\n",
      "### [ output ] => { data/quants/SRR13349123_quant }\n",
      "Logs will be written to data/quants/SRR13349123_quant/logs\n",
      "\u001b[00m[2022-03-15 02:40:57.016] [jointLog] [info] setting maxHashResizeThreads to 8\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:40:57.016] [jointLog] [info] Fragment incompatibility prior below threshold.  Incompatible fragments will be ignored.\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:40:57.016] [jointLog] [info] Usage of --validateMappings implies use of minScoreFraction. Since not explicitly specified, it is being set to 0.65\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:40:57.016] [jointLog] [info] Setting consensusSlack to selective-alignment default of 0.35.\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:40:57.016] [jointLog] [info] parsing read library format\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:40:57.016] [jointLog] [info] There is 1 library.\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:40:57.057] [jointLog] [info] Loading pufferfish index\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:40:57.057] [jointLog] [info] Loading dense pufferfish index.\n",
      "\u001b[00m-----------------------------------------\n",
      "| Loading contig table | Time = 1.7575 ms\n",
      "-----------------------------------------\n",
      "size = 10431\n",
      "-----------------------------------------\n",
      "| Loading contig offsets | Time = 119.31 us\n",
      "-----------------------------------------\n",
      "-----------------------------------------\n",
      "| Loading reference lengths | Time = 20.348 us\n",
      "-----------------------------------------\n",
      "-----------------------------------------\n",
      "| Loading mphf table | Time = 804.2 us\n",
      "-----------------------------------------\n",
      "size = 5330463\n",
      "Number of ones: 10430\n",
      "Number of ones per inventory item: 512\n",
      "Inventory entries filled: 21\n",
      "-----------------------------------------\n",
      "| Loading contig boundaries | Time = 13.051 ms\n",
      "-----------------------------------------\n",
      "size = 5330463\n",
      "-----------------------------------------\n",
      "| Loading sequence | Time = 890.92 us\n",
      "-----------------------------------------\n",
      "size = 5017563\n",
      "-----------------------------------------\n",
      "| Loading positions | Time = 9.1443 ms\n",
      "-----------------------------------------\n",
      "size = 9684819\n",
      "-----------------------------------------\n",
      "| Loading reference sequence | Time = 1.5942 ms\n",
      "-----------------------------------------\n",
      "-----------------------------------------\n",
      "| Loading reference accumulative lengths | Time = 35.615 us\n",
      "-----------------------------------------\n",
      "\u001b[00m[2022-03-15 02:40:57.084] [jointLog] [info] done\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:40:57.084] [jointLog] [info] Index contained 4907 targets\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:40:57.086] [jointLog] [info] Number of decoys : 1\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:40:57.086] [jointLog] [info] First decoy index : 4906 \n",
      "\u001b[00m\n",
      "\n",
      "\n",
      "\n",
      "\u001b[32mprocessed\u001b[31m 500001 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 1000000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 1500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 2000000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 2500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 3000000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 3500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 4000001 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 4500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 5000000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 5500001 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 6500002 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 7000001 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 7500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 8000001 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 8500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 9000000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 9500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 10000000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 10500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 11000000 \u001b[32mfragments\u001b[0m\n",
      "hits: 10546700; hits per frag:  0.959752\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\u001b[00m[2022-03-15 02:41:16.591] [jointLog] [info] Thread saw mini-batch with a maximum of 0.48% zero probability fragments\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:41:16.597] [jointLog] [info] Thread saw mini-batch with a maximum of 0.44% zero probability fragments\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:41:16.605] [jointLog] [info] Thread saw mini-batch with a maximum of 0.48% zero probability fragments\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:41:16.609] [jointLog] [info] Thread saw mini-batch with a maximum of 0.46% zero probability fragments\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:41:16.610] [jointLog] [info] Thread saw mini-batch with a maximum of 0.46% zero probability fragments\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:41:16.611] [jointLog] [info] Thread saw mini-batch with a maximum of 0.42% zero probability fragments\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:41:16.613] [jointLog] [info] Thread saw mini-batch with a maximum of 0.40% zero probability fragments\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:41:16.614] [jointLog] [info] Thread saw mini-batch with a maximum of 0.40% zero probability fragments\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:41:16.631] [jointLog] [info] Computed 4657 rich equivalence classes for further processing\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:41:16.631] [jointLog] [info] Counted 10613007 total reads in the equivalence classes \n",
      "\u001b[00m\u001b[00m[2022-03-15 02:41:16.635] [jointLog] [info] Number of mappings discarded because of alignment score : 158751\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:41:16.635] [jointLog] [info] Number of fragments entirely discarded because of alignment score : 143045\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:41:16.635] [jointLog] [info] Number of fragments discarded because they are best-mapped to decoys : 80723\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:41:16.635] [jointLog] [info] Number of fragments discarded because they have only dovetail (discordant) mappings to valid targets : 0\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:41:16.635] [jointLog] [info] Mapping rate = 95.6449%\n",
      "\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:41:16.635] [jointLog] [info] finished quantifyLibrary()\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:41:16.635] [jointLog] [info] Starting optimizer\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:41:16.638] [jointLog] [info] Marked 0 weighted equivalence classes as degenerate\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:41:16.638] [jointLog] [info] iteration = 0 | max rel diff. = 3407.4\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:41:16.713] [jointLog] [info] iteration = 100 | max rel diff. = 2.09062e-16\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:41:16.714] [jointLog] [info] Finished optimizer\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:41:16.714] [jointLog] [info] writing output \n",
      "\n",
      "\u001b[00mVersion Info: ### PLEASE UPGRADE SALMON ###\n",
      "### A newer version of salmon with important bug fixes and improvements is available. ####\n",
      "###\n",
      "The newest version, available at https://github.com/COMBINE-lab/salmon/releases\n",
      "contains new features, improvements, and bug fixes; please upgrade at your\n",
      "earliest convenience.\n",
      "###\n",
      "Sign up for the salmon mailing list to hear about new versions, features and updates at:\n",
      "https://oceangenomics.com/subscribe\n",
      "###### salmon (selective-alignment-based) v1.5.2\n",
      "### [ program ] => salmon \n",
      "### [ command ] => quant \n",
      "### [ index ] => { /home/jupyter/rnaseq-myco-notebook/data/reference/transcriptome_index }\n",
      "### [ libType ] => { SR }\n",
      "### [ unmatedReads ] => { data/trimmed/SRR13349124_1_trimmed.fastq.gz }\n",
      "### [ threads ] => { 8 }\n",
      "### [ validateMappings ] => { }\n",
      "### [ output ] => { data/quants/SRR13349124_quant }\n",
      "Logs will be written to data/quants/SRR13349124_quant/logs\n",
      "\u001b[00m[2022-03-15 02:41:17.260] [jointLog] [info] setting maxHashResizeThreads to 8\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:41:17.260] [jointLog] [info] Fragment incompatibility prior below threshold.  Incompatible fragments will be ignored.\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:41:17.260] [jointLog] [info] Usage of --validateMappings implies use of minScoreFraction. Since not explicitly specified, it is being set to 0.65\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:41:17.260] [jointLog] [info] Setting consensusSlack to selective-alignment default of 0.35.\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:41:17.260] [jointLog] [info] parsing read library format\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:41:17.260] [jointLog] [info] There is 1 library.\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:41:17.300] [jointLog] [info] Loading pufferfish index\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:41:17.300] [jointLog] [info] Loading dense pufferfish index.\n",
      "\u001b[00m-----------------------------------------\n",
      "| Loading contig table | Time = 1.8107 ms\n",
      "-----------------------------------------\n",
      "size = 10431\n",
      "-----------------------------------------\n",
      "| Loading contig offsets | Time = 108.88 us\n",
      "-----------------------------------------\n",
      "-----------------------------------------\n",
      "| Loading reference lengths | Time = 19.766 us\n",
      "-----------------------------------------\n",
      "-----------------------------------------\n",
      "| Loading mphf table | Time = 825.32 us\n",
      "-----------------------------------------\n",
      "size = 5330463\n",
      "Number of ones: 10430\n",
      "Number of ones per inventory item: 512\n",
      "Inventory entries filled: 21\n",
      "-----------------------------------------\n",
      "| Loading contig boundaries | Time = 13.402 ms\n",
      "-----------------------------------------\n",
      "size = 5330463\n",
      "-----------------------------------------\n",
      "| Loading sequence | Time = 1.0373 ms\n",
      "-----------------------------------------\n",
      "size = 5017563\n",
      "-----------------------------------------\n",
      "| Loading positions | Time = 9.5193 ms\n",
      "-----------------------------------------\n",
      "size = 9684819\n",
      "-----------------------------------------\n",
      "| Loading reference sequence | Time = 1.6964 ms\n",
      "-----------------------------------------\n",
      "-----------------------------------------\n",
      "| Loading reference accumulative lengths | Time = 42.604 us\n",
      "-----------------------------------------\n",
      "\u001b[00m[2022-03-15 02:41:17.329] [jointLog] [info] done\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:41:17.329] [jointLog] [info] Index contained 4907 targets\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:41:17.331] [jointLog] [info] Number of decoys : 1\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:41:17.331] [jointLog] [info] First decoy index : 4906 \n",
      "\u001b[00m\n",
      "\n",
      "\n",
      "\n",
      "\u001b[32mprocessed\u001b[31m 500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 1000000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 1500001 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 2000000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 2500002 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 3000001 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 3500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 4000000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 4500001 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 5000000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 5500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 6000000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 6500001 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 7000000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 7500001 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 8000000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 8500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 9000000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 9500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 10000001 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 10500000 \u001b[32mfragments\u001b[0m\n",
      "hits: 9990840; hits per frag:  0.956142\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\u001b[00m[2022-03-15 02:41:36.358] [jointLog] [info] Thread saw mini-batch with a maximum of 0.32% zero probability fragments\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:41:36.359] [jointLog] [info] Thread saw mini-batch with a maximum of 0.28% zero probability fragments\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:41:36.359] [jointLog] [info] Thread saw mini-batch with a maximum of 0.36% zero probability fragments\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:41:36.360] [jointLog] [info] Thread saw mini-batch with a maximum of 0.30% zero probability fragments\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:41:36.369] [jointLog] [info] Thread saw mini-batch with a maximum of 0.30% zero probability fragments\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:41:36.369] [jointLog] [info] Thread saw mini-batch with a maximum of 0.34% zero probability fragments\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:41:36.372] [jointLog] [info] Thread saw mini-batch with a maximum of 0.32% zero probability fragments\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:41:36.378] [jointLog] [info] Thread saw mini-batch with a maximum of 0.32% zero probability fragments\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:41:36.396] [jointLog] [info] Computed 4575 rich equivalence classes for further processing\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:41:36.396] [jointLog] [info] Counted 10172390 total reads in the equivalence classes \n",
      "\u001b[00m\u001b[00m[2022-03-15 02:41:36.399] [jointLog] [info] Number of mappings discarded because of alignment score : 187877\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:41:36.399] [jointLog] [info] Number of fragments entirely discarded because of alignment score : 144611\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:41:36.399] [jointLog] [info] Number of fragments discarded because they are best-mapped to decoys : 64442\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:41:36.399] [jointLog] [info] Number of fragments discarded because they have only dovetail (discordant) mappings to valid targets : 0\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:41:36.399] [jointLog] [info] Mapping rate = 94.9788%\n",
      "\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:41:36.399] [jointLog] [info] finished quantifyLibrary()\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:41:36.399] [jointLog] [info] Starting optimizer\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:41:36.402] [jointLog] [info] Marked 0 weighted equivalence classes as degenerate\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:41:36.403] [jointLog] [info] iteration = 0 | max rel diff. = 1650.82\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:41:36.464] [jointLog] [info] iteration = 100 | max rel diff. = 1.4345e-16\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:41:36.465] [jointLog] [info] Finished optimizer\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:41:36.465] [jointLog] [info] writing output \n",
      "\n",
      "\u001b[00mVersion Info: ### PLEASE UPGRADE SALMON ###\n",
      "### A newer version of salmon with important bug fixes and improvements is available. ####\n",
      "###\n",
      "The newest version, available at https://github.com/COMBINE-lab/salmon/releases\n",
      "contains new features, improvements, and bug fixes; please upgrade at your\n",
      "earliest convenience.\n",
      "###\n",
      "Sign up for the salmon mailing list to hear about new versions, features and updates at:\n",
      "https://oceangenomics.com/subscribe\n",
      "###### salmon (selective-alignment-based) v1.5.2\n",
      "### [ program ] => salmon \n",
      "### [ command ] => quant \n",
      "### [ index ] => { /home/jupyter/rnaseq-myco-notebook/data/reference/transcriptome_index }\n",
      "### [ libType ] => { SR }\n",
      "### [ unmatedReads ] => { data/trimmed/SRR13349125_1_trimmed.fastq.gz }\n",
      "### [ threads ] => { 8 }\n",
      "### [ validateMappings ] => { }\n",
      "### [ output ] => { data/quants/SRR13349125_quant }\n",
      "Logs will be written to data/quants/SRR13349125_quant/logs\n",
      "\u001b[00m[2022-03-15 02:41:36.993] [jointLog] [info] setting maxHashResizeThreads to 8\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:41:36.993] [jointLog] [info] Fragment incompatibility prior below threshold.  Incompatible fragments will be ignored.\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:41:36.993] [jointLog] [info] Usage of --validateMappings implies use of minScoreFraction. Since not explicitly specified, it is being set to 0.65\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:41:36.993] [jointLog] [info] Setting consensusSlack to selective-alignment default of 0.35.\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:41:36.993] [jointLog] [info] parsing read library format\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:41:36.993] [jointLog] [info] There is 1 library.\n",
      "\u001b[00m-----------------------------------------\n",
      "| Loading contig table | Time = 1.8722 ms\n",
      "-----------------------------------------\n",
      "size = 10431\n",
      "-----------------------------------------\n",
      "| Loading contig offsets | Time = 90.36 us\n",
      "-----------------------------------------\n",
      "-----------------------------------------\n",
      "| Loading reference lengths | Time = 19.053 us\n",
      "-----------------------------------------\n",
      "-----------------------------------------\n",
      "| Loading mphf table | Time = 748.06 us\n",
      "-----------------------------------------\n",
      "size = 5330463\n",
      "Number of ones: 10430\n",
      "Number of ones per inventory item: 512\n",
      "Inventory entries filled: 21\n",
      "-----------------------------------------\n",
      "| Loading contig boundaries | Time = 12.904 ms\n",
      "-----------------------------------------\n",
      "size = 5330463\n",
      "-----------------------------------------\n",
      "| Loading sequence | Time = 948.45 us\n",
      "-----------------------------------------\n",
      "size = 5017563\n",
      "\u001b[00m[2022-03-15 02:41:37.034] [jointLog] [info] Loading pufferfish index\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:41:37.034] [jointLog] [info] Loading dense pufferfish index.\n",
      "\u001b[00m-----------------------------------------\n",
      "| Loading positions | Time = 9.4835 ms\n",
      "-----------------------------------------\n",
      "size = 9684819\n",
      "-----------------------------------------\n",
      "| Loading reference sequence | Time = 1.5448 ms\n",
      "-----------------------------------------\n",
      "-----------------------------------------\n",
      "| Loading reference accumulative lengths | Time = 34.814 us\n",
      "-----------------------------------------\n",
      "\u001b[00m[2022-03-15 02:41:37.062] [jointLog] [info] done\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:41:37.062] [jointLog] [info] Index contained 4907 targets\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:41:37.064] [jointLog] [info] Number of decoys : 1\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:41:37.064] [jointLog] [info] First decoy index : 4906 \n",
      "\u001b[00m\n",
      "\n",
      "\n",
      "\n",
      "\u001b[32mprocessed\u001b[31m 500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 1000000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 1500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 2000001 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 2500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 3000000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 3500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 4000000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 4500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 5000000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 5500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 6000000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 6500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 7000001 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 7500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 8000000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 8500001 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 9000001 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 9500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 10000000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 10500000 \u001b[32mfragments\u001b[0m\n",
      "hits: 10044455; hits per frag:  0.957714\u001b[00m[2022-03-15 02:41:56.258] [jointLog] [info] Thread saw mini-batch with a maximum of 0.38% zero probability fragments\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:41:56.259] [jointLog] [info] Thread saw mini-batch with a maximum of 0.36% zero probability fragments\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:41:56.259] [jointLog] [info] Thread saw mini-batch with a maximum of 0.36% zero probability fragments\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:41:56.263] [jointLog] [info] Thread saw mini-batch with a maximum of 0.42% zero probability fragments\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:41:56.266] [jointLog] [info] Thread saw mini-batch with a maximum of 0.40% zero probability fragments\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:41:56.278] [jointLog] [info] Thread saw mini-batch with a maximum of 0.38% zero probability fragments\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:41:56.279] [jointLog] [info] Thread saw mini-batch with a maximum of 0.38% zero probability fragments\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:41:56.282] [jointLog] [info] Thread saw mini-batch with a maximum of 0.38% zero probability fragments\n",
      "\u001b[00m\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\u001b[00m[2022-03-15 02:41:56.300] [jointLog] [info] Computed 4586 rich equivalence classes for further processing\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:41:56.300] [jointLog] [info] Counted 10429221 total reads in the equivalence classes \n",
      "\u001b[00m\u001b[00m[2022-03-15 02:41:56.303] [jointLog] [info] Number of mappings discarded because of alignment score : 190142\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:41:56.303] [jointLog] [info] Number of fragments entirely discarded because of alignment score : 150020\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:41:56.303] [jointLog] [info] Number of fragments discarded because they are best-mapped to decoys : 70256\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:41:56.303] [jointLog] [info] Number of fragments discarded because they have only dovetail (discordant) mappings to valid targets : 0\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:41:56.303] [jointLog] [info] Mapping rate = 95.4666%\n",
      "\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:41:56.303] [jointLog] [info] finished quantifyLibrary()\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:41:56.304] [jointLog] [info] Starting optimizer\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:41:56.307] [jointLog] [info] Marked 0 weighted equivalence classes as degenerate\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:41:56.308] [jointLog] [info] iteration = 0 | max rel diff. = 1681.61\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:41:56.369] [jointLog] [info] iteration = 100 | max rel diff. = 1.62169e-16\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:41:56.370] [jointLog] [info] Finished optimizer\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:41:56.370] [jointLog] [info] writing output \n",
      "\n",
      "\u001b[00mVersion Info: ### PLEASE UPGRADE SALMON ###\n",
      "### A newer version of salmon with important bug fixes and improvements is available. ####\n",
      "###\n",
      "The newest version, available at https://github.com/COMBINE-lab/salmon/releases\n",
      "contains new features, improvements, and bug fixes; please upgrade at your\n",
      "earliest convenience.\n",
      "###\n",
      "Sign up for the salmon mailing list to hear about new versions, features and updates at:\n",
      "https://oceangenomics.com/subscribe\n",
      "###### salmon (selective-alignment-based) v1.5.2\n",
      "### [ program ] => salmon \n",
      "### [ command ] => quant \n",
      "### [ index ] => { /home/jupyter/rnaseq-myco-notebook/data/reference/transcriptome_index }\n",
      "### [ libType ] => { SR }\n",
      "### [ unmatedReads ] => { data/trimmed/SRR13349126_1_trimmed.fastq.gz }\n",
      "### [ threads ] => { 8 }\n",
      "### [ validateMappings ] => { }\n",
      "### [ output ] => { data/quants/SRR13349126_quant }\n",
      "Logs will be written to data/quants/SRR13349126_quant/logs\n",
      "\u001b[00m[2022-03-15 02:41:56.751] [jointLog] [info] setting maxHashResizeThreads to 8\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:41:56.751] [jointLog] [info] Fragment incompatibility prior below threshold.  Incompatible fragments will be ignored.\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:41:56.751] [jointLog] [info] Usage of --validateMappings implies use of minScoreFraction. Since not explicitly specified, it is being set to 0.65\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:41:56.751] [jointLog] [info] Setting consensusSlack to selective-alignment default of 0.35.\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:41:56.751] [jointLog] [info] parsing read library format\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:41:56.751] [jointLog] [info] There is 1 library.\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:41:56.792] [jointLog] [info] Loading pufferfish index\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:41:56.792] [jointLog] [info] Loading dense pufferfish index.\n",
      "\u001b[00m-----------------------------------------\n",
      "| Loading contig table | Time = 1.8016 ms\n",
      "-----------------------------------------\n",
      "size = 10431\n",
      "-----------------------------------------\n",
      "| Loading contig offsets | Time = 96.738 us\n",
      "-----------------------------------------\n",
      "-----------------------------------------\n",
      "| Loading reference lengths | Time = 15.376 us\n",
      "-----------------------------------------\n",
      "-----------------------------------------\n",
      "| Loading mphf table | Time = 714.16 us\n",
      "-----------------------------------------\n",
      "size = 5330463\n",
      "Number of ones: 10430\n",
      "Number of ones per inventory item: 512\n",
      "Inventory entries filled: 21\n",
      "-----------------------------------------\n",
      "| Loading contig boundaries | Time = 12.994 ms\n",
      "-----------------------------------------\n",
      "size = 5330463\n",
      "-----------------------------------------\n",
      "| Loading sequence | Time = 912.18 us\n",
      "-----------------------------------------\n",
      "size = 5017563\n",
      "-----------------------------------------\n",
      "| Loading positions | Time = 8.9263 ms\n",
      "-----------------------------------------\n",
      "size = 9684819\n",
      "-----------------------------------------\n",
      "| Loading reference sequence | Time = 1.7347 ms\n",
      "-----------------------------------------\n",
      "-----------------------------------------\n",
      "| Loading reference accumulative lengths | Time = 36.769 us\n",
      "-----------------------------------------\n",
      "\u001b[00m[2022-03-15 02:41:56.820] [jointLog] [info] done\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:41:56.820] [jointLog] [info] Index contained 4907 targets\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:41:56.821] [jointLog] [info] Number of decoys : 1\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:41:56.821] [jointLog] [info] First decoy index : 4906 \n",
      "\u001b[00m\n",
      "\n",
      "\n",
      "\n",
      "\u001b[32mprocessed\u001b[31m 500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 1000000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 1500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 2000001 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 2500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 3000001 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 3500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 4000000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 4500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 5000000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 5500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 6000001 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 6500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 7000000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 7500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 8000000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 8500001 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 9000000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 9500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 10000001 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 10500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 11000000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 11500001 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 12000000 \u001b[32mfragments\u001b[0m\n",
      "hits: 11496209; hits per frag:  0.960171\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\u001b[00m[2022-03-15 02:42:18.568] [jointLog] [info] Thread saw mini-batch with a maximum of 0.28% zero probability fragments\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:42:18.572] [jointLog] [info] Thread saw mini-batch with a maximum of 0.28% zero probability fragments\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:42:18.577] [jointLog] [info] Thread saw mini-batch with a maximum of 0.30% zero probability fragments\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:42:18.579] [jointLog] [info] Thread saw mini-batch with a maximum of 0.32% zero probability fragments\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:42:18.581] [jointLog] [info] Thread saw mini-batch with a maximum of 0.28% zero probability fragments\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:42:18.581] [jointLog] [info] Thread saw mini-batch with a maximum of 0.36% zero probability fragments\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:42:18.584] [jointLog] [info] Thread saw mini-batch with a maximum of 0.30% zero probability fragments\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:42:18.586] [jointLog] [info] Thread saw mini-batch with a maximum of 0.28% zero probability fragments\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:42:18.604] [jointLog] [info] Computed 4591 rich equivalence classes for further processing\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:42:18.604] [jointLog] [info] Counted 11715938 total reads in the equivalence classes \n",
      "\u001b[00m\u001b[00m[2022-03-15 02:42:18.607] [jointLog] [info] Number of mappings discarded because of alignment score : 186660\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:42:18.607] [jointLog] [info] Number of fragments entirely discarded because of alignment score : 144709\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:42:18.607] [jointLog] [info] Number of fragments discarded because they are best-mapped to decoys : 65046\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:42:18.607] [jointLog] [info] Number of fragments discarded because they have only dovetail (discordant) mappings to valid targets : 0\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:42:18.607] [jointLog] [info] Mapping rate = 95.6559%\n",
      "\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:42:18.607] [jointLog] [info] finished quantifyLibrary()\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:42:18.608] [jointLog] [info] Starting optimizer\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:42:18.611] [jointLog] [info] Marked 0 weighted equivalence classes as degenerate\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:42:18.611] [jointLog] [info] iteration = 0 | max rel diff. = 1827.75\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:42:18.673] [jointLog] [info] iteration = 100 | max rel diff. = 1.34367e-16\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:42:18.674] [jointLog] [info] Finished optimizer\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:42:18.674] [jointLog] [info] writing output \n",
      "\n",
      "\u001b[00mVersion Info: ### PLEASE UPGRADE SALMON ###\n",
      "### A newer version of salmon with important bug fixes and improvements is available. ####\n",
      "###\n",
      "The newest version, available at https://github.com/COMBINE-lab/salmon/releases\n",
      "contains new features, improvements, and bug fixes; please upgrade at your\n",
      "earliest convenience.\n",
      "###\n",
      "Sign up for the salmon mailing list to hear about new versions, features and updates at:\n",
      "https://oceangenomics.com/subscribe\n",
      "###### salmon (selective-alignment-based) v1.5.2\n",
      "### [ program ] => salmon \n",
      "### [ command ] => quant \n",
      "### [ index ] => { /home/jupyter/rnaseq-myco-notebook/data/reference/transcriptome_index }\n",
      "### [ libType ] => { SR }\n",
      "### [ unmatedReads ] => { data/trimmed/SRR13349127_1_trimmed.fastq.gz }\n",
      "### [ threads ] => { 8 }\n",
      "### [ validateMappings ] => { }\n",
      "### [ output ] => { data/quants/SRR13349127_quant }\n",
      "Logs will be written to data/quants/SRR13349127_quant/logs\n",
      "\u001b[00m[2022-03-15 02:42:19.509] [jointLog] [info] setting maxHashResizeThreads to 8\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:42:19.509] [jointLog] [info] Fragment incompatibility prior below threshold.  Incompatible fragments will be ignored.\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:42:19.509] [jointLog] [info] Usage of --validateMappings implies use of minScoreFraction. Since not explicitly specified, it is being set to 0.65\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:42:19.509] [jointLog] [info] Setting consensusSlack to selective-alignment default of 0.35.\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:42:19.509] [jointLog] [info] parsing read library format\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:42:19.509] [jointLog] [info] There is 1 library.\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:42:19.549] [jointLog] [info] Loading pufferfish index\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:42:19.549] [jointLog] [info] Loading dense pufferfish index.\n",
      "\u001b[00m-----------------------------------------\n",
      "| Loading contig table | Time = 1.9179 ms\n",
      "-----------------------------------------\n",
      "size = 10431\n",
      "-----------------------------------------\n",
      "| Loading contig offsets | Time = 94.318 us\n",
      "-----------------------------------------\n",
      "-----------------------------------------\n",
      "| Loading reference lengths | Time = 18.154 us\n",
      "-----------------------------------------\n",
      "-----------------------------------------\n",
      "| Loading mphf table | Time = 749.39 us\n",
      "-----------------------------------------\n",
      "size = 5330463\n",
      "Number of ones: 10430\n",
      "Number of ones per inventory item: 512\n",
      "Inventory entries filled: 21\n",
      "-----------------------------------------\n",
      "| Loading contig boundaries | Time = 12.908 ms\n",
      "-----------------------------------------\n",
      "size = 5330463\n",
      "-----------------------------------------\n",
      "| Loading sequence | Time = 907.4 us\n",
      "-----------------------------------------\n",
      "size = 5017563\n",
      "-----------------------------------------\n",
      "| Loading positions | Time = 8.5146 ms\n",
      "-----------------------------------------\n",
      "size = 9684819\n",
      "-----------------------------------------\n",
      "| Loading reference sequence | Time = 1.5221 ms\n",
      "-----------------------------------------\n",
      "-----------------------------------------\n",
      "| Loading reference accumulative lengths | Time = 37.322 us\n",
      "-----------------------------------------\n",
      "\u001b[00m[2022-03-15 02:42:19.576] [jointLog] [info] done\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:42:19.576] [jointLog] [info] Index contained 4907 targets\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:42:19.577] [jointLog] [info] Number of decoys : 1\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:42:19.577] [jointLog] [info] First decoy index : 4906 \n",
      "\u001b[00m\n",
      "\n",
      "\n",
      "\n",
      "\u001b[32mprocessed\u001b[31m 500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 1000001 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 1500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 2000000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 2500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 3000001 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 3500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 4000001 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 4500002 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 5000000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 5500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 6000001 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 6500001 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 7000000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 7500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 8000001 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 8500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 9000000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 9500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 10000000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 10500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 11000000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 11500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 12000000 \u001b[32mfragments\u001b[0m\n",
      "hits: 11555795; hits per frag:  0.965195\n",
      "\n",
      "\n",
      "\n",
      "\u001b[00m[2022-03-15 02:42:41.259] [jointLog] [info] Thread saw mini-batch with a maximum of 0.32% zero probability fragments\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:42:41.261] [jointLog] [info] Thread saw mini-batch with a maximum of 0.32% zero probability fragments\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:42:41.261] [jointLog] [info] Thread saw mini-batch with a maximum of 0.36% zero probability fragments\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:42:41.273] [jointLog] [info] Thread saw mini-batch with a maximum of 0.36% zero probability fragments\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:42:41.276] [jointLog] [info] Thread saw mini-batch with a maximum of 0.38% zero probability fragments\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:42:41.279] [jointLog] [info] Thread saw mini-batch with a maximum of 0.38% zero probability fragments\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:42:41.279] [jointLog] [info] Thread saw mini-batch with a maximum of 0.36% zero probability fragments\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:42:41.284] [jointLog] [info] Thread saw mini-batch with a maximum of 0.32% zero probability fragments\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:42:41.302] [jointLog] [info] Computed 4632 rich equivalence classes for further processing\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:42:41.302] [jointLog] [info] Counted 12002454 total reads in the equivalence classes \n",
      "\u001b[00m\n",
      "\n",
      "\n",
      "\n",
      "\u001b[00m[2022-03-15 02:42:41.306] [jointLog] [info] Number of mappings discarded because of alignment score : 188621\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:42:41.306] [jointLog] [info] Number of fragments entirely discarded because of alignment score : 150239\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:42:41.306] [jointLog] [info] Number of fragments discarded because they are best-mapped to decoys : 70855\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:42:41.306] [jointLog] [info] Number of fragments discarded because they have only dovetail (discordant) mappings to valid targets : 0\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:42:41.306] [jointLog] [info] Mapping rate = 96.1307%\n",
      "\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:42:41.306] [jointLog] [info] finished quantifyLibrary()\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:42:41.307] [jointLog] [info] Starting optimizer\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:42:41.310] [jointLog] [info] Marked 0 weighted equivalence classes as degenerate\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:42:41.310] [jointLog] [info] iteration = 0 | max rel diff. = 3717.9\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:42:41.372] [jointLog] [info] iteration = 100 | max rel diff. = 2.76183e-16\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:42:41.373] [jointLog] [info] Finished optimizer\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:42:41.373] [jointLog] [info] writing output \n",
      "\n",
      "\u001b[00mVersion Info: ### PLEASE UPGRADE SALMON ###\n",
      "### A newer version of salmon with important bug fixes and improvements is available. ####\n",
      "###\n",
      "The newest version, available at https://github.com/COMBINE-lab/salmon/releases\n",
      "contains new features, improvements, and bug fixes; please upgrade at your\n",
      "earliest convenience.\n",
      "###\n",
      "Sign up for the salmon mailing list to hear about new versions, features and updates at:\n",
      "https://oceangenomics.com/subscribe\n",
      "###### salmon (selective-alignment-based) v1.5.2\n",
      "### [ program ] => salmon \n",
      "### [ command ] => quant \n",
      "### [ index ] => { /home/jupyter/rnaseq-myco-notebook/data/reference/transcriptome_index }\n",
      "### [ libType ] => { SR }\n",
      "### [ unmatedReads ] => { data/trimmed/SRR13349128_1_trimmed.fastq.gz }\n",
      "### [ threads ] => { 8 }\n",
      "### [ validateMappings ] => { }\n",
      "### [ output ] => { data/quants/SRR13349128_quant }\n",
      "Logs will be written to data/quants/SRR13349128_quant/logs\n",
      "\u001b[00m[2022-03-15 02:42:42.250] [jointLog] [info] setting maxHashResizeThreads to 8\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:42:42.250] [jointLog] [info] Fragment incompatibility prior below threshold.  Incompatible fragments will be ignored.\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:42:42.250] [jointLog] [info] Usage of --validateMappings implies use of minScoreFraction. Since not explicitly specified, it is being set to 0.65\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:42:42.250] [jointLog] [info] Setting consensusSlack to selective-alignment default of 0.35.\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:42:42.250] [jointLog] [info] parsing read library format\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:42:42.250] [jointLog] [info] There is 1 library.\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:42:42.289] [jointLog] [info] Loading pufferfish index\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:42:42.289] [jointLog] [info] Loading dense pufferfish index.\n",
      "\u001b[00m-----------------------------------------\n",
      "| Loading contig table | Time = 2.0064 ms\n",
      "-----------------------------------------\n",
      "size = 10431\n",
      "-----------------------------------------\n",
      "| Loading contig offsets | Time = 93.476 us\n",
      "-----------------------------------------\n",
      "-----------------------------------------\n",
      "| Loading reference lengths | Time = 15.975 us\n",
      "-----------------------------------------\n",
      "-----------------------------------------\n",
      "| Loading mphf table | Time = 750.42 us\n",
      "-----------------------------------------\n",
      "size = 5330463\n",
      "Number of ones: 10430\n",
      "Number of ones per inventory item: 512\n",
      "Inventory entries filled: 21\n",
      "-----------------------------------------\n",
      "| Loading contig boundaries | Time = 12.996 ms\n",
      "-----------------------------------------\n",
      "size = 5330463\n",
      "-----------------------------------------\n",
      "| Loading sequence | Time = 909.88 us\n",
      "-----------------------------------------\n",
      "size = 5017563\n",
      "-----------------------------------------\n",
      "| Loading positions | Time = 8.5701 ms\n",
      "-----------------------------------------\n",
      "size = 9684819\n",
      "-----------------------------------------\n",
      "| Loading reference sequence | Time = 1.4919 ms\n",
      "-----------------------------------------\n",
      "-----------------------------------------\n",
      "| Loading reference accumulative lengths | Time = 36.474 us\n",
      "-----------------------------------------\n",
      "\u001b[00m[2022-03-15 02:42:42.317] [jointLog] [info] done\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:42:42.317] [jointLog] [info] Index contained 4907 targets\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:42:42.318] [jointLog] [info] Number of decoys : 1\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:42:42.318] [jointLog] [info] First decoy index : 4906 \n",
      "\u001b[00m\n",
      "\n",
      "\n",
      "\n",
      "\u001b[32mprocessed\u001b[31m 500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 1000000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 1500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 2000000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 2500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 3000000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 3500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 4000001 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 4500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 5000000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 5500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 6000001 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 6500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 7000000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 7500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 8000000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 8500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 9000000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 9500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 10000000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 10500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 11000000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 11500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 12000000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 12500000 \u001b[32mfragments\u001b[0m\n",
      "hits: 11844193; hits per frag:  0.949629\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\u001b[00m[2022-03-15 02:43:04.897] [jointLog] [info] Thread saw mini-batch with a maximum of 0.28% zero probability fragments\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:43:04.898] [jointLog] [info] Thread saw mini-batch with a maximum of 0.30% zero probability fragments\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:43:04.898] [jointLog] [info] Thread saw mini-batch with a maximum of 0.26% zero probability fragments\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:43:04.899] [jointLog] [info] Thread saw mini-batch with a maximum of 0.26% zero probability fragments\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:43:04.903] [jointLog] [info] Thread saw mini-batch with a maximum of 0.28% zero probability fragments\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:43:04.905] [jointLog] [info] Thread saw mini-batch with a maximum of 0.26% zero probability fragments\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:43:04.907] [jointLog] [info] Thread saw mini-batch with a maximum of 0.32% zero probability fragments\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:43:04.909] [jointLog] [info] Thread saw mini-batch with a maximum of 0.28% zero probability fragments\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:43:04.926] [jointLog] [info] Computed 4521 rich equivalence classes for further processing\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:43:04.926] [jointLog] [info] Counted 11952118 total reads in the equivalence classes \n",
      "\u001b[00m\u001b[00m[2022-03-15 02:43:04.930] [jointLog] [info] Number of mappings discarded because of alignment score : 443265\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:43:04.930] [jointLog] [info] Number of fragments entirely discarded because of alignment score : 259906\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:43:04.930] [jointLog] [info] Number of fragments discarded because they are best-mapped to decoys : 48485\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:43:04.930] [jointLog] [info] Number of fragments discarded because they have only dovetail (discordant) mappings to valid targets : 0\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:43:04.930] [jointLog] [info] Mapping rate = 94.618%\n",
      "\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:43:04.930] [jointLog] [info] finished quantifyLibrary()\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:43:04.930] [jointLog] [info] Starting optimizer\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:43:04.933] [jointLog] [info] Marked 0 weighted equivalence classes as degenerate\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:43:04.934] [jointLog] [info] iteration = 0 | max rel diff. = 1853.46\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:43:04.995] [jointLog] [info] iteration = 100 | max rel diff. = 0\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:43:04.996] [jointLog] [info] Finished optimizer\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:43:04.996] [jointLog] [info] writing output \n",
      "\n",
      "\u001b[00mVersion Info: ### PLEASE UPGRADE SALMON ###\n",
      "### A newer version of salmon with important bug fixes and improvements is available. ####\n",
      "###\n",
      "The newest version, available at https://github.com/COMBINE-lab/salmon/releases\n",
      "contains new features, improvements, and bug fixes; please upgrade at your\n",
      "earliest convenience.\n",
      "###\n",
      "Sign up for the salmon mailing list to hear about new versions, features and updates at:\n",
      "https://oceangenomics.com/subscribe\n",
      "###### salmon (selective-alignment-based) v1.5.2\n",
      "### [ program ] => salmon \n",
      "### [ command ] => quant \n",
      "### [ index ] => { /home/jupyter/rnaseq-myco-notebook/data/reference/transcriptome_index }\n",
      "### [ libType ] => { SR }\n",
      "### [ unmatedReads ] => { data/trimmed/SRR13349129_1_trimmed.fastq.gz }\n",
      "### [ threads ] => { 8 }\n",
      "### [ validateMappings ] => { }\n",
      "### [ output ] => { data/quants/SRR13349129_quant }\n",
      "Logs will be written to data/quants/SRR13349129_quant/logs\n",
      "\u001b[00m[2022-03-15 02:43:05.499] [jointLog] [info] setting maxHashResizeThreads to 8\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:43:05.499] [jointLog] [info] Fragment incompatibility prior below threshold.  Incompatible fragments will be ignored.\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:43:05.499] [jointLog] [info] Usage of --validateMappings implies use of minScoreFraction. Since not explicitly specified, it is being set to 0.65\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:43:05.499] [jointLog] [info] Setting consensusSlack to selective-alignment default of 0.35.\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:43:05.499] [jointLog] [info] parsing read library format\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:43:05.499] [jointLog] [info] There is 1 library.\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:43:05.539] [jointLog] [info] Loading pufferfish index\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:43:05.539] [jointLog] [info] Loading dense pufferfish index.\n",
      "\u001b[00m-----------------------------------------\n",
      "| Loading contig table | Time = 1.8301 ms\n",
      "-----------------------------------------\n",
      "size = 10431\n",
      "-----------------------------------------\n",
      "| Loading contig offsets | Time = 99.749 us\n",
      "-----------------------------------------\n",
      "-----------------------------------------\n",
      "| Loading reference lengths | Time = 17.898 us\n",
      "-----------------------------------------\n",
      "-----------------------------------------\n",
      "| Loading mphf table | Time = 770.75 us\n",
      "-----------------------------------------\n",
      "size = 5330463\n",
      "Number of ones: 10430\n",
      "Number of ones per inventory item: 512\n",
      "Inventory entries filled: 21\n",
      "-----------------------------------------\n",
      "| Loading contig boundaries | Time = 14.343 ms\n",
      "-----------------------------------------\n",
      "size = 5330463\n",
      "-----------------------------------------\n",
      "| Loading sequence | Time = 1.1121 ms\n",
      "-----------------------------------------\n",
      "size = 5017563\n",
      "-----------------------------------------\n",
      "| Loading positions | Time = 9.5127 ms\n",
      "-----------------------------------------\n",
      "size = 9684819\n",
      "-----------------------------------------\n",
      "| Loading reference sequence | Time = 1.7121 ms\n",
      "-----------------------------------------\n",
      "-----------------------------------------\n",
      "| Loading reference accumulative lengths | Time = 34.349 us\n",
      "-----------------------------------------\n",
      "\u001b[00m[2022-03-15 02:43:05.568] [jointLog] [info] done\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:43:05.568] [jointLog] [info] Index contained 4907 targets\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:43:05.570] [jointLog] [info] Number of decoys : 1\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:43:05.570] [jointLog] [info] First decoy index : 4906 \n",
      "\u001b[00m\n",
      "\n",
      "\n",
      "\n",
      "\u001b[32mprocessed\u001b[31m 500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 1000001 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 1500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 2000000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 2500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 3000001 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 3500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 4000000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 4500001 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 5000000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 5500001 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 6000000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 6500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 7000000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 7500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 8000000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 8500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 9000000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 9500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 10000000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 10500001 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 11000000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 11500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 12000000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 12500001 \u001b[32mfragments\u001b[0m\n",
      "hits: 11907107; hits per frag:  0.953953\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\u001b[00m[2022-03-15 02:43:28.595] [jointLog] [info] Thread saw mini-batch with a maximum of 0.36% zero probability fragments\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:43:28.595] [jointLog] [info] Thread saw mini-batch with a maximum of 0.30% zero probability fragments\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:43:28.607] [jointLog] [info] Thread saw mini-batch with a maximum of 0.36% zero probability fragments\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:43:28.616] [jointLog] [info] Thread saw mini-batch with a maximum of 0.32% zero probability fragments\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:43:28.617] [jointLog] [info] Thread saw mini-batch with a maximum of 0.32% zero probability fragments\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:43:28.618] [jointLog] [info] Thread saw mini-batch with a maximum of 0.34% zero probability fragments\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:43:28.622] [jointLog] [info] Thread saw mini-batch with a maximum of 0.30% zero probability fragments\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:43:28.624] [jointLog] [info] Thread saw mini-batch with a maximum of 0.32% zero probability fragments\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:43:28.642] [jointLog] [info] Computed 4572 rich equivalence classes for further processing\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:43:28.642] [jointLog] [info] Counted 12250892 total reads in the equivalence classes \n",
      "\u001b[00m\u001b[00m[2022-03-15 02:43:28.645] [jointLog] [info] Number of mappings discarded because of alignment score : 443985\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:43:28.645] [jointLog] [info] Number of fragments entirely discarded because of alignment score : 265177\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:43:28.645] [jointLog] [info] Number of fragments discarded because they are best-mapped to decoys : 54924\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:43:28.645] [jointLog] [info] Number of fragments discarded because they have only dovetail (discordant) mappings to valid targets : 0\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:43:28.645] [jointLog] [info] Mapping rate = 95.1013%\n",
      "\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:43:28.645] [jointLog] [info] finished quantifyLibrary()\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:43:28.646] [jointLog] [info] Starting optimizer\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:43:28.648] [jointLog] [info] Marked 0 weighted equivalence classes as degenerate\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:43:28.649] [jointLog] [info] iteration = 0 | max rel diff. = 3769.81\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:43:28.711] [jointLog] [info] iteration = 100 | max rel diff. = 1.34867e-16\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:43:28.711] [jointLog] [info] Finished optimizer\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:43:28.711] [jointLog] [info] writing output \n",
      "\n",
      "\u001b[00mVersion Info: ### PLEASE UPGRADE SALMON ###\n",
      "### A newer version of salmon with important bug fixes and improvements is available. ####\n",
      "###\n",
      "The newest version, available at https://github.com/COMBINE-lab/salmon/releases\n",
      "contains new features, improvements, and bug fixes; please upgrade at your\n",
      "earliest convenience.\n",
      "###\n",
      "Sign up for the salmon mailing list to hear about new versions, features and updates at:\n",
      "https://oceangenomics.com/subscribe\n",
      "###### salmon (selective-alignment-based) v1.5.2\n",
      "### [ program ] => salmon \n",
      "### [ command ] => quant \n",
      "### [ index ] => { /home/jupyter/rnaseq-myco-notebook/data/reference/transcriptome_index }\n",
      "### [ libType ] => { SR }\n",
      "### [ unmatedReads ] => { data/trimmed/SRR13349130_1_trimmed.fastq.gz }\n",
      "### [ threads ] => { 8 }\n",
      "### [ validateMappings ] => { }\n",
      "### [ output ] => { data/quants/SRR13349130_quant }\n",
      "Logs will be written to data/quants/SRR13349130_quant/logs\n",
      "\u001b[00m[2022-03-15 02:43:29.241] [jointLog] [info] setting maxHashResizeThreads to 8\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:43:29.241] [jointLog] [info] Fragment incompatibility prior below threshold.  Incompatible fragments will be ignored.\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:43:29.241] [jointLog] [info] Usage of --validateMappings implies use of minScoreFraction. Since not explicitly specified, it is being set to 0.65\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:43:29.241] [jointLog] [info] Setting consensusSlack to selective-alignment default of 0.35.\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:43:29.241] [jointLog] [info] parsing read library format\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:43:29.241] [jointLog] [info] There is 1 library.\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:43:29.282] [jointLog] [info] Loading pufferfish index\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:43:29.282] [jointLog] [info] Loading dense pufferfish index.\n",
      "\u001b[00m-----------------------------------------\n",
      "| Loading contig table | Time = 1.8094 ms\n",
      "-----------------------------------------\n",
      "size = 10431\n",
      "-----------------------------------------\n",
      "| Loading contig offsets | Time = 90.804 us\n",
      "-----------------------------------------\n",
      "-----------------------------------------\n",
      "| Loading reference lengths | Time = 18.643 us\n",
      "-----------------------------------------\n",
      "-----------------------------------------\n",
      "| Loading mphf table | Time = 789.44 us\n",
      "-----------------------------------------\n",
      "size = 5330463\n",
      "Number of ones: 10430\n",
      "Number of ones per inventory item: 512\n",
      "Inventory entries filled: 21\n",
      "-----------------------------------------\n",
      "| Loading contig boundaries | Time = 13.353 ms\n",
      "-----------------------------------------\n",
      "size = 5330463\n",
      "-----------------------------------------\n",
      "| Loading sequence | Time = 992.24 us\n",
      "-----------------------------------------\n",
      "size = 5017563\n",
      "-----------------------------------------\n",
      "| Loading positions | Time = 9.3521 ms\n",
      "-----------------------------------------\n",
      "size = 9684819\n",
      "-----------------------------------------\n",
      "| Loading reference sequence | Time = 1.623 ms\n",
      "-----------------------------------------\n",
      "-----------------------------------------\n",
      "| Loading reference accumulative lengths | Time = 37.169 us\n",
      "-----------------------------------------\n",
      "\u001b[00m[2022-03-15 02:43:29.310] [jointLog] [info] done\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:43:29.310] [jointLog] [info] Index contained 4907 targets\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:43:29.312] [jointLog] [info] Number of decoys : 1\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:43:29.312] [jointLog] [info] First decoy index : 4906 \n",
      "\u001b[00m\n",
      "\n",
      "\n",
      "\n",
      "\u001b[32mprocessed\u001b[31m 500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 1000000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 1500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 2000002 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 2500002 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 3000000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 3500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 4000001 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 4500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 5000000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 5500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 6000000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 6500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 7000000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 7500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 8000001 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 8500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 9000000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 9500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 10000001 \u001b[32mfragments\u001b[0m\n",
      "hits: 9580684; hits per frag:  0.960316\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\u001b[00m[2022-03-15 02:43:47.477] [jointLog] [info] Thread saw mini-batch with a maximum of 0.36% zero probability fragments\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:43:47.478] [jointLog] [info] Thread saw mini-batch with a maximum of 0.38% zero probability fragments\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:43:47.479] [jointLog] [info] Thread saw mini-batch with a maximum of 0.32% zero probability fragments\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:43:47.480] [jointLog] [info] Thread saw mini-batch with a maximum of 0.30% zero probability fragments\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:43:47.483] [jointLog] [info] Thread saw mini-batch with a maximum of 0.36% zero probability fragments\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:43:47.485] [jointLog] [info] Thread saw mini-batch with a maximum of 0.32% zero probability fragments\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:43:47.489] [jointLog] [info] Thread saw mini-batch with a maximum of 0.30% zero probability fragments\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:43:47.490] [jointLog] [info] Thread saw mini-batch with a maximum of 0.34% zero probability fragments\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:43:47.508] [jointLog] [info] Computed 4484 rich equivalence classes for further processing\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:43:47.508] [jointLog] [info] Counted 9628756 total reads in the equivalence classes \n",
      "\u001b[00m\u001b[00m[2022-03-15 02:43:47.511] [jointLog] [info] Number of mappings discarded because of alignment score : 163164\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:43:47.511] [jointLog] [info] Number of fragments entirely discarded because of alignment score : 115960\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:43:47.511] [jointLog] [info] Number of fragments discarded because they are best-mapped to decoys : 44119\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:43:47.511] [jointLog] [info] Number of fragments discarded because they have only dovetail (discordant) mappings to valid targets : 0\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:43:47.511] [jointLog] [info] Mapping rate = 95.646%\n",
      "\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:43:47.511] [jointLog] [info] finished quantifyLibrary()\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:43:47.511] [jointLog] [info] Starting optimizer\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:43:47.514] [jointLog] [info] Marked 0 weighted equivalence classes as degenerate\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:43:47.515] [jointLog] [info] iteration = 0 | max rel diff. = 1583.88\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:43:47.576] [jointLog] [info] iteration = 100 | max rel diff. = 1.35198e-16\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:43:47.576] [jointLog] [info] Finished optimizer\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:43:47.576] [jointLog] [info] writing output \n",
      "\n",
      "\u001b[00mVersion Info: ### PLEASE UPGRADE SALMON ###\n",
      "### A newer version of salmon with important bug fixes and improvements is available. ####\n",
      "###\n",
      "The newest version, available at https://github.com/COMBINE-lab/salmon/releases\n",
      "contains new features, improvements, and bug fixes; please upgrade at your\n",
      "earliest convenience.\n",
      "###\n",
      "Sign up for the salmon mailing list to hear about new versions, features and updates at:\n",
      "https://oceangenomics.com/subscribe\n",
      "###### salmon (selective-alignment-based) v1.5.2\n",
      "### [ program ] => salmon \n",
      "### [ command ] => quant \n",
      "### [ index ] => { /home/jupyter/rnaseq-myco-notebook/data/reference/transcriptome_index }\n",
      "### [ libType ] => { SR }\n",
      "### [ unmatedReads ] => { data/trimmed/SRR13349131_1_trimmed.fastq.gz }\n",
      "### [ threads ] => { 8 }\n",
      "### [ validateMappings ] => { }\n",
      "### [ output ] => { data/quants/SRR13349131_quant }\n",
      "Logs will be written to data/quants/SRR13349131_quant/logs\n",
      "\u001b[00m[2022-03-15 02:43:48.478] [jointLog] [info] setting maxHashResizeThreads to 8\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:43:48.478] [jointLog] [info] Fragment incompatibility prior below threshold.  Incompatible fragments will be ignored.\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:43:48.478] [jointLog] [info] Usage of --validateMappings implies use of minScoreFraction. Since not explicitly specified, it is being set to 0.65\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:43:48.478] [jointLog] [info] Setting consensusSlack to selective-alignment default of 0.35.\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:43:48.478] [jointLog] [info] parsing read library format\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:43:48.478] [jointLog] [info] There is 1 library.\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:43:48.519] [jointLog] [info] Loading pufferfish index\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:43:48.519] [jointLog] [info] Loading dense pufferfish index.\n",
      "\u001b[00m-----------------------------------------\n",
      "| Loading contig table | Time = 1.8009 ms\n",
      "-----------------------------------------\n",
      "size = 10431\n",
      "-----------------------------------------\n",
      "| Loading contig offsets | Time = 79.016 us\n",
      "-----------------------------------------\n",
      "-----------------------------------------\n",
      "| Loading reference lengths | Time = 16.797 us\n",
      "-----------------------------------------\n",
      "-----------------------------------------\n",
      "| Loading mphf table | Time = 754.06 us\n",
      "-----------------------------------------\n",
      "size = 5330463\n",
      "Number of ones: 10430\n",
      "Number of ones per inventory item: 512\n",
      "Inventory entries filled: 21\n",
      "-----------------------------------------\n",
      "| Loading contig boundaries | Time = 13.167 ms\n",
      "-----------------------------------------\n",
      "size = 5330463\n",
      "-----------------------------------------\n",
      "| Loading sequence | Time = 876.11 us\n",
      "-----------------------------------------\n",
      "size = 5017563\n",
      "-----------------------------------------\n",
      "| Loading positions | Time = 8.6625 ms\n",
      "-----------------------------------------\n",
      "size = 9684819\n",
      "-----------------------------------------\n",
      "| Loading reference sequence | Time = 1.5159 ms\n",
      "-----------------------------------------\n",
      "-----------------------------------------\n",
      "| Loading reference accumulative lengths | Time = 33.991 us\n",
      "-----------------------------------------\n",
      "\u001b[00m[2022-03-15 02:43:48.546] [jointLog] [info] done\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:43:48.546] [jointLog] [info] Index contained 4907 targets\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:43:48.547] [jointLog] [info] Number of decoys : 1\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:43:48.547] [jointLog] [info] First decoy index : 4906 \n",
      "\u001b[00m\n",
      "\n",
      "\n",
      "\n",
      "\u001b[32mprocessed\u001b[31m 500001 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 1000000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 1500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 2000000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 2500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 3000000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 3500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 4000000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 4500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 5000000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 5500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 6000000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 6500001 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 7000000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 7500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 8000000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 8500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 9000000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 9500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 10000000 \u001b[32mfragments\u001b[0m\n",
      "hits: 9635740; hits per frag:  0.967577\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\u001b[00m[2022-03-15 02:44:07.111] [jointLog] [info] Thread saw mini-batch with a maximum of 0.36% zero probability fragments\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:44:07.164] [jointLog] [info] Thread saw mini-batch with a maximum of 0.36% zero probability fragments\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:44:07.165] [jointLog] [info] Thread saw mini-batch with a maximum of 0.38% zero probability fragments\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:44:07.166] [jointLog] [info] Thread saw mini-batch with a maximum of 0.36% zero probability fragments\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:44:07.169] [jointLog] [info] Thread saw mini-batch with a maximum of 0.36% zero probability fragments\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:44:07.172] [jointLog] [info] Thread saw mini-batch with a maximum of 0.36% zero probability fragments\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:44:07.172] [jointLog] [info] Thread saw mini-batch with a maximum of 0.34% zero probability fragments\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:44:07.173] [jointLog] [info] Thread saw mini-batch with a maximum of 0.36% zero probability fragments\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:44:07.191] [jointLog] [info] Computed 4550 rich equivalence classes for further processing\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:44:07.191] [jointLog] [info] Counted 10028030 total reads in the equivalence classes \n",
      "\u001b[00m\u001b[00m[2022-03-15 02:44:07.195] [jointLog] [info] Number of mappings discarded because of alignment score : 168772\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:44:07.195] [jointLog] [info] Number of fragments entirely discarded because of alignment score : 123620\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:44:07.195] [jointLog] [info] Number of fragments discarded because they are best-mapped to decoys : 50352\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:44:07.195] [jointLog] [info] Number of fragments discarded because they have only dovetail (discordant) mappings to valid targets : 0\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:44:07.195] [jointLog] [info] Mapping rate = 96.1719%\n",
      "\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:44:07.195] [jointLog] [info] finished quantifyLibrary()\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:44:07.195] [jointLog] [info] Starting optimizer\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:44:07.198] [jointLog] [info] Marked 0 weighted equivalence classes as degenerate\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:44:07.198] [jointLog] [info] iteration = 0 | max rel diff. = 1633.38\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:44:07.260] [jointLog] [info] iteration = 100 | max rel diff. = 1.84245e-16\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:44:07.260] [jointLog] [info] Finished optimizer\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:44:07.260] [jointLog] [info] writing output \n",
      "\n",
      "\u001b[00mVersion Info: ### PLEASE UPGRADE SALMON ###\n",
      "### A newer version of salmon with important bug fixes and improvements is available. ####\n",
      "###\n",
      "The newest version, available at https://github.com/COMBINE-lab/salmon/releases\n",
      "contains new features, improvements, and bug fixes; please upgrade at your\n",
      "earliest convenience.\n",
      "###\n",
      "Sign up for the salmon mailing list to hear about new versions, features and updates at:\n",
      "https://oceangenomics.com/subscribe\n",
      "###### salmon (selective-alignment-based) v1.5.2\n",
      "### [ program ] => salmon \n",
      "### [ command ] => quant \n",
      "### [ index ] => { /home/jupyter/rnaseq-myco-notebook/data/reference/transcriptome_index }\n",
      "### [ libType ] => { SR }\n",
      "### [ unmatedReads ] => { data/trimmed/SRR13349132_1_trimmed.fastq.gz }\n",
      "### [ threads ] => { 8 }\n",
      "### [ validateMappings ] => { }\n",
      "### [ output ] => { data/quants/SRR13349132_quant }\n",
      "Logs will be written to data/quants/SRR13349132_quant/logs\n",
      "\u001b[00m[2022-03-15 02:44:07.722] [jointLog] [info] setting maxHashResizeThreads to 8\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:44:07.722] [jointLog] [info] Fragment incompatibility prior below threshold.  Incompatible fragments will be ignored.\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:44:07.722] [jointLog] [info] Usage of --validateMappings implies use of minScoreFraction. Since not explicitly specified, it is being set to 0.65\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:44:07.722] [jointLog] [info] Setting consensusSlack to selective-alignment default of 0.35.\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:44:07.722] [jointLog] [info] parsing read library format\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:44:07.722] [jointLog] [info] There is 1 library.\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:44:07.763] [jointLog] [info] Loading pufferfish index\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:44:07.763] [jointLog] [info] Loading dense pufferfish index.\n",
      "\u001b[00m-----------------------------------------\n",
      "| Loading contig table | Time = 1.788 ms\n",
      "-----------------------------------------\n",
      "size = 10431\n",
      "-----------------------------------------\n",
      "| Loading contig offsets | Time = 102.92 us\n",
      "-----------------------------------------\n",
      "-----------------------------------------\n",
      "| Loading reference lengths | Time = 15.362 us\n",
      "-----------------------------------------\n",
      "-----------------------------------------\n",
      "| Loading mphf table | Time = 702.13 us\n",
      "-----------------------------------------\n",
      "size = 5330463\n",
      "Number of ones: 10430\n",
      "Number of ones per inventory item: 512\n",
      "Inventory entries filled: 21\n",
      "-----------------------------------------\n",
      "| Loading contig boundaries | Time = 12.891 ms\n",
      "-----------------------------------------\n",
      "size = 5330463\n",
      "-----------------------------------------\n",
      "| Loading sequence | Time = 902.73 us\n",
      "-----------------------------------------\n",
      "size = 5017563\n",
      "-----------------------------------------\n",
      "| Loading positions | Time = 8.6202 ms\n",
      "-----------------------------------------\n",
      "size = 9684819\n",
      "-----------------------------------------\n",
      "| Loading reference sequence | Time = 1.4828 ms\n",
      "-----------------------------------------\n",
      "-----------------------------------------\n",
      "| Loading reference accumulative lengths | Time = 38.198 us\n",
      "-----------------------------------------\n",
      "\u001b[00m[2022-03-15 02:44:07.790] [jointLog] [info] done\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:44:07.790] [jointLog] [info] Index contained 4907 targets\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:44:07.791] [jointLog] [info] Number of decoys : 1\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:44:07.791] [jointLog] [info] First decoy index : 4906 \n",
      "\u001b[00m\n",
      "\n",
      "\n",
      "\n",
      "\u001b[32mprocessed\u001b[31m 500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 1000000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 1500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 2000000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 2500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 3000000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 3500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 4000000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 4500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 5000000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 5500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 6000000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 6500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 7000000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 7500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 8000001 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 8500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 9000000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 9500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 10000000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 10500001 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 11000001 \u001b[32mfragments\u001b[0m\n",
      "hits: 10524260; hits per frag:  0.960703\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\u001b[00m[2022-03-15 02:44:28.107] [jointLog] [info] Thread saw mini-batch with a maximum of 0.36% zero probability fragments\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:44:28.107] [jointLog] [info] Thread saw mini-batch with a maximum of 0.34% zero probability fragments\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:44:28.107] [jointLog] [info] Thread saw mini-batch with a maximum of 0.38% zero probability fragments\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:44:28.107] [jointLog] [info] Thread saw mini-batch with a maximum of 0.38% zero probability fragments\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:44:28.116] [jointLog] [info] Thread saw mini-batch with a maximum of 0.38% zero probability fragments\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:44:28.119] [jointLog] [info] Thread saw mini-batch with a maximum of 0.36% zero probability fragments\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:44:28.122] [jointLog] [info] Thread saw mini-batch with a maximum of 0.34% zero probability fragments\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:44:28.124] [jointLog] [info] Thread saw mini-batch with a maximum of 0.36% zero probability fragments\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:44:28.142] [jointLog] [info] Computed 4570 rich equivalence classes for further processing\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:44:28.142] [jointLog] [info] Counted 10811229 total reads in the equivalence classes \n",
      "\u001b[00m\u001b[00m[2022-03-15 02:44:28.145] [jointLog] [info] Number of mappings discarded because of alignment score : 182403\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:44:28.145] [jointLog] [info] Number of fragments entirely discarded because of alignment score : 138010\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:44:28.145] [jointLog] [info] Number of fragments discarded because they are best-mapped to decoys : 62855\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:44:28.145] [jointLog] [info] Number of fragments discarded because they have only dovetail (discordant) mappings to valid targets : 0\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:44:28.145] [jointLog] [info] Mapping rate = 95.4795%\n",
      "\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:44:28.145] [jointLog] [info] finished quantifyLibrary()\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:44:28.146] [jointLog] [info] Starting optimizer\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:44:28.148] [jointLog] [info] Marked 0 weighted equivalence classes as degenerate\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:44:28.149] [jointLog] [info] iteration = 0 | max rel diff. = 1726.4\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:44:28.211] [jointLog] [info] iteration = 100 | max rel diff. = 1.68972e-16\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:44:28.212] [jointLog] [info] Finished optimizer\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:44:28.212] [jointLog] [info] writing output \n",
      "\n",
      "\u001b[00mVersion Info: ### PLEASE UPGRADE SALMON ###\n",
      "### A newer version of salmon with important bug fixes and improvements is available. ####\n",
      "###\n",
      "The newest version, available at https://github.com/COMBINE-lab/salmon/releases\n",
      "contains new features, improvements, and bug fixes; please upgrade at your\n",
      "earliest convenience.\n",
      "###\n",
      "Sign up for the salmon mailing list to hear about new versions, features and updates at:\n",
      "https://oceangenomics.com/subscribe\n",
      "###### salmon (selective-alignment-based) v1.5.2\n",
      "### [ program ] => salmon \n",
      "### [ command ] => quant \n",
      "### [ index ] => { /home/jupyter/rnaseq-myco-notebook/data/reference/transcriptome_index }\n",
      "### [ libType ] => { SR }\n",
      "### [ unmatedReads ] => { data/trimmed/SRR13349133_1_trimmed.fastq.gz }\n",
      "### [ threads ] => { 8 }\n",
      "### [ validateMappings ] => { }\n",
      "### [ output ] => { data/quants/SRR13349133_quant }\n",
      "Logs will be written to data/quants/SRR13349133_quant/logs\n",
      "\u001b[00m[2022-03-15 02:44:28.967] [jointLog] [info] setting maxHashResizeThreads to 8\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:44:28.967] [jointLog] [info] Fragment incompatibility prior below threshold.  Incompatible fragments will be ignored.\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:44:28.967] [jointLog] [info] Usage of --validateMappings implies use of minScoreFraction. Since not explicitly specified, it is being set to 0.65\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:44:28.967] [jointLog] [info] Setting consensusSlack to selective-alignment default of 0.35.\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:44:28.967] [jointLog] [info] parsing read library format\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:44:28.967] [jointLog] [info] There is 1 library.\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:44:29.008] [jointLog] [info] Loading pufferfish index\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:44:29.008] [jointLog] [info] Loading dense pufferfish index.\n",
      "\u001b[00m-----------------------------------------\n",
      "| Loading contig table | Time = 1.8275 ms\n",
      "-----------------------------------------\n",
      "size = 10431\n",
      "-----------------------------------------\n",
      "| Loading contig offsets | Time = 112.44 us\n",
      "-----------------------------------------\n",
      "-----------------------------------------\n",
      "| Loading reference lengths | Time = 16.588 us\n",
      "-----------------------------------------\n",
      "-----------------------------------------\n",
      "| Loading mphf table | Time = 845.35 us\n",
      "-----------------------------------------\n",
      "size = 5330463\n",
      "Number of ones: 10430\n",
      "Number of ones per inventory item: 512\n",
      "Inventory entries filled: 21\n",
      "-----------------------------------------\n",
      "| Loading contig boundaries | Time = 13.301 ms\n",
      "-----------------------------------------\n",
      "size = 5330463\n",
      "-----------------------------------------\n",
      "| Loading sequence | Time = 948.57 us\n",
      "-----------------------------------------\n",
      "size = 5017563\n",
      "-----------------------------------------\n",
      "| Loading positions | Time = 8.6322 ms\n",
      "-----------------------------------------\n",
      "size = 9684819\n",
      "-----------------------------------------\n",
      "| Loading reference sequence | Time = 1.5243 ms\n",
      "-----------------------------------------\n",
      "-----------------------------------------\n",
      "| Loading reference accumulative lengths | Time = 32.79 us\n",
      "-----------------------------------------\n",
      "\u001b[00m[2022-03-15 02:44:29.036] [jointLog] [info] done\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:44:29.036] [jointLog] [info] Index contained 4907 targets\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:44:29.037] [jointLog] [info] Number of decoys : 1\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:44:29.037] [jointLog] [info] First decoy index : 4906 \n",
      "\u001b[00m\n",
      "\n",
      "\n",
      "\n",
      "\u001b[32mprocessed\u001b[31m 500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 1000000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 1500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 2000000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 2500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 3000000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 3500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 4000001 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 4500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 5000000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 5500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 6000000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 6500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 7000000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 7500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 8000000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 8500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 9000000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 9500000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 10000000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 10500001 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 11000000 \u001b[32mfragments\u001b[0m\n",
      "\u001b[32mprocessed\u001b[31m 11500000 \u001b[32mfragments\u001b[0m\n",
      "hits: 11058584; hits per frag:  0.964305\u001b[00m[2022-03-15 02:44:49.723] [jointLog] [info] Thread saw mini-batch with a maximum of 0.44% zero probability fragments\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:44:49.724] [jointLog] [info] Thread saw mini-batch with a maximum of 0.46% zero probability fragments\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:44:49.731] [jointLog] [info] Thread saw mini-batch with a maximum of 0.48% zero probability fragments\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:44:49.733] [jointLog] [info] Thread saw mini-batch with a maximum of 0.40% zero probability fragments\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:44:49.738] [jointLog] [info] Thread saw mini-batch with a maximum of 0.42% zero probability fragments\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:44:49.739] [jointLog] [info] Thread saw mini-batch with a maximum of 0.42% zero probability fragments\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:44:49.746] [jointLog] [info] Thread saw mini-batch with a maximum of 0.42% zero probability fragments\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:44:49.751] [jointLog] [info] Thread saw mini-batch with a maximum of 0.44% zero probability fragments\n",
      "\u001b[00m\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\u001b[00m[2022-03-15 02:44:49.770] [jointLog] [info] Computed 4585 rich equivalence classes for further processing\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:44:49.770] [jointLog] [info] Counted 11060584 total reads in the equivalence classes \n",
      "\u001b[00m\u001b[00m[2022-03-15 02:44:49.773] [jointLog] [info] Number of mappings discarded because of alignment score : 185118\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:44:49.773] [jointLog] [info] Number of fragments entirely discarded because of alignment score : 144946\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:44:49.773] [jointLog] [info] Number of fragments discarded because they are best-mapped to decoys : 69884\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:44:49.773] [jointLog] [info] Number of fragments discarded because they have only dovetail (discordant) mappings to valid targets : 0\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:44:49.773] [jointLog] [info] Mapping rate = 95.9172%\n",
      "\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:44:49.773] [jointLog] [info] finished quantifyLibrary()\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:44:49.773] [jointLog] [info] Starting optimizer\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:44:49.776] [jointLog] [info] Marked 0 weighted equivalence classes as degenerate\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:44:49.777] [jointLog] [info] iteration = 0 | max rel diff. = 1755\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:44:49.840] [jointLog] [info] iteration = 100 | max rel diff. = 1.01982e-12\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:44:49.841] [jointLog] [info] Finished optimizer\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:44:49.841] [jointLog] [info] writing output \n",
      "\n",
      "\u001b[00m"
     ]
    }
   ],
   "source": [
    "!cat data/raw_fastq/list_of_accesionIDS.txt | xargs -I {} salmon quant -i /home/jupyter/rnaseq-myco-notebook/data/reference/transcriptome_index -l SR -r \"data/trimmed/{}_1_trimmed.fastq.gz\" -p 8 --validateMappings -o \"data/quants/{}_quant\"\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### STEP 11: Report the top 10 most highly expressed genes in the samples"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Top 10 most highly expressed genes in each wild-type sample.\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 138,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Name\tLength\tEffectiveLength\tTPM\tNumReads\n",
      "rna-BB28_RS07080\t117\t3.947\t366889.819745\t16507.000\n",
      "rna-BB28_RS07075\t3116\t2866.000\t234649.688207\t7665384.000\n",
      "rna-BB28_RS07070\t1516\t1266.000\t130544.614501\t1883781.000\n",
      "rna-BB28_RS17330\t369\t119.000\t72555.739838\t98414.000\n",
      "gene-BB28_RS02220\t204\t9.377\t8373.726433\t895.000\n",
      "gene-BB28_RS20695\t231\t14.032\t6858.936071\t1097.000\n",
      "rna-BB28_RS09710\t405\t155.000\t5466.039721\t9657.000\n",
      "gene-BB28_RS18745\t300\t51.326\t3680.213081\t2153.000\n",
      "gene-BB28_RS18945\t222\t12.150\t3530.894602\t489.000\n",
      "gene-BB28_RS24750\t102\t3.543\t3392.339786\t137.000\n",
      "sort: write failed: 'standard output': Broken pipe\n",
      "sort: write error\n",
      "rna-BB28_RS07080\t117\t3.947\t348464.132877\t16529.000\n",
      "rna-BB28_RS07075\t3116\t2866.000\t224110.719521\t7718494.000\n",
      "rna-BB28_RS07070\t1516\t1266.000\t132619.568495\t2017600.000\n",
      "rna-BB28_RS17330\t369\t119.000\t76318.688035\t109137.000\n",
      "gene-BB28_RS02220\t204\t9.377\t8723.549796\t983.000\n",
      "gene-BB28_RS20695\t231\t14.032\t7312.353823\t1233.000\n",
      "rna-BB28_RS09710\t405\t155.000\t6225.085179\t11595.000\n",
      "gene-BB28_RS18745\t300\t51.326\t3998.213678\t2466.000\n",
      "gene-BB28_RS18945\t222\t12.150\t3773.735759\t551.000\n",
      "gene-BB28_RS24750\t102\t3.543\t3452.551034\t147.000\n",
      "sort: write failed: 'standard output': Broken pipe\n",
      "sort: write error\n",
      "rna-BB28_RS07080\t117\t3.947\t471373.491225\t23945.000\n",
      "rna-BB28_RS07075\t3116\t2866.000\t210066.635757\t7747975.000\n",
      "rna-BB28_RS07070\t1516\t1266.000\t109003.973827\t1775951.000\n",
      "rna-BB28_RS17330\t369\t119.000\t58941.596177\t90266.000\n",
      "gene-BB28_RS02220\t204\t9.377\t5709.495815\t689.000\n",
      "gene-BB28_RS20695\t231\t14.032\t5576.508929\t1007.000\n",
      "rna-BB28_RS09710\t405\t155.000\t4726.928047\t9429.000\n",
      "gene-BB28_RS18745\t300\t51.326\t2812.920437\t1858.000\n",
      "gene-BB28_RS18945\t222\t12.150\t2705.198053\t423.000\n",
      "gene-BB28_RS24750\t102\t3.543\t2653.670233\t121.000\n",
      "sort: write failed: 'standard output': Broken pipe\n",
      "sort: write error\n",
      "rna-BB28_RS07080\t117\t3.947\t454456.088270\t24081.000\n",
      "rna-BB28_RS07075\t3116\t2866.000\t202000.055452\t7771691.000\n",
      "rna-BB28_RS07070\t1516\t1266.000\t111292.575649\t1891419.000\n",
      "rna-BB28_RS17330\t369\t119.000\t63375.473250\t101241.000\n",
      "gene-BB28_RS02220\t204\t9.377\t5711.820118\t719.000\n",
      "rna-BB28_RS09710\t405\t155.000\t5621.535273\t11697.000\n",
      "gene-BB28_RS20695\t231\t14.032\t5568.978149\t1049.000\n",
      "gene-BB28_RS24750\t102\t3.543\t3132.672144\t149.000\n",
      "gene-BB28_RS18745\t300\t51.326\t2984.021469\t2056.000\n",
      "gene-BB28_RS18945\t222\t12.150\t2887.664276\t471.000\n",
      "sort: write failed: 'standard output': Broken pipe\n",
      "sort: write error\n",
      "rna-BB28_RS07080\t117\t3.947\t355610.619291\t16373.000\n",
      "rna-BB28_RS07075\t3116\t2866.000\t274208.716011\t9166768.000\n",
      "rna-BB28_RS07070\t1516\t1266.000\t127630.663764\t1884723.000\n",
      "rna-BB28_RS17330\t369\t119.000\t68728.626182\t95399.000\n",
      "gene-BB28_RS02220\t204\t9.377\t7442.161455\t814.000\n",
      "gene-BB28_RS20695\t231\t14.032\t6268.684864\t1026.000\n",
      "rna-BB28_RS09710\t405\t155.000\t4239.015431\t7664.000\n",
      "gene-BB28_RS18945\t222\t12.150\t3760.815719\t533.000\n",
      "gene-BB28_RS24750\t102\t3.543\t3411.747177\t141.000\n",
      "gene-BB28_RS18745\t300\t51.326\t2908.082049\t1741.000\n",
      "sort: write failed: 'standard output': Broken pipe\n",
      "sort: write error\n",
      "rna-BB28_RS07080\t117\t3.947\t342846.188962\t16587.000\n",
      "rna-BB28_RS07075\t3116\t2866.000\t261522.353002\t9186684.000\n",
      "rna-BB28_RS07070\t1516\t1266.000\t130870.033318\t2030709.000\n",
      "rna-BB28_RS17330\t369\t119.000\t72868.976078\t106283.000\n",
      "gene-BB28_RS02220\t204\t9.377\t7473.996129\t859.000\n",
      "gene-BB28_RS20695\t231\t14.032\t6268.055884\t1078.000\n",
      "rna-BB28_RS09710\t405\t155.000\t4786.846193\t9094.000\n",
      "gene-BB28_RS18945\t222\t12.150\t3773.776554\t562.000\n",
      "gene-BB28_RS18745\t300\t51.326\t3228.515158\t2031.000\n",
      "gene-BB28_RS24750\t102\t3.543\t2809.328861\t122.000\n",
      "sort: write failed: 'standard output': Broken pipe\n",
      "sort: write error\n"
     ]
    }
   ],
   "source": [
    "!head data/quants/SRR13349122_quant/quant.sf -n 1\n",
    "!sort -nrk 4,4 data/quants/SRR13349122_quant/quant.sf | head -10\n",
    "!sort -nrk 4,4 data/quants/SRR13349123_quant/quant.sf | head -10\n",
    "!sort -nrk 4,4 data/quants/SRR13349124_quant/quant.sf | head -10\n",
    "!sort -nrk 4,4 data/quants/SRR13349125_quant/quant.sf | head -10\n",
    "!sort -nrk 4,4 data/quants/SRR13349126_quant/quant.sf | head -10\n",
    "!sort -nrk 4,4 data/quants/SRR13349127_quant/quant.sf | head -10\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Top 10 most highly expressed genes in the double lysogen samples.\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 139,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Name\tLength\tEffectiveLength\tTPM\tNumReads\n",
      "rna-BB28_RS07080\t117\t3.947\t453962.554944\t24652.000\n",
      "rna-BB28_RS07075\t3116\t2866.000\t246655.204473\t9725319.000\n",
      "rna-BB28_RS07070\t1516\t1266.000\t92462.501485\t1610411.000\n",
      "rna-BB28_RS17330\t369\t119.000\t77568.389870\t126990.000\n",
      "gene-BB28_RS20695\t231\t14.032\t4926.415147\t951.000\n",
      "rna-BB28_RS09710\t405\t155.000\t4308.287043\t9187.000\n",
      "gene-BB28_RS02220\t204\t9.377\t3790.573785\t489.000\n",
      "gene-BB28_RS14885\t195\t8.348\t2603.559715\t299.000\n",
      "gene-BB28_RS06975\t216\t11.099\t2541.095506\t388.000\n",
      "gene-BB28_RS21780\t213\t10.625\t2223.501125\t325.000\n",
      "sort: write failed: 'standard output': Broken pipe\n",
      "sort: write error\n",
      "rna-BB28_RS07080\t117\t3.947\t437397.002569\t24834.000\n",
      "rna-BB28_RS07075\t3116\t2866.000\t237318.253525\t9783257.000\n",
      "rna-BB28_RS07070\t1516\t1266.000\t95599.035600\t1740858.000\n",
      "rna-BB28_RS17330\t369\t119.000\t83089.537568\t142223.000\n",
      "gene-BB28_RS20695\t231\t14.032\t5400.553107\t1090.000\n",
      "rna-BB28_RS09710\t405\t155.000\t5016.367275\t11184.000\n",
      "gene-BB28_RS02220\t204\t9.377\t4122.229249\t556.000\n",
      "gene-BB28_RS14885\t195\t8.348\t3206.404708\t385.000\n",
      "gene-BB28_RS06975\t216\t11.099\t2800.000048\t447.000\n",
      "gene-BB28_RS21780\t213\t10.625\t2434.210804\t372.000\n",
      "sort: write failed: 'standard output': Broken pipe\n",
      "sort: write error\n",
      "rna-BB28_RS07080\t117\t3.947\t400790.315794\t15594.000\n",
      "rna-BB28_RS07075\t3116\t2866.000\t280304.877022\t7918673.000\n",
      "rna-BB28_RS07070\t1516\t1266.000\t94999.070212\t1185492.000\n",
      "rna-BB28_RS17330\t369\t119.000\t76649.660848\t89909.000\n",
      "rna-BB28_RS09710\t405\t155.000\t4707.960295\t7193.000\n",
      "gene-BB28_RS02220\t204\t9.377\t4046.313376\t374.000\n",
      "gene-BB28_RS20695\t231\t14.032\t3998.226912\t553.000\n",
      "gene-BB28_RS14885\t195\t8.348\t2540.004334\t209.000\n",
      "gene-BB28_RS20665\t1293\t1043.000\t2491.034277\t25610.000\n",
      "gene-BB28_RS24750\t102\t3.543\t2118.862026\t74.000\n",
      "sort: write failed: 'standard output': Broken pipe\n",
      "sort: write error\n",
      "rna-BB28_RS07080\t117\t3.947\t382840.293956\t15621.000\n",
      "rna-BB28_RS07075\t3116\t2866.000\t273993.659363\t8117329.000\n",
      "rna-BB28_RS07070\t1516\t1266.000\t98392.106467\t1287628.000\n",
      "rna-BB28_RS17330\t369\t119.000\t81017.983597\t99661.000\n",
      "rna-BB28_RS09710\t405\t155.000\t5293.207433\t8481.000\n",
      "gene-BB28_RS02220\t204\t9.377\t4425.825879\t429.000\n",
      "gene-BB28_RS20695\t231\t14.032\t4357.210126\t632.000\n",
      "gene-BB28_RS14885\t195\t8.348\t3476.630578\t300.000\n",
      "gene-BB28_RS20665\t1293\t1043.000\t2925.185374\t31538.000\n",
      "gene-BB28_RS18745\t300\t51.326\t2320.215300\t1231.000\n",
      "sort: write failed: 'standard output': Broken pipe\n",
      "sort: write error\n",
      "rna-BB28_RS07080\t117\t3.947\t413489.577066\t19297.000\n",
      "rna-BB28_RS07075\t3116\t2866.000\t262171.629861\t8883669.000\n",
      "rna-BB28_RS07070\t1516\t1266.000\t78623.588158\t1176839.000\n",
      "rna-BB28_RS17330\t369\t119.000\t68320.253231\t96123.000\n",
      "rna-BB28_RS09710\t405\t155.000\t5474.804605\t10033.000\n",
      "gene-BB28_RS20695\t231\t14.032\t4894.557292\t812.000\n",
      "gene-BB28_RS02220\t204\t9.377\t4185.245841\t464.000\n",
      "gene-BB28_RS14885\t195\t8.348\t3627.323410\t358.000\n",
      "gene-BB28_RS20665\t1293\t1043.000\t3421.571432\t42193.000\n",
      "gene-BB28_RS18945\t222\t12.150\t2749.667941\t395.000\n",
      "sort: write failed: 'standard output': Broken pipe\n",
      "sort: write error\n",
      "rna-BB28_RS07080\t117\t3.947\t398339.037417\t19555.000\n",
      "rna-BB28_RS07075\t3116\t2866.000\t249742.504095\t8901821.000\n",
      "rna-BB28_RS07070\t1516\t1266.000\t80654.134701\t1269903.000\n",
      "rna-BB28_RS17330\t369\t119.000\t71668.251678\t106068.000\n",
      "rna-BB28_RS09710\t405\t155.000\t6362.987227\t12266.000\n",
      "gene-BB28_RS20695\t231\t14.032\t5415.137332\t945.000\n",
      "gene-BB28_RS02220\t204\t9.377\t4544.636394\t530.000\n",
      "gene-BB28_RS14885\t195\t8.348\t4093.666525\t425.000\n",
      "gene-BB28_RS20665\t1293\t1043.000\t3958.562219\t51349.000\n",
      "gene-BB28_RS06975\t216\t11.099\t2999.278403\t414.000\n",
      "sort: write failed: 'standard output': Broken pipe\n",
      "sort: write error\n"
     ]
    }
   ],
   "source": [
    "!head data/quants/SRR13349122_quant/quant.sf -n 1\n",
    "!sort -nrk 4,4 data/quants/SRR13349128_quant/quant.sf | head -10\n",
    "!sort -nrk 4,4 data/quants/SRR13349129_quant/quant.sf | head -10\n",
    "!sort -nrk 4,4 data/quants/SRR13349130_quant/quant.sf | head -10\n",
    "!sort -nrk 4,4 data/quants/SRR13349131_quant/quant.sf | head -10\n",
    "!sort -nrk 4,4 data/quants/SRR13349132_quant/quant.sf | head -10\n",
    "!sort -nrk 4,4 data/quants/SRR13349133_quant/quant.sf | head -10\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### STEP 12: Report the expression of a putative acyl-ACP desaturase (BB28_RS16545) that was downregulated in the double lysogen relative to wild-type\n",
    "A acyl-transferase was reported to be downregulated in the double lysogen as shown in the table of the top 20 upregulated and downregulated genes from the paper describing the study.\n",
    "![RNA-Seq workflow](images/table-cushman.png)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Use `grep` to report the expression in the wild-type sample. The fields in the Salmon `quant.sf` file are as follows. The level of expression is reported in the Transcripts Per Million (`TPM`) and number of reads (`NumReads`) fields:  \n",
    "`Name    Length  EffectiveLength TPM     NumReads`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 140,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "gene-BB28_RS16545\t987\t737.000\t53.330147\t448.000\n",
      "gene-BB28_RS16545\t987\t737.000\t50.245685\t445.000\n",
      "gene-BB28_RS16545\t987\t737.000\t42.489598\t403.000\n",
      "gene-BB28_RS16545\t987\t737.000\t44.574169\t441.000\n",
      "gene-BB28_RS16545\t987\t737.000\t52.578979\t452.000\n",
      "gene-BB28_RS16545\t987\t737.000\t53.912280\t487.000\n"
     ]
    }
   ],
   "source": [
    "!grep 'BB28_RS16545' data/quants/SRR13349122_quant/quant.sf\n",
    "!grep 'BB28_RS16545' data/quants/SRR13349123_quant/quant.sf\n",
    "!grep 'BB28_RS16545' data/quants/SRR13349124_quant/quant.sf\n",
    "!grep 'BB28_RS16545' data/quants/SRR13349125_quant/quant.sf\n",
    "!grep 'BB28_RS16545' data/quants/SRR13349126_quant/quant.sf\n",
    "!grep 'BB28_RS16545' data/quants/SRR13349127_quant/quant.sf\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Use `grep` to report the expression in the double lysogen sample. The fields in the Salmon `quant.sf` file are as follows. The level of expression is reported in the Transcripts Per Million (`TPM`) and number of reads (`NumReads`) fields:  \n",
    "`Name    Length  EffectiveLength TPM     NumReads`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 141,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "gene-BB28_RS16545\t987\t737.000\t4.635462\t47.000\n",
      "gene-BB28_RS16545\t987\t737.000\t6.508868\t69.000\n",
      "gene-BB28_RS16545\t987\t737.000\t5.093176\t37.000\n",
      "gene-BB28_RS16545\t987\t737.000\t5.906750\t45.000\n",
      "gene-BB28_RS16545\t987\t737.000\t4.934811\t43.000\n",
      "gene-BB28_RS16545\t987\t737.000\t6.218665\t57.000\n"
     ]
    }
   ],
   "source": [
    "!grep 'BB28_RS16545' data/quants/SRR13349128_quant/quant.sf\n",
    "!grep 'BB28_RS16545' data/quants/SRR13349129_quant/quant.sf\n",
    "!grep 'BB28_RS16545' data/quants/SRR13349130_quant/quant.sf\n",
    "!grep 'BB28_RS16545' data/quants/SRR13349131_quant/quant.sf\n",
    "!grep 'BB28_RS16545' data/quants/SRR13349132_quant/quant.sf\n",
    "!grep 'BB28_RS16545' data/quants/SRR13349133_quant/quant.sf"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### STEP 12: Combine Genecounts to a Single Genecount File\n",
    "Commonly, the readcounts for each sample are combined into a single table, where the rows contain the gene ID, and the columns identify the sample."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 144,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Version Info: ### PLEASE UPGRADE SALMON ###\n",
      "### A newer version of salmon with important bug fixes and improvements is available. ####\n",
      "###\n",
      "The newest version, available at https://github.com/COMBINE-lab/salmon/releases\n",
      "contains new features, improvements, and bug fixes; please upgrade at your\n",
      "earliest convenience.\n",
      "###\n",
      "Sign up for the salmon mailing list to hear about new versions, features and updates at:\n",
      "https://oceangenomics.com/subscribe\n",
      "###\u001b[00m[2022-03-15 02:50:59.621] [mergeLog] [info] samples: [ data/quants/SRR13349122_quant, data/quants/SRR13349123_quant, data/quants/SRR13349124_quant, data/quants/SRR13349125_quant, data/quants/SRR13349126_quant, data/quants/SRR13349127_quant, data/quants/SRR13349128_quant, data/quants/SRR13349129_quant, data/quants/SRR13349130_quant, data/quants/SRR13349131_quant, data/quants/SRR13349132_quant, data/quants/SRR13349133_quant ]\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:50:59.621] [mergeLog] [info] sample names : [ SRR13349122_quant, SRR13349123_quant, SRR13349124_quant, SRR13349125_quant, SRR13349126_quant, SRR13349127_quant, SRR13349128_quant, SRR13349129_quant, SRR13349130_quant, SRR13349131_quant, SRR13349132_quant, SRR13349133_quant ]\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:50:59.621] [mergeLog] [info] output column : NUMREADS\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:50:59.621] [mergeLog] [info] output file : data/quants/merged_quants.txt\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:50:59.621] [mergeLog] [info] Parsing data/quants/SRR13349122_quant/quant.sf\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:50:59.630] [mergeLog] [info] Parsing data/quants/SRR13349123_quant/quant.sf\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:50:59.639] [mergeLog] [info] Parsing data/quants/SRR13349124_quant/quant.sf\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:50:59.649] [mergeLog] [info] Parsing data/quants/SRR13349125_quant/quant.sf\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:50:59.659] [mergeLog] [info] Parsing data/quants/SRR13349126_quant/quant.sf\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:50:59.669] [mergeLog] [info] Parsing data/quants/SRR13349127_quant/quant.sf\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:50:59.678] [mergeLog] [info] Parsing data/quants/SRR13349128_quant/quant.sf\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:50:59.687] [mergeLog] [info] Parsing data/quants/SRR13349129_quant/quant.sf\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:50:59.695] [mergeLog] [info] Parsing data/quants/SRR13349130_quant/quant.sf\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:50:59.706] [mergeLog] [info] Parsing data/quants/SRR13349131_quant/quant.sf\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:50:59.716] [mergeLog] [info] Parsing data/quants/SRR13349132_quant/quant.sf\n",
      "\u001b[00m\u001b[00m[2022-03-15 02:50:59.726] [mergeLog] [info] Parsing data/quants/SRR13349133_quant/quant.sf\n",
      "\u001b[00m"
     ]
    }
   ],
   "source": [
    "##first merge salmon files by number of reads.\n",
    "!salmon quantmerge --column numreads --quants data/quants/*_quant -o data/quants/merged_quants.txt\n",
    "##optinally we can rename the columns\n",
    "!sed -i \"1s/.*/Name\\tSRR13349122\\tSRR13349123\\tSRR13349124\\tSRR13349125\\tSRR13349126\\tSRR13349127\\tSRR13349128\\tSRR13349129\\tSRR13349130\\tSRR13349131\\tSRR13349132\\tSRR13349133/\" data/quants/merged_quants.txt\n",
    "\n",
    "##for further formatting, it may be easier in our r-code to later merge\n",
    "##if we remove the gene- and rna- prefix\n",
    "!sed -i \"s/gene-//\" data/quants/merged_quants.txt\n",
    "!sed -i \"s/rna-//\" data/quants/merged_quants.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "tags": []
   },
   "source": [
    "## <a name=\"workflow\">Additional Workflows</a>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now that you have read counts per gene, feel free to explore the R workflow which creates plots and analyses using these readcount files, or try other alternate workflows for creating read count files, such as using snakemake."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "\n",
    "[Workflow One:](Tutorial_1.ipynb) A short introduction to downloading and mapping sequences to a transcriptome using Trimmomatic and Salmon. Here is a link to the YouTube video demonstrating the tutorial: <https://www.youtube.com/watch?v=NG1U7D4l31o&t=26s>.\n",
    "\n",
    "[Workflow One (Snakemake):](Tutorial_2_Snakemake.ipynb) Using snakemake to run workflow one.\n",
    "\n",
    "[Extended Workflow One:](Tutorial_1_Extended.ipynb) An extended version of workflow one. Once you have got your feet wet, you can retry workflow one with this extended version that covers the entire dataset, and includes elaboration such as using SRA tools for sequence downloading, and examples of running batches of fastq files through the pipeline. This workflow may take around an hour to run.\n",
    "\n",
    "[Workflow Two:](Tutorial_3_R.ipynb) Using Deseq2 and R to conduct clustering and differential gene expression analysis.\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "![RNA-Seq workflow](images/RNA-Seq_Notebook_Homepage.png)"
   ]
  }
 ],
 "metadata": {
  "environment": {
   "kernel": "python3",
   "name": "r-cpu.4-1.m92",
   "type": "gcloud",
   "uri": "gcr.io/deeplearning-platform-release/r-cpu.4-1:m92"
  },
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
